<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { color: #00769e; background-color: #f1f3f5; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #00769e; } /* Normal */
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #657422; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #00769e; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #00769e; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #00769e; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #5e5e5e; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>


  <!--radix_placeholder_meta_tags-->
  <title>G000191: Preprocessing the G000396 mini-bulk data set</title>



  <!--  https://schema.org/Article -->
  <meta property="article:published" itemprop="datePublished" content="2023-11-14"/>
  <meta property="article:created" itemprop="dateCreated" content="2023-11-14"/>
  <meta name="article:author" content="Peter Hickey"/>

  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="G000191: Preprocessing the G000396 mini-bulk data set"/>
  <meta property="og:type" content="article"/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="G000191"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="G000191: Preprocessing the G000396 mini-bulk data set"/>

  <!--/radix_placeholder_meta_tags-->
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","date","editor_options"]}},"value":[{"type":"character","attributes":{},"value":["Preprocessing the G000396 mini-bulk data set"]},{"type":"NULL"},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url","affiliation","affiliation_url"]}},"value":[{"type":"character","attributes":{},"value":["Peter Hickey"]},{"type":"character","attributes":{},"value":["https://peterhickey.org"]},{"type":"character","attributes":{},"value":["Cellular Genomics Projects Team (formerly SCORE), WEHI"]},{"type":"character","attributes":{},"value":["https://www.wehi.edu.au/people/rory-bowden/4536/wehi-advanced-genomics-facility"]}]}]},{"type":"character","attributes":{},"value":["2023-11-14"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["chunk_output_type"]}},"value":[{"type":"character","attributes":{},"value":["console"]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <script src="site_libs/autocomplete-0.37.1/autocomplete.min.js"></script>
  <script src="site_libs/fuse-6.4.1/fuse.min.js"></script>

  <script type="application/javascript">

  function getMeta(metaName) {
    var metas = document.getElementsByTagName('meta');
    for (let i = 0; i < metas.length; i++) {
      if (metas[i].getAttribute('name') === metaName) {
        return metas[i].getAttribute('content');
      }
    }
    return '';
  }

  function offsetURL(url) {
    var offset = getMeta('distill:offset');
    return offset ? offset + '/' + url : url;
  }

  function createFuseIndex() {

    // create fuse index
    var options = {
      keys: [
        { name: 'title', weight: 20 },
        { name: 'categories', weight: 15 },
        { name: 'description', weight: 10 },
        { name: 'contents', weight: 5 },
      ],
      ignoreLocation: true,
      threshold: 0
    };
    var fuse = new window.Fuse([], options);

    // fetch the main search.json
    return fetch(offsetURL('search.json'))
      .then(function(response) {
        if (response.status == 200) {
          return response.json().then(function(json) {
            // index main articles
            json.articles.forEach(function(article) {
              fuse.add(article);
            });
            // download collections and index their articles
            return Promise.all(json.collections.map(function(collection) {
              return fetch(offsetURL(collection)).then(function(response) {
                if (response.status === 200) {
                  return response.json().then(function(articles) {
                    articles.forEach(function(article) {
                      fuse.add(article);
                    });
                  })
                } else {
                  return Promise.reject(
                    new Error('Unexpected status from search index request: ' +
                              response.status)
                  );
                }
              });
            })).then(function() {
              return fuse;
            });
          });

        } else {
          return Promise.reject(
            new Error('Unexpected status from search index request: ' +
                        response.status)
          );
        }
      });
  }

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // get search element (bail if we don't have one)
    var searchEl = window.document.getElementById('distill-search');
    if (!searchEl)
      return;

    createFuseIndex()
      .then(function(fuse) {

        // make search box visible
        searchEl.classList.remove('hidden');

        // initialize autocomplete
        var options = {
          autoselect: true,
          hint: false,
          minLength: 2,
        };
        window.autocomplete(searchEl, options, [{
          source: function(query, callback) {
            const searchOptions = {
              isCaseSensitive: false,
              shouldSort: true,
              minMatchCharLength: 2,
              limit: 10,
            };
            var results = fuse.search(query, searchOptions);
            callback(results
              .map(function(result) { return result.item; })
            );
          },
          templates: {
            suggestion: function(suggestion) {
              var img = suggestion.preview && Object.keys(suggestion.preview).length > 0
                ? `<img src="${offsetURL(suggestion.preview)}"</img>`
                : '';
              var html = `
                <div class="search-item">
                  <h3>${suggestion.title}</h3>
                  <div class="search-item-description">
                    ${suggestion.description || ''}
                  </div>
                  <div class="search-item-preview">
                    ${img}
                  </div>
                </div>
              `;
              return html;
            }
          }
        }]).on('autocomplete:selected', function(event, suggestion) {
          window.location.href = offsetURL(suggestion.path);
        });
        // remove inline display style on autocompleter (we want to
        // manage responsive display via css)
        $('.algolia-autocomplete').css("display", "");
      })
      .catch(function(error) {
        console.log(error);
      });

  });

  </script>

  <style type="text/css">

  .nav-search {
    font-size: x-small;
  }

  /* Algolioa Autocomplete */

  .algolia-autocomplete {
    display: inline-block;
    margin-left: 10px;
    vertical-align: sub;
    background-color: white;
    color: black;
    padding: 6px;
    padding-top: 8px;
    padding-bottom: 0;
    border-radius: 6px;
    border: 1px #0F2E3D solid;
    width: 180px;
  }


  @media screen and (max-width: 768px) {
    .distill-site-nav .algolia-autocomplete {
      display: none;
      visibility: hidden;
    }
    .distill-site-nav.responsive .algolia-autocomplete {
      display: inline-block;
      visibility: visible;
    }
    .distill-site-nav.responsive .algolia-autocomplete .aa-dropdown-menu {
      margin-left: 0;
      width: 400px;
      max-height: 400px;
    }
  }

  .algolia-autocomplete .aa-input, .algolia-autocomplete .aa-hint {
    width: 90%;
    outline: none;
    border: none;
  }

  .algolia-autocomplete .aa-hint {
    color: #999;
  }
  .algolia-autocomplete .aa-dropdown-menu {
    width: 550px;
    max-height: 70vh;
    overflow-x: visible;
    overflow-y: scroll;
    padding: 5px;
    margin-top: 3px;
    margin-left: -150px;
    background-color: #fff;
    border-radius: 5px;
    border: 1px solid #999;
    border-top: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion {
    cursor: pointer;
    padding: 5px 4px;
    border-bottom: 1px solid #eee;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion:last-of-type {
    border-bottom: none;
    margin-bottom: 2px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item {
    overflow: hidden;
    font-size: 0.8em;
    line-height: 1.4em;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item h3 {
    font-size: 1rem;
    margin-block-start: 0;
    margin-block-end: 5px;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-description {
    display: inline-block;
    overflow: hidden;
    height: 2.8em;
    width: 80%;
    margin-right: 4%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview {
    display: inline-block;
    width: 15%;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img {
    height: 3em;
    width: auto;
    display: none;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion .search-item-preview img[src] {
    display: initial;
  }

  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
    background-color: #eee;
  }
  .algolia-autocomplete .aa-dropdown-menu .aa-suggestion em {
    font-weight: bold;
    font-style: normal;
  }

  </style>


  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
    font-size: 100%;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  hr.section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    margin: 0px;
  }


  d-byline {
    border-top: none;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
    border-top: none;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  /* tweak for Pandoc numbered line within distill */
  d-article pre.numberSource code > span {
      left: -2em;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // separator
    var separator = '<hr class="section-separator" style="clear: both"/>';
    // prepend separator above appendix
    $('.d-byline').before(separator);
    $('.d-article').before(separator);

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme, except when numbering line
    // in code chunk
    $('pre:not(.numberLines) code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        var author_name = front_matter.authors[i].author
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', author_name ? 'ORCID ID for ' + author_name : 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        const citeChild = $(this).children()[0]
        // Do not process if @xyz has been used without escaping and without bibliography activated
        // https://github.com/rstudio/distill/issues/466
        if (citeChild === undefined) return true

        if (citeChild.nodeName == "D-FOOTNOTE") {
          var fn = citeChild
          $(this).html(fn.shadowRoot.querySelector("sup"))
          $(this).id = fn.id
          fn.remove()
        }
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          // Could use CSS.escape too here, we insure backward compatibility in navigator
          return "<p>" + $('div[id="ref-' + ref + '"]').html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // fix footnotes in tables (#411)
      // replacing broken distill.pub feature
      $('table d-footnote').each(function() {
        // we replace internal showAtNode methode which is triggered when hovering a footnote
        this.hoverBox.showAtNode = function(node) {
          // ported from https://github.com/distillpub/template/pull/105/files
          calcOffset = function(elem) {
              let x = elem.offsetLeft;
              let y = elem.offsetTop;
              // Traverse upwards until an `absolute` element is found or `elem`
              // becomes null.
              while (elem = elem.offsetParent && elem.style.position != 'absolute') {
                  x += elem.offsetLeft;
                  y += elem.offsetTop;
              }

              return { left: x, top: y };
          }
          // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetTop
          const bbox = node.getBoundingClientRect();
          const offset = calcOffset(node);
          this.show([offset.left + bbox.width, offset.top + bbox.height]);
        }
      })

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      // ignore leaflet img layers (#106)
      figures = figures.filter(':not(img[class*="leaflet"])')
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace(/^index[.]html/, "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
  <link href="site_libs/panelset-0.2.6/panelset.css" rel="stylesheet" />
  <script src="site_libs/panelset-0.2.6/panelset.js"></script>
  <script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
  <link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
  <script src="site_libs/datatables-binding-0.30/datatables.js"></script>
  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
  <link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
  <script src="site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
  <link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
  <script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Preprocessing the G000396 mini-bulk data set","authors":[{"author":"Peter Hickey","authorURL":"https://peterhickey.org","affiliation":"Cellular Genomics Projects Team (formerly SCORE), WEHI","affiliationURL":"https://www.wehi.edu.au/people/rory-bowden/4536/wehi-advanced-genomics-facility","orcidID":""}],"publishedDate":"2023-11-14T00:00:00.000+00:00","citationText":"Hickey, 2023"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">G000191</a>
<a href="index.html">Introduction</a>
<a href="G000396_Danu.preprocess.html">Preprocessing</a>
<a href="G000396_Danu.differential_expression.html">Differential expression</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Supplementary files
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="../output/scPipe/">scPipe reports</a>
</div>
</div>
</div>
<div class="nav-right">
<a href="https://github.com/WEHISCORE/G000396_Danu">
<i class="fa fa-github" aria-hidden="true"></i>
</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Preprocessing the G000396 mini-bulk data set</h1>

<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->

</div>

<div class="d-byline">
  Peter Hickey <a href="https://peterhickey.org" class="uri">https://peterhickey.org</a> (Cellular Genomics Projects Team (formerly SCORE), WEHI)<a href="https://www.wehi.edu.au/people/rory-bowden/4536/wehi-advanced-genomics-facility" class="uri">https://www.wehi.edu.au/people/rory-bowden/4536/wehi-advanced-genomics-facility</a>
  
<br/>2023-11-14
</div>

<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#setting-up-the-data" id="toc-setting-up-the-data">Setting up the data</a>
<ul>
<li><a href="#experimental-design" id="toc-experimental-design">Experimental design</a>
<ul>
<li><a href="#introduction" id="toc-introduction">Introduction</a></li>
</ul></li>
<li><a href="#incorporating-gene-based-annotation" id="toc-incorporating-gene-based-annotation">Incorporating gene-based annotation</a></li>
</ul></li>
<li><a href="#read-mapping-metrics" id="toc-read-mapping-metrics">Read mapping metrics</a>
<ul>
<li><a href="#motivation" id="toc-motivation">Motivation</a></li>
<li><a href="#analysis" id="toc-analysis">Analysis</a></li>
<li><a href="#summary" id="toc-summary">Summary</a></li>
</ul></li>
<li><a href="#analysis-of-empty-wells" id="toc-analysis-of-empty-wells">Analysis of empty wells</a>
<ul>
<li><a href="#background" id="toc-background">Background</a></li>
<li><a href="#analysis-1" id="toc-analysis-1">Analysis</a></li>
<li><a href="#summary-1" id="toc-summary-1">Summary</a></li>
</ul></li>
<li><a href="#have-we-sequenced-enough" id="toc-have-we-sequenced-enough">Have we sequenced enough?</a>
<ul>
<li><a href="#motivation-1" id="toc-motivation-1">Motivation</a></li>
<li><a href="#analysis-2" id="toc-analysis-2">Analysis</a></li>
<li><a href="#summary-2" id="toc-summary-2">Summary</a></li>
</ul></li>
<li><a href="#umi-counts-vs.-read-counts" id="toc-umi-counts-vs.-read-counts">UMI counts vs.read counts</a>
<ul>
<li><a href="#motivation-2" id="toc-motivation-2">Motivation</a></li>
<li><a href="#analysis-3" id="toc-analysis-3">Analysis</a></li>
<li><a href="#summary-3" id="toc-summary-3">Summary</a></li>
</ul></li>
<li><a href="#quality-control-of-cells" id="toc-quality-control-of-cells">Quality control of cells</a>
<ul>
<li><a href="#defining-the-quality-control-metrics" id="toc-defining-the-quality-control-metrics">Defining the quality control metrics</a></li>
<li><a href="#visualizing-the-qc-metrics" id="toc-visualizing-the-qc-metrics">Visualizing the QC metrics</a>
<ul>
<li><a href="#individual-technical-replicates" id="toc-individual-technical-replicates">Individual technical replicates</a></li>
<li><a href="#aggregated-technical-replicates" id="toc-aggregated-technical-replicates">Aggregated technical replicates</a></li>
</ul></li>
<li><a href="#identifying-outliers-by-each-metric" id="toc-identifying-outliers-by-each-metric">Identifying outliers by each metric</a></li>
<li><a href="#summary-4" id="toc-summary-4">Summary</a></li>
</ul></li>
<li><a href="#examining-gene-level-metrics" id="toc-examining-gene-level-metrics">Examining gene-level metrics</a>
<ul>
<li><a href="#inspecting-the-most-highly-expressed-genes" id="toc-inspecting-the-most-highly-expressed-genes">Inspecting the most highly expressed genes</a></li>
</ul></li>
<li><a href="#mds-plot" id="toc-mds-plot">MDS plot</a>
<ul>
<li><a href="#invividual-technical-replicates" id="toc-invividual-technical-replicates">Invividual technical replicates</a></li>
<li><a href="#aggregated-technical-replicates-1" id="toc-aggregated-technical-replicates-1">Aggregated technical replicates</a></li>
<li><a href="#summary-5" id="toc-summary-5">Summary</a></li>
</ul></li>
<li><a href="#concluding-remarks" id="toc-concluding-remarks">Concluding remarks</a></li>
<li><a href="#additional-information" id="toc-additional-information">Additional information</a></li>
<li><a href="#session-info" id="toc-session-info">Session info</a></li>
</ul>
</nav>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>SingleCellExperiment</span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://here.r-lib.org/'>here</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://scales.r-lib.org'>scales</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://ggplot2.tidyverse.org'>ggplot2</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://wilkelab.org/cowplot/'>cowplot</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://patchwork.data-imaginist.com'>patchwork</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/slowkow/ggrepel'>ggrepel</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://gt.rstudio.com'>gt</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='http://bioconductor.org/packages/scater/'>scater</a></span><span class='op'>)</span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://bioinf.wehi.edu.au/limma/'>limma</a></span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'>knitr</span><span class='fu'>::</span><span class='va'><a href='https://rdrr.io/pkg/knitr/man/opts_chunk.html'>opts_chunk</a></span><span class='op'>$</span><span class='fu'>set</span><span class='op'>(</span>fig.path <span class='op'>=</span> <span class='st'>"G000396_Danu.preprocess_files/"</span><span class='op'>)</span></span>
<span></span>
<span><span class='co'># NOTE: The following is a workaround to the lack of support for tabsets in </span></span>
<span><span class='co'>#       distill (see https://github.com/rstudio/distill/issues/11 and </span></span>
<span><span class='co'>#       https://github.com/rstudio/distill/issues/11#issuecomment-692142414 in </span></span>
<span><span class='co'>#       particular).</span></span>
<span><span class='fu'>xaringanExtra</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/xaringanExtra/man/panelset.html'>use_panelset</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<h1 id="setting-up-the-data">Setting up the data</h1>
<p>We start from the count matrices created in <a href="../code/scPipe.R%60"><code>code/scPipe.R</code></a>, specifically the <a href="../data/SCEs/G000396_Danu.UMI_deduped.scPipe.SCE.rds">UMI counts (a.k.a. UMI-deduplicated) data</a> and <a href="../data/SCEs/G000396_Danu.not_UMI_deduped.scPipe.SCE.rds">read counts (a.k.a. not UMI-deduplicated) data</a>.</p>
<aside>
UMI stands for unique molecular identifier.
</aside>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>sce_deduped</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"SCEs"</span>, <span class='st'>"G000396_Danu.UMI_deduped.scPipe.SCE.rds"</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>sce_not_deduped</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>readRDS</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"SCEs"</span>, <span class='st'>"G000396_Danu.not_UMI_deduped.scPipe.SCE.rds"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce_not_deduped</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>sub</a></span><span class='op'>(</span></span>
<span>  <span class='st'>"\\.not_UMI_deduped"</span>,</span>
<span>  <span class='st'>""</span>,</span>
<span>  <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce_not_deduped</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/stopifnot.html'>stopifnot</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/base/identical.html'>identical</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce_deduped</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce_not_deduped</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  <span class='fu'><a href='https://rdrr.io/r/base/identical.html'>identical</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce_deduped</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce_not_deduped</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span> </span>
<span></span>
<span><span class='co'># Combine UMI and read counts a single SCE.</span></span>
<span><span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='va'>sce_deduped</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce_deduped</span>, <span class='st'>"counts"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce_not_deduped</span>, <span class='st'>"counts"</span><span class='op'>)</span></span>
<span><span class='co'># NOTE: Nullify some now-unrequired data.</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"counts"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span></span>
<span><span class='va'>sce</span><span class='op'>$</span><span class='va'>UMI_deduped</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span></span></code></pre>
</div>
</details>
</div>
<p>We combine these into a single object<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a> containing both the UMI counts and the read counts for 5,724 genes and 376 samples.</p>
<h2 id="experimental-design">Experimental design</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Initial tidying of sample metadata</span></span>
<span><span class='va'>sce</span><span class='op'>$</span><span class='va'>plate_number</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>plate_number</span><span class='op'>)</span></span>
<span><span class='va'>sce</span><span class='op'>$</span><span class='va'>well_position</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span><span class='op'>$</span><span class='va'>well_position</span>,</span>
<span>  <span class='fu'><a href='https://rdrr.io/r/base/unlist.html'>unlist</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>LETTERS</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>16</span><span class='op'>]</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='fl'>1</span><span class='op'>:</span><span class='fl'>24</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_type</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_type</span><span class='op'>)</span></span>
<span><span class='va'>sce</span><span class='op'>$</span><span class='va'>original_plate</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>original_plate</span><span class='op'>)</span></span>
<span><span class='va'>sce</span><span class='op'>$</span><span class='va'>cell_line</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span><span class='op'>$</span><span class='va'>cell_line</span>,</span>
<span>  levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"GID1KO"</span>, <span class='st'>"GID2KO"</span>, <span class='st'>"GID7KO"</span>, <span class='st'>"GID8KO"</span>, <span class='st'>"GID9KO"</span>, <span class='st'>"WT"</span>, <span class='st'>"N/A"</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>sce</span><span class='op'>$</span><span class='va'>timepoint</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span><span class='op'>$</span><span class='va'>timepoint</span>,</span>
<span>  levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"Day_"</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>3</span>, <span class='fl'>6</span>, <span class='fl'>9</span>, <span class='fl'>12</span><span class='op'>)</span><span class='op'>)</span>, <span class='st'>"N/A"</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>sce</span><span class='op'>$</span><span class='va'>biological_replicate</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span><span class='op'>$</span><span class='va'>biological_replicate</span>,</span>
<span>  levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>5</span>, <span class='st'>"N/A"</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>sce</span><span class='op'>$</span><span class='va'>technical_replicate</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span><span class='op'>$</span><span class='va'>technical_replicate</span>, </span>
<span>  levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span>, <span class='st'>"N/A"</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sequencing_run</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sequencing_run</span><span class='op'>)</span></span>
<span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sample</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/interaction.html'>interaction</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span><span class='op'>$</span><span class='va'>cell_line</span>, </span>
<span>  <span class='va'>sce</span><span class='op'>$</span><span class='va'>timepoint</span>, </span>
<span>  <span class='va'>sce</span><span class='op'>$</span><span class='va'>biological_replicate</span>, </span>
<span>  drop <span class='op'>=</span> <span class='cn'>TRUE</span>, </span>
<span>  lex.order <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<h3 id="introduction">Introduction</h3>
<p>This is a large and somewhat complex experiment.
There are 120 biological samples, each done in technical duplicate for a total of 240 mini-bulk libraries.
There are also136 negative control empty libraries<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>.
For now, we will retain the negative control empty libraries mini-bulk libraries for QC purposes.
The tables below summarise the samples in this experiment.</p>
<aside>
The <code>Day_3</code> and <code>Day_6</code> samples were originally plated together on one 96-well plate (<code>Plate A</code>) while the <code>Day_9</code> and <code>Day_12</code> were originally plated together on another 96-well plate (<code>Plate B</code>).
</aside>
<div class="panelset">
<section id="technical-replicates" class="panel">
<h4>Technical replicates</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>tech_reps_tbl</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/group_by.html'>group_by</a></span><span class='op'>(</span><span class='va'>cell_line</span>, <span class='va'>timepoint</span>, <span class='va'>biological_replicate</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/count.html'>count</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>|&gt;</span> </span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/group_by.html'>group_by</a></span><span class='op'>(</span><span class='va'>cell_line</span>, <span class='va'>timepoint</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://gt.rstudio.com/reference/gt.html'>gt</a></span><span class='op'>(</span></span>
<span>  <span class='va'>tech_reps_tbl</span>,</span>
<span>  caption <span class='op'>=</span> <span class='fu'><a href='https://gt.rstudio.com/reference/md.html'>md</a></span><span class='op'>(</span></span>
<span>    <span class='st'>"Number of technical replicates per `cell_line`-`timepoint`-`biological_replicate` combination."</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://gt.rstudio.com/reference/fmt_number.html'>fmt_number</a></span><span class='op'>(</span>columns <span class='op'>=</span> <span class='va'>n</span>, decimals <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://gt.rstudio.com/reference/summary_rows.html'>summary_rows</a></span><span class='op'>(</span></span>
<span>    columns <span class='op'>=</span> <span class='va'>n</span>, </span>
<span>    fns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>subtotal</span> <span class='op'>~</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>.</span><span class='op'>)</span><span class='op'>)</span>, </span>
<span>    fmt <span class='op'>=</span> <span class='op'>~</span> <span class='fu'><a href='https://gt.rstudio.com/reference/fmt_number.html'>fmt_number</a></span><span class='op'>(</span><span class='va'>.</span>, decimals <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://gt.rstudio.com/reference/grand_summary_rows.html'>grand_summary_rows</a></span><span class='op'>(</span></span>
<span>    columns <span class='op'>=</span> <span class='va'>n</span>, </span>
<span>    fns <span class='op'>=</span> <span class='va'>total</span> <span class='op'>~</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>.</span><span class='op'>)</span>, </span>
<span>    fmt <span class='op'>=</span> <span class='op'>~</span> <span class='fu'><a href='https://gt.rstudio.com/reference/fmt_number.html'>fmt_number</a></span><span class='op'>(</span><span class='va'>.</span>, decimals <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://gt.rstudio.com/reference/tab_options.html'>tab_options</a></span><span class='op'>(</span></span>
<span>    summary_row.background.color <span class='op'>=</span> <span class='st'>"lightgrey"</span>,</span>
<span>    grand_summary_row.background.color <span class='op'>=</span> <span class='st'>"darkgrey"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div id="phpbwgvmdd" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#phpbwgvmdd table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#phpbwgvmdd thead, #phpbwgvmdd tbody, #phpbwgvmdd tfoot, #phpbwgvmdd tr, #phpbwgvmdd td, #phpbwgvmdd th {
  border-style: none;
}

#phpbwgvmdd p {
  margin: 0;
  padding: 0;
}

#phpbwgvmdd .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#phpbwgvmdd .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#phpbwgvmdd .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#phpbwgvmdd .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#phpbwgvmdd .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#phpbwgvmdd .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#phpbwgvmdd .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#phpbwgvmdd .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#phpbwgvmdd .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#phpbwgvmdd .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#phpbwgvmdd .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#phpbwgvmdd .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#phpbwgvmdd .gt_spanner_row {
  border-bottom-style: hidden;
}

#phpbwgvmdd .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#phpbwgvmdd .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#phpbwgvmdd .gt_from_md > :first-child {
  margin-top: 0;
}

#phpbwgvmdd .gt_from_md > :last-child {
  margin-bottom: 0;
}

#phpbwgvmdd .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#phpbwgvmdd .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#phpbwgvmdd .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#phpbwgvmdd .gt_row_group_first td {
  border-top-width: 2px;
}

#phpbwgvmdd .gt_row_group_first th {
  border-top-width: 2px;
}

#phpbwgvmdd .gt_summary_row {
  color: #333333;
  background-color: #D3D3D3;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#phpbwgvmdd .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#phpbwgvmdd .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#phpbwgvmdd .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#phpbwgvmdd .gt_grand_summary_row {
  color: #FFFFFF;
  background-color: #A9A9A9;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#phpbwgvmdd .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#phpbwgvmdd .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#phpbwgvmdd .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#phpbwgvmdd .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#phpbwgvmdd .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#phpbwgvmdd .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#phpbwgvmdd .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#phpbwgvmdd .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#phpbwgvmdd .gt_left {
  text-align: left;
}

#phpbwgvmdd .gt_center {
  text-align: center;
}

#phpbwgvmdd .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#phpbwgvmdd .gt_font_normal {
  font-weight: normal;
}

#phpbwgvmdd .gt_font_bold {
  font-weight: bold;
}

#phpbwgvmdd .gt_font_italic {
  font-style: italic;
}

#phpbwgvmdd .gt_super {
  font-size: 65%;
}

#phpbwgvmdd .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#phpbwgvmdd .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#phpbwgvmdd .gt_indent_1 {
  text-indent: 5px;
}

#phpbwgvmdd .gt_indent_2 {
  text-indent: 10px;
}

#phpbwgvmdd .gt_indent_3 {
  text-indent: 15px;
}

#phpbwgvmdd .gt_indent_4 {
  text-indent: 20px;
}

#phpbwgvmdd .gt_indent_5 {
  text-indent: 25px;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <caption><span id="tab:unnamed-chunk-3">Table 1: </span>Number of technical replicates per <code>cell_line</code>-<code>timepoint</code>-<code>biological_replicate</code> combination.</caption>
  <thead>
    
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id=""></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="biological_replicate">biological_replicate</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID1KO - Day_3">GID1KO - Day_3</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_1" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_3 stub_1_1 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID1KO - Day_3 stub_1_1 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_2" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_3 stub_1_2 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID1KO - Day_3 stub_1_2 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_3" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_3 stub_1_3 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID1KO - Day_3 stub_1_3 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_4" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_3 stub_1_4 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID1KO - Day_3 stub_1_4 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_5" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_3 stub_1_5 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID1KO - Day_3 stub_1_5 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID1KO - Day_3_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID1KO - Day_3 summary_stub_GID1KO - Day_3_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID1KO - Day_3 summary_stub_GID1KO - Day_3_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID1KO - Day_6">GID1KO - Day_6</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_6" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_6 stub_1_6 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID1KO - Day_6 stub_1_6 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_7" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_6 stub_1_7 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID1KO - Day_6 stub_1_7 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_8" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_6 stub_1_8 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID1KO - Day_6 stub_1_8 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_9" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_6 stub_1_9 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID1KO - Day_6 stub_1_9 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_10" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_6 stub_1_10 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID1KO - Day_6 stub_1_10 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID1KO - Day_6_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID1KO - Day_6 summary_stub_GID1KO - Day_6_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID1KO - Day_6 summary_stub_GID1KO - Day_6_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID1KO - Day_9">GID1KO - Day_9</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_11" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_9 stub_1_11 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID1KO - Day_9 stub_1_11 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_12" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_9 stub_1_12 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID1KO - Day_9 stub_1_12 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_13" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_9 stub_1_13 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID1KO - Day_9 stub_1_13 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_14" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_9 stub_1_14 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID1KO - Day_9 stub_1_14 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_15" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_9 stub_1_15 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID1KO - Day_9 stub_1_15 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID1KO - Day_9_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID1KO - Day_9 summary_stub_GID1KO - Day_9_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID1KO - Day_9 summary_stub_GID1KO - Day_9_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID1KO - Day_12">GID1KO - Day_12</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_16" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_12 stub_1_16 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID1KO - Day_12 stub_1_16 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_17" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_12 stub_1_17 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID1KO - Day_12 stub_1_17 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_18" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_12 stub_1_18 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID1KO - Day_12 stub_1_18 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_19" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_12 stub_1_19 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID1KO - Day_12 stub_1_19 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_20" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO - Day_12 stub_1_20 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID1KO - Day_12 stub_1_20 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID1KO - Day_12_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID1KO - Day_12 summary_stub_GID1KO - Day_12_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID1KO - Day_12 summary_stub_GID1KO - Day_12_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID2KO - Day_3">GID2KO - Day_3</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_21" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_3 stub_1_21 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID2KO - Day_3 stub_1_21 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_22" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_3 stub_1_22 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID2KO - Day_3 stub_1_22 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_23" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_3 stub_1_23 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID2KO - Day_3 stub_1_23 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_24" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_3 stub_1_24 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID2KO - Day_3 stub_1_24 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_25" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_3 stub_1_25 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID2KO - Day_3 stub_1_25 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID2KO - Day_3_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID2KO - Day_3 summary_stub_GID2KO - Day_3_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID2KO - Day_3 summary_stub_GID2KO - Day_3_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID2KO - Day_6">GID2KO - Day_6</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_26" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_6 stub_1_26 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID2KO - Day_6 stub_1_26 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_27" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_6 stub_1_27 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID2KO - Day_6 stub_1_27 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_28" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_6 stub_1_28 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID2KO - Day_6 stub_1_28 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_29" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_6 stub_1_29 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID2KO - Day_6 stub_1_29 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_30" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_6 stub_1_30 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID2KO - Day_6 stub_1_30 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID2KO - Day_6_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID2KO - Day_6 summary_stub_GID2KO - Day_6_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID2KO - Day_6 summary_stub_GID2KO - Day_6_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID2KO - Day_9">GID2KO - Day_9</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_31" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_9 stub_1_31 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID2KO - Day_9 stub_1_31 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_32" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_9 stub_1_32 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID2KO - Day_9 stub_1_32 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_33" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_9 stub_1_33 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID2KO - Day_9 stub_1_33 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_34" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_9 stub_1_34 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID2KO - Day_9 stub_1_34 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_35" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_9 stub_1_35 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID2KO - Day_9 stub_1_35 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID2KO - Day_9_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID2KO - Day_9 summary_stub_GID2KO - Day_9_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID2KO - Day_9 summary_stub_GID2KO - Day_9_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID2KO - Day_12">GID2KO - Day_12</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_36" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_12 stub_1_36 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID2KO - Day_12 stub_1_36 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_37" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_12 stub_1_37 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID2KO - Day_12 stub_1_37 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_38" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_12 stub_1_38 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID2KO - Day_12 stub_1_38 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_39" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_12 stub_1_39 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID2KO - Day_12 stub_1_39 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_40" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO - Day_12 stub_1_40 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID2KO - Day_12 stub_1_40 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID2KO - Day_12_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID2KO - Day_12 summary_stub_GID2KO - Day_12_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID2KO - Day_12 summary_stub_GID2KO - Day_12_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID7KO - Day_3">GID7KO - Day_3</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_41" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_3 stub_1_41 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID7KO - Day_3 stub_1_41 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_42" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_3 stub_1_42 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID7KO - Day_3 stub_1_42 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_43" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_3 stub_1_43 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID7KO - Day_3 stub_1_43 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_44" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_3 stub_1_44 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID7KO - Day_3 stub_1_44 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_45" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_3 stub_1_45 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID7KO - Day_3 stub_1_45 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID7KO - Day_3_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID7KO - Day_3 summary_stub_GID7KO - Day_3_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID7KO - Day_3 summary_stub_GID7KO - Day_3_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID7KO - Day_6">GID7KO - Day_6</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_46" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_6 stub_1_46 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID7KO - Day_6 stub_1_46 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_47" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_6 stub_1_47 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID7KO - Day_6 stub_1_47 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_48" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_6 stub_1_48 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID7KO - Day_6 stub_1_48 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_49" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_6 stub_1_49 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID7KO - Day_6 stub_1_49 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_50" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_6 stub_1_50 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID7KO - Day_6 stub_1_50 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID7KO - Day_6_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID7KO - Day_6 summary_stub_GID7KO - Day_6_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID7KO - Day_6 summary_stub_GID7KO - Day_6_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID7KO - Day_9">GID7KO - Day_9</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_51" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_9 stub_1_51 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID7KO - Day_9 stub_1_51 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_52" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_9 stub_1_52 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID7KO - Day_9 stub_1_52 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_53" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_9 stub_1_53 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID7KO - Day_9 stub_1_53 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_54" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_9 stub_1_54 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID7KO - Day_9 stub_1_54 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_55" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_9 stub_1_55 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID7KO - Day_9 stub_1_55 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID7KO - Day_9_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID7KO - Day_9 summary_stub_GID7KO - Day_9_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID7KO - Day_9 summary_stub_GID7KO - Day_9_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID7KO - Day_12">GID7KO - Day_12</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_56" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_12 stub_1_56 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID7KO - Day_12 stub_1_56 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_57" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_12 stub_1_57 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID7KO - Day_12 stub_1_57 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_58" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_12 stub_1_58 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID7KO - Day_12 stub_1_58 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_59" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_12 stub_1_59 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID7KO - Day_12 stub_1_59 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_60" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO - Day_12 stub_1_60 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID7KO - Day_12 stub_1_60 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID7KO - Day_12_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID7KO - Day_12 summary_stub_GID7KO - Day_12_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID7KO - Day_12 summary_stub_GID7KO - Day_12_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID8KO - Day_3">GID8KO - Day_3</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_61" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_3 stub_1_61 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID8KO - Day_3 stub_1_61 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_62" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_3 stub_1_62 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID8KO - Day_3 stub_1_62 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_63" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_3 stub_1_63 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID8KO - Day_3 stub_1_63 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_64" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_3 stub_1_64 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID8KO - Day_3 stub_1_64 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_65" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_3 stub_1_65 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID8KO - Day_3 stub_1_65 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID8KO - Day_3_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID8KO - Day_3 summary_stub_GID8KO - Day_3_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID8KO - Day_3 summary_stub_GID8KO - Day_3_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID8KO - Day_6">GID8KO - Day_6</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_66" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_6 stub_1_66 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID8KO - Day_6 stub_1_66 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_67" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_6 stub_1_67 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID8KO - Day_6 stub_1_67 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_68" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_6 stub_1_68 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID8KO - Day_6 stub_1_68 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_69" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_6 stub_1_69 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID8KO - Day_6 stub_1_69 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_70" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_6 stub_1_70 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID8KO - Day_6 stub_1_70 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID8KO - Day_6_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID8KO - Day_6 summary_stub_GID8KO - Day_6_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID8KO - Day_6 summary_stub_GID8KO - Day_6_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID8KO - Day_9">GID8KO - Day_9</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_71" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_9 stub_1_71 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID8KO - Day_9 stub_1_71 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_72" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_9 stub_1_72 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID8KO - Day_9 stub_1_72 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_73" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_9 stub_1_73 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID8KO - Day_9 stub_1_73 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_74" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_9 stub_1_74 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID8KO - Day_9 stub_1_74 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_75" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_9 stub_1_75 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID8KO - Day_9 stub_1_75 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID8KO - Day_9_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID8KO - Day_9 summary_stub_GID8KO - Day_9_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID8KO - Day_9 summary_stub_GID8KO - Day_9_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID8KO - Day_12">GID8KO - Day_12</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_76" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_12 stub_1_76 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID8KO - Day_12 stub_1_76 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_77" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_12 stub_1_77 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID8KO - Day_12 stub_1_77 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_78" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_12 stub_1_78 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID8KO - Day_12 stub_1_78 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_79" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_12 stub_1_79 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID8KO - Day_12 stub_1_79 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_80" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO - Day_12 stub_1_80 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID8KO - Day_12 stub_1_80 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID8KO - Day_12_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID8KO - Day_12 summary_stub_GID8KO - Day_12_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID8KO - Day_12 summary_stub_GID8KO - Day_12_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID9KO - Day_3">GID9KO - Day_3</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_81" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_3 stub_1_81 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID9KO - Day_3 stub_1_81 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_82" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_3 stub_1_82 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID9KO - Day_3 stub_1_82 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_83" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_3 stub_1_83 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID9KO - Day_3 stub_1_83 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_84" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_3 stub_1_84 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID9KO - Day_3 stub_1_84 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_85" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_3 stub_1_85 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID9KO - Day_3 stub_1_85 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID9KO - Day_3_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID9KO - Day_3 summary_stub_GID9KO - Day_3_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID9KO - Day_3 summary_stub_GID9KO - Day_3_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID9KO - Day_6">GID9KO - Day_6</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_86" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_6 stub_1_86 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID9KO - Day_6 stub_1_86 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_87" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_6 stub_1_87 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID9KO - Day_6 stub_1_87 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_88" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_6 stub_1_88 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID9KO - Day_6 stub_1_88 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_89" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_6 stub_1_89 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID9KO - Day_6 stub_1_89 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_90" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_6 stub_1_90 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID9KO - Day_6 stub_1_90 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID9KO - Day_6_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID9KO - Day_6 summary_stub_GID9KO - Day_6_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID9KO - Day_6 summary_stub_GID9KO - Day_6_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID9KO - Day_9">GID9KO - Day_9</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_91" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_9 stub_1_91 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID9KO - Day_9 stub_1_91 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_92" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_9 stub_1_92 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID9KO - Day_9 stub_1_92 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_93" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_9 stub_1_93 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID9KO - Day_9 stub_1_93 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_94" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_9 stub_1_94 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID9KO - Day_9 stub_1_94 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_95" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_9 stub_1_95 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID9KO - Day_9 stub_1_95 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID9KO - Day_9_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID9KO - Day_9 summary_stub_GID9KO - Day_9_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID9KO - Day_9 summary_stub_GID9KO - Day_9_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID9KO - Day_12">GID9KO - Day_12</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_96" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_12 stub_1_96 biological_replicate" class="gt_row gt_center">1</td>
<td headers="GID9KO - Day_12 stub_1_96 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_97" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_12 stub_1_97 biological_replicate" class="gt_row gt_center">2</td>
<td headers="GID9KO - Day_12 stub_1_97 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_98" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_12 stub_1_98 biological_replicate" class="gt_row gt_center">3</td>
<td headers="GID9KO - Day_12 stub_1_98 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_99" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_12 stub_1_99 biological_replicate" class="gt_row gt_center">4</td>
<td headers="GID9KO - Day_12 stub_1_99 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_100" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO - Day_12 stub_1_100 biological_replicate" class="gt_row gt_center">5</td>
<td headers="GID9KO - Day_12 stub_1_100 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_GID9KO - Day_12_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID9KO - Day_12 summary_stub_GID9KO - Day_12_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID9KO - Day_12 summary_stub_GID9KO - Day_12_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="WT - Day_3">WT - Day_3</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_101" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_3 stub_1_101 biological_replicate" class="gt_row gt_center">1</td>
<td headers="WT - Day_3 stub_1_101 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_102" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_3 stub_1_102 biological_replicate" class="gt_row gt_center">2</td>
<td headers="WT - Day_3 stub_1_102 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_103" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_3 stub_1_103 biological_replicate" class="gt_row gt_center">3</td>
<td headers="WT - Day_3 stub_1_103 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_104" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_3 stub_1_104 biological_replicate" class="gt_row gt_center">4</td>
<td headers="WT - Day_3 stub_1_104 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_105" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_3 stub_1_105 biological_replicate" class="gt_row gt_center">5</td>
<td headers="WT - Day_3 stub_1_105 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_WT - Day_3_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="WT - Day_3 summary_stub_WT - Day_3_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="WT - Day_3 summary_stub_WT - Day_3_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="WT - Day_6">WT - Day_6</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_106" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_6 stub_1_106 biological_replicate" class="gt_row gt_center">1</td>
<td headers="WT - Day_6 stub_1_106 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_107" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_6 stub_1_107 biological_replicate" class="gt_row gt_center">2</td>
<td headers="WT - Day_6 stub_1_107 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_108" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_6 stub_1_108 biological_replicate" class="gt_row gt_center">3</td>
<td headers="WT - Day_6 stub_1_108 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_109" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_6 stub_1_109 biological_replicate" class="gt_row gt_center">4</td>
<td headers="WT - Day_6 stub_1_109 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_110" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_6 stub_1_110 biological_replicate" class="gt_row gt_center">5</td>
<td headers="WT - Day_6 stub_1_110 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_WT - Day_6_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="WT - Day_6 summary_stub_WT - Day_6_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="WT - Day_6 summary_stub_WT - Day_6_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="WT - Day_9">WT - Day_9</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_111" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_9 stub_1_111 biological_replicate" class="gt_row gt_center">1</td>
<td headers="WT - Day_9 stub_1_111 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_112" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_9 stub_1_112 biological_replicate" class="gt_row gt_center">2</td>
<td headers="WT - Day_9 stub_1_112 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_113" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_9 stub_1_113 biological_replicate" class="gt_row gt_center">3</td>
<td headers="WT - Day_9 stub_1_113 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_114" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_9 stub_1_114 biological_replicate" class="gt_row gt_center">4</td>
<td headers="WT - Day_9 stub_1_114 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_115" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_9 stub_1_115 biological_replicate" class="gt_row gt_center">5</td>
<td headers="WT - Day_9 stub_1_115 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_WT - Day_9_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="WT - Day_9 summary_stub_WT - Day_9_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="WT - Day_9 summary_stub_WT - Day_9_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="WT - Day_12">WT - Day_12</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_116" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_12 stub_1_116 biological_replicate" class="gt_row gt_center">1</td>
<td headers="WT - Day_12 stub_1_116 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_117" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_12 stub_1_117 biological_replicate" class="gt_row gt_center">2</td>
<td headers="WT - Day_12 stub_1_117 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_118" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_12 stub_1_118 biological_replicate" class="gt_row gt_center">3</td>
<td headers="WT - Day_12 stub_1_118 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_119" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_12 stub_1_119 biological_replicate" class="gt_row gt_center">4</td>
<td headers="WT - Day_12 stub_1_119 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="stub_1_120" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT - Day_12 stub_1_120 biological_replicate" class="gt_row gt_center">5</td>
<td headers="WT - Day_12 stub_1_120 n" class="gt_row gt_right">2</td></tr>
    <tr><th id="summary_stub_WT - Day_12_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="WT - Day_12 summary_stub_WT - Day_12_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="WT - Day_12 summary_stub_WT - Day_12_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">10</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="N/A - N/A">N/A - N/A</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_121" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="N/A - N/A stub_1_121 biological_replicate" class="gt_row gt_center">N/A</td>
<td headers="N/A - N/A stub_1_121 n" class="gt_row gt_right">136</td></tr>
    <tr><th id="summary_stub_N/A - N/A_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="N/A - N/A summary_stub_N/A - N/A_1 biological_replicate" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="N/A - N/A summary_stub_N/A - N/A_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">136</td></tr>
    <tr><th id="grand_summary_stub_1" scope="row" class="gt_row gt_left gt_stub gt_grand_summary_row gt_first_grand_summary_row gt_last_summary_row">total</th>
<td headers="grand_summary_stub_1 biological_replicate" class="gt_row gt_center gt_grand_summary_row gt_first_grand_summary_row gt_last_summary_row"></td>
<td headers="grand_summary_stub_1 n" class="gt_row gt_right gt_grand_summary_row gt_first_grand_summary_row gt_last_summary_row">376</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</section>
<section id="biological-replicates-per-cell-line" class="panel">
<h4>Biological replicates (per cell line)</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>bio_reps_tbl</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span><span class='va'>cell_line</span>, <span class='va'>timepoint</span>, <span class='va'>biological_replicate</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/group_by.html'>group_by</a></span><span class='op'>(</span><span class='va'>cell_line</span>, <span class='va'>timepoint</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/distinct.html'>distinct</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/count.html'>count</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/group_by.html'>group_by</a></span><span class='op'>(</span><span class='va'>cell_line</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://gt.rstudio.com/reference/gt.html'>gt</a></span><span class='op'>(</span></span>
<span>  <span class='va'>bio_reps_tbl</span>,</span>
<span>  caption <span class='op'>=</span> <span class='fu'><a href='https://gt.rstudio.com/reference/md.html'>md</a></span><span class='op'>(</span></span>
<span>    <span class='st'>"Number of biological replicates per `cell_line`-`timepoint` combination."</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://gt.rstudio.com/reference/fmt_number.html'>fmt_number</a></span><span class='op'>(</span>columns <span class='op'>=</span> <span class='va'>n</span>, decimals <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://gt.rstudio.com/reference/summary_rows.html'>summary_rows</a></span><span class='op'>(</span></span>
<span>    columns <span class='op'>=</span> <span class='va'>n</span>, </span>
<span>    fns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>subtotal</span> <span class='op'>~</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>.</span><span class='op'>)</span><span class='op'>)</span>, </span>
<span>    fmt <span class='op'>=</span> <span class='op'>~</span> <span class='fu'><a href='https://gt.rstudio.com/reference/fmt_number.html'>fmt_number</a></span><span class='op'>(</span><span class='va'>.</span>, decimals <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://gt.rstudio.com/reference/grand_summary_rows.html'>grand_summary_rows</a></span><span class='op'>(</span></span>
<span>    columns <span class='op'>=</span> <span class='va'>n</span>, </span>
<span>    fns <span class='op'>=</span> <span class='va'>total</span> <span class='op'>~</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>.</span><span class='op'>)</span>, </span>
<span>    fmt <span class='op'>=</span> <span class='op'>~</span> <span class='fu'><a href='https://gt.rstudio.com/reference/fmt_number.html'>fmt_number</a></span><span class='op'>(</span><span class='va'>.</span>, decimals <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://gt.rstudio.com/reference/tab_options.html'>tab_options</a></span><span class='op'>(</span></span>
<span>    summary_row.background.color <span class='op'>=</span> <span class='st'>"lightgrey"</span>,</span>
<span>    grand_summary_row.background.color <span class='op'>=</span> <span class='st'>"darkgrey"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div id="edlocqqwbj" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#edlocqqwbj table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#edlocqqwbj thead, #edlocqqwbj tbody, #edlocqqwbj tfoot, #edlocqqwbj tr, #edlocqqwbj td, #edlocqqwbj th {
  border-style: none;
}

#edlocqqwbj p {
  margin: 0;
  padding: 0;
}

#edlocqqwbj .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#edlocqqwbj .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#edlocqqwbj .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#edlocqqwbj .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#edlocqqwbj .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#edlocqqwbj .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#edlocqqwbj .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#edlocqqwbj .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#edlocqqwbj .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#edlocqqwbj .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#edlocqqwbj .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#edlocqqwbj .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#edlocqqwbj .gt_spanner_row {
  border-bottom-style: hidden;
}

#edlocqqwbj .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#edlocqqwbj .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#edlocqqwbj .gt_from_md > :first-child {
  margin-top: 0;
}

#edlocqqwbj .gt_from_md > :last-child {
  margin-bottom: 0;
}

#edlocqqwbj .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#edlocqqwbj .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#edlocqqwbj .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#edlocqqwbj .gt_row_group_first td {
  border-top-width: 2px;
}

#edlocqqwbj .gt_row_group_first th {
  border-top-width: 2px;
}

#edlocqqwbj .gt_summary_row {
  color: #333333;
  background-color: #D3D3D3;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#edlocqqwbj .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#edlocqqwbj .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#edlocqqwbj .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#edlocqqwbj .gt_grand_summary_row {
  color: #FFFFFF;
  background-color: #A9A9A9;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#edlocqqwbj .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#edlocqqwbj .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#edlocqqwbj .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#edlocqqwbj .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#edlocqqwbj .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#edlocqqwbj .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#edlocqqwbj .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#edlocqqwbj .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#edlocqqwbj .gt_left {
  text-align: left;
}

#edlocqqwbj .gt_center {
  text-align: center;
}

#edlocqqwbj .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#edlocqqwbj .gt_font_normal {
  font-weight: normal;
}

#edlocqqwbj .gt_font_bold {
  font-weight: bold;
}

#edlocqqwbj .gt_font_italic {
  font-style: italic;
}

#edlocqqwbj .gt_super {
  font-size: 65%;
}

#edlocqqwbj .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#edlocqqwbj .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#edlocqqwbj .gt_indent_1 {
  text-indent: 5px;
}

#edlocqqwbj .gt_indent_2 {
  text-indent: 10px;
}

#edlocqqwbj .gt_indent_3 {
  text-indent: 15px;
}

#edlocqqwbj .gt_indent_4 {
  text-indent: 20px;
}

#edlocqqwbj .gt_indent_5 {
  text-indent: 25px;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <caption><span id="tab:unnamed-chunk-4">Table 2: </span>Number of biological replicates per <code>cell_line</code>-<code>timepoint</code> combination.</caption>
  <thead>
    
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id=""></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="timepoint">timepoint</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID1KO">GID1KO</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_1" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO stub_1_1 timepoint" class="gt_row gt_center">Day_3</td>
<td headers="GID1KO stub_1_1 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_2" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO stub_1_2 timepoint" class="gt_row gt_center">Day_6</td>
<td headers="GID1KO stub_1_2 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_3" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO stub_1_3 timepoint" class="gt_row gt_center">Day_9</td>
<td headers="GID1KO stub_1_3 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_4" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID1KO stub_1_4 timepoint" class="gt_row gt_center">Day_12</td>
<td headers="GID1KO stub_1_4 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="summary_stub_GID1KO_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID1KO summary_stub_GID1KO_1 timepoint" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID1KO summary_stub_GID1KO_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">20</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID2KO">GID2KO</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_5" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO stub_1_5 timepoint" class="gt_row gt_center">Day_3</td>
<td headers="GID2KO stub_1_5 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_6" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO stub_1_6 timepoint" class="gt_row gt_center">Day_6</td>
<td headers="GID2KO stub_1_6 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_7" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO stub_1_7 timepoint" class="gt_row gt_center">Day_9</td>
<td headers="GID2KO stub_1_7 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_8" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID2KO stub_1_8 timepoint" class="gt_row gt_center">Day_12</td>
<td headers="GID2KO stub_1_8 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="summary_stub_GID2KO_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID2KO summary_stub_GID2KO_1 timepoint" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID2KO summary_stub_GID2KO_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">20</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID7KO">GID7KO</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_9" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO stub_1_9 timepoint" class="gt_row gt_center">Day_3</td>
<td headers="GID7KO stub_1_9 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_10" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO stub_1_10 timepoint" class="gt_row gt_center">Day_6</td>
<td headers="GID7KO stub_1_10 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_11" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO stub_1_11 timepoint" class="gt_row gt_center">Day_9</td>
<td headers="GID7KO stub_1_11 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_12" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID7KO stub_1_12 timepoint" class="gt_row gt_center">Day_12</td>
<td headers="GID7KO stub_1_12 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="summary_stub_GID7KO_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID7KO summary_stub_GID7KO_1 timepoint" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID7KO summary_stub_GID7KO_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">20</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID8KO">GID8KO</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_13" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO stub_1_13 timepoint" class="gt_row gt_center">Day_3</td>
<td headers="GID8KO stub_1_13 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_14" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO stub_1_14 timepoint" class="gt_row gt_center">Day_6</td>
<td headers="GID8KO stub_1_14 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_15" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO stub_1_15 timepoint" class="gt_row gt_center">Day_9</td>
<td headers="GID8KO stub_1_15 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_16" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID8KO stub_1_16 timepoint" class="gt_row gt_center">Day_12</td>
<td headers="GID8KO stub_1_16 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="summary_stub_GID8KO_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID8KO summary_stub_GID8KO_1 timepoint" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID8KO summary_stub_GID8KO_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">20</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="GID9KO">GID9KO</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_17" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO stub_1_17 timepoint" class="gt_row gt_center">Day_3</td>
<td headers="GID9KO stub_1_17 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_18" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO stub_1_18 timepoint" class="gt_row gt_center">Day_6</td>
<td headers="GID9KO stub_1_18 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_19" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO stub_1_19 timepoint" class="gt_row gt_center">Day_9</td>
<td headers="GID9KO stub_1_19 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_20" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="GID9KO stub_1_20 timepoint" class="gt_row gt_center">Day_12</td>
<td headers="GID9KO stub_1_20 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="summary_stub_GID9KO_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="GID9KO summary_stub_GID9KO_1 timepoint" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="GID9KO summary_stub_GID9KO_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">20</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="WT">WT</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_21" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT stub_1_21 timepoint" class="gt_row gt_center">Day_3</td>
<td headers="WT stub_1_21 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_22" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT stub_1_22 timepoint" class="gt_row gt_center">Day_6</td>
<td headers="WT stub_1_22 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_23" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT stub_1_23 timepoint" class="gt_row gt_center">Day_9</td>
<td headers="WT stub_1_23 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_24" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="WT stub_1_24 timepoint" class="gt_row gt_center">Day_12</td>
<td headers="WT stub_1_24 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="summary_stub_WT_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="WT summary_stub_WT_1 timepoint" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="WT summary_stub_WT_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">20</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="N/A">N/A</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_25" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="N/A stub_1_25 timepoint" class="gt_row gt_center">N/A</td>
<td headers="N/A stub_1_25 n" class="gt_row gt_right">1</td></tr>
    <tr><th id="summary_stub_N/A_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="N/A summary_stub_N/A_1 timepoint" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="N/A summary_stub_N/A_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">1</td></tr>
    <tr><th id="grand_summary_stub_1" scope="row" class="gt_row gt_left gt_stub gt_grand_summary_row gt_first_grand_summary_row gt_last_summary_row">total</th>
<td headers="grand_summary_stub_1 timepoint" class="gt_row gt_center gt_grand_summary_row gt_first_grand_summary_row gt_last_summary_row"></td>
<td headers="grand_summary_stub_1 n" class="gt_row gt_right gt_grand_summary_row gt_first_grand_summary_row gt_last_summary_row">121</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</section>
<section id="biological-replicates-per-timepoint" class="panel">
<h4>Biological replicates (per timepoint)</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>bio_reps_tbl</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span><span class='va'>timepoint</span>, <span class='va'>cell_line</span>, <span class='va'>biological_replicate</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/group_by.html'>group_by</a></span><span class='op'>(</span><span class='va'>timepoint</span>, <span class='va'>cell_line</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/distinct.html'>distinct</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/count.html'>count</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/group_by.html'>group_by</a></span><span class='op'>(</span><span class='va'>timepoint</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://gt.rstudio.com/reference/gt.html'>gt</a></span><span class='op'>(</span></span>
<span>  <span class='va'>bio_reps_tbl</span>,</span>
<span>  caption <span class='op'>=</span> <span class='fu'><a href='https://gt.rstudio.com/reference/md.html'>md</a></span><span class='op'>(</span></span>
<span>    <span class='st'>"Number of biological replicates per `timepoint`-`cell_line` combination."</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://gt.rstudio.com/reference/fmt_number.html'>fmt_number</a></span><span class='op'>(</span>columns <span class='op'>=</span> <span class='va'>n</span>, decimals <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://gt.rstudio.com/reference/summary_rows.html'>summary_rows</a></span><span class='op'>(</span></span>
<span>    columns <span class='op'>=</span> <span class='va'>n</span>, </span>
<span>    fns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>subtotal</span> <span class='op'>~</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>.</span><span class='op'>)</span><span class='op'>)</span>, </span>
<span>    fmt <span class='op'>=</span> <span class='op'>~</span> <span class='fu'><a href='https://gt.rstudio.com/reference/fmt_number.html'>fmt_number</a></span><span class='op'>(</span><span class='va'>.</span>, decimals <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://gt.rstudio.com/reference/grand_summary_rows.html'>grand_summary_rows</a></span><span class='op'>(</span></span>
<span>    columns <span class='op'>=</span> <span class='va'>n</span>, </span>
<span>    fns <span class='op'>=</span> <span class='va'>total</span> <span class='op'>~</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>.</span><span class='op'>)</span>, </span>
<span>    fmt <span class='op'>=</span> <span class='op'>~</span> <span class='fu'><a href='https://gt.rstudio.com/reference/fmt_number.html'>fmt_number</a></span><span class='op'>(</span><span class='va'>.</span>, decimals <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://gt.rstudio.com/reference/tab_options.html'>tab_options</a></span><span class='op'>(</span></span>
<span>    summary_row.background.color <span class='op'>=</span> <span class='st'>"lightgrey"</span>,</span>
<span>    grand_summary_row.background.color <span class='op'>=</span> <span class='st'>"darkgrey"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div id="ezlvpfywkf" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#ezlvpfywkf table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#ezlvpfywkf thead, #ezlvpfywkf tbody, #ezlvpfywkf tfoot, #ezlvpfywkf tr, #ezlvpfywkf td, #ezlvpfywkf th {
  border-style: none;
}

#ezlvpfywkf p {
  margin: 0;
  padding: 0;
}

#ezlvpfywkf .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#ezlvpfywkf .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#ezlvpfywkf .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#ezlvpfywkf .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#ezlvpfywkf .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ezlvpfywkf .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ezlvpfywkf .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ezlvpfywkf .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#ezlvpfywkf .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#ezlvpfywkf .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#ezlvpfywkf .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#ezlvpfywkf .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#ezlvpfywkf .gt_spanner_row {
  border-bottom-style: hidden;
}

#ezlvpfywkf .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#ezlvpfywkf .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#ezlvpfywkf .gt_from_md > :first-child {
  margin-top: 0;
}

#ezlvpfywkf .gt_from_md > :last-child {
  margin-bottom: 0;
}

#ezlvpfywkf .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#ezlvpfywkf .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#ezlvpfywkf .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#ezlvpfywkf .gt_row_group_first td {
  border-top-width: 2px;
}

#ezlvpfywkf .gt_row_group_first th {
  border-top-width: 2px;
}

#ezlvpfywkf .gt_summary_row {
  color: #333333;
  background-color: #D3D3D3;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ezlvpfywkf .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#ezlvpfywkf .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#ezlvpfywkf .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ezlvpfywkf .gt_grand_summary_row {
  color: #FFFFFF;
  background-color: #A9A9A9;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ezlvpfywkf .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#ezlvpfywkf .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#ezlvpfywkf .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#ezlvpfywkf .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ezlvpfywkf .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ezlvpfywkf .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ezlvpfywkf .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ezlvpfywkf .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ezlvpfywkf .gt_left {
  text-align: left;
}

#ezlvpfywkf .gt_center {
  text-align: center;
}

#ezlvpfywkf .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#ezlvpfywkf .gt_font_normal {
  font-weight: normal;
}

#ezlvpfywkf .gt_font_bold {
  font-weight: bold;
}

#ezlvpfywkf .gt_font_italic {
  font-style: italic;
}

#ezlvpfywkf .gt_super {
  font-size: 65%;
}

#ezlvpfywkf .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#ezlvpfywkf .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#ezlvpfywkf .gt_indent_1 {
  text-indent: 5px;
}

#ezlvpfywkf .gt_indent_2 {
  text-indent: 10px;
}

#ezlvpfywkf .gt_indent_3 {
  text-indent: 15px;
}

#ezlvpfywkf .gt_indent_4 {
  text-indent: 20px;
}

#ezlvpfywkf .gt_indent_5 {
  text-indent: 25px;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <caption><span id="tab:unnamed-chunk-5">Table 3: </span>Number of biological replicates per <code>timepoint</code>-<code>cell_line</code> combination.</caption>
  <thead>
    
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id=""></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="cell_line">cell_line</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="n">n</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="Day_3">Day_3</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_1" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_3 stub_1_1 cell_line" class="gt_row gt_center">GID1KO</td>
<td headers="Day_3 stub_1_1 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_2" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_3 stub_1_2 cell_line" class="gt_row gt_center">GID2KO</td>
<td headers="Day_3 stub_1_2 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_3" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_3 stub_1_3 cell_line" class="gt_row gt_center">GID7KO</td>
<td headers="Day_3 stub_1_3 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_4" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_3 stub_1_4 cell_line" class="gt_row gt_center">GID8KO</td>
<td headers="Day_3 stub_1_4 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_5" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_3 stub_1_5 cell_line" class="gt_row gt_center">GID9KO</td>
<td headers="Day_3 stub_1_5 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_6" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_3 stub_1_6 cell_line" class="gt_row gt_center">WT</td>
<td headers="Day_3 stub_1_6 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="summary_stub_Day_3_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="Day_3 summary_stub_Day_3_1 cell_line" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="Day_3 summary_stub_Day_3_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">30</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="Day_6">Day_6</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_7" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_6 stub_1_7 cell_line" class="gt_row gt_center">GID1KO</td>
<td headers="Day_6 stub_1_7 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_8" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_6 stub_1_8 cell_line" class="gt_row gt_center">GID2KO</td>
<td headers="Day_6 stub_1_8 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_9" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_6 stub_1_9 cell_line" class="gt_row gt_center">GID7KO</td>
<td headers="Day_6 stub_1_9 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_10" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_6 stub_1_10 cell_line" class="gt_row gt_center">GID8KO</td>
<td headers="Day_6 stub_1_10 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_11" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_6 stub_1_11 cell_line" class="gt_row gt_center">GID9KO</td>
<td headers="Day_6 stub_1_11 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_12" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_6 stub_1_12 cell_line" class="gt_row gt_center">WT</td>
<td headers="Day_6 stub_1_12 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="summary_stub_Day_6_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="Day_6 summary_stub_Day_6_1 cell_line" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="Day_6 summary_stub_Day_6_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">30</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="Day_9">Day_9</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_13" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_9 stub_1_13 cell_line" class="gt_row gt_center">GID1KO</td>
<td headers="Day_9 stub_1_13 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_14" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_9 stub_1_14 cell_line" class="gt_row gt_center">GID2KO</td>
<td headers="Day_9 stub_1_14 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_15" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_9 stub_1_15 cell_line" class="gt_row gt_center">GID7KO</td>
<td headers="Day_9 stub_1_15 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_16" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_9 stub_1_16 cell_line" class="gt_row gt_center">GID8KO</td>
<td headers="Day_9 stub_1_16 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_17" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_9 stub_1_17 cell_line" class="gt_row gt_center">GID9KO</td>
<td headers="Day_9 stub_1_17 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_18" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_9 stub_1_18 cell_line" class="gt_row gt_center">WT</td>
<td headers="Day_9 stub_1_18 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="summary_stub_Day_9_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="Day_9 summary_stub_Day_9_1 cell_line" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="Day_9 summary_stub_Day_9_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">30</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="Day_12">Day_12</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_19" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_12 stub_1_19 cell_line" class="gt_row gt_center">GID1KO</td>
<td headers="Day_12 stub_1_19 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_20" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_12 stub_1_20 cell_line" class="gt_row gt_center">GID2KO</td>
<td headers="Day_12 stub_1_20 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_21" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_12 stub_1_21 cell_line" class="gt_row gt_center">GID7KO</td>
<td headers="Day_12 stub_1_21 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_22" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_12 stub_1_22 cell_line" class="gt_row gt_center">GID8KO</td>
<td headers="Day_12 stub_1_22 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_23" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_12 stub_1_23 cell_line" class="gt_row gt_center">GID9KO</td>
<td headers="Day_12 stub_1_23 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="stub_1_24" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="Day_12 stub_1_24 cell_line" class="gt_row gt_center">WT</td>
<td headers="Day_12 stub_1_24 n" class="gt_row gt_right">5</td></tr>
    <tr><th id="summary_stub_Day_12_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="Day_12 summary_stub_Day_12_1 cell_line" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="Day_12 summary_stub_Day_12_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">30</td></tr>
    <tr class="gt_group_heading_row">
      <th colspan="3" class="gt_group_heading" scope="colgroup" id="N/A">N/A</th>
    </tr>
    <tr class="gt_row_group_first"><th id="stub_1_25" scope="row" class="gt_row gt_left gt_stub"></th>
<td headers="N/A stub_1_25 cell_line" class="gt_row gt_center">N/A</td>
<td headers="N/A stub_1_25 n" class="gt_row gt_right">1</td></tr>
    <tr><th id="summary_stub_N/A_1" scope="row" class="gt_row gt_left gt_stub gt_summary_row gt_first_summary_row thick gt_last_summary_row">subtotal</th>
<td headers="N/A summary_stub_N/A_1 cell_line" class="gt_row gt_center gt_summary_row gt_first_summary_row thick gt_last_summary_row"></td>
<td headers="N/A summary_stub_N/A_1 n" class="gt_row gt_right gt_summary_row gt_first_summary_row thick gt_last_summary_row">1</td></tr>
    <tr><th id="grand_summary_stub_1" scope="row" class="gt_row gt_left gt_stub gt_grand_summary_row gt_first_grand_summary_row gt_last_summary_row">total</th>
<td headers="grand_summary_stub_1 cell_line" class="gt_row gt_center gt_grand_summary_row gt_first_grand_summary_row gt_last_summary_row"></td>
<td headers="grand_summary_stub_1 n" class="gt_row gt_right gt_grand_summary_row gt_first_grand_summary_row gt_last_summary_row">121</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</section>
</div>
<h2 id="incorporating-gene-based-annotation">Incorporating gene-based annotation</h2>
<p>We obtain gene-based annotations, such as the chromosome, using the information in the GFF file, <code>PlasmoDB-51_Pfalciparum3D7.gff</code><a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>ids</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span></span>
<span></span>
<span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://bioconductor.org/packages/GenomicFeatures'>GenomicFeatures</a></span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>txdb</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbFromGFF.html'>makeTxDbFromGFF</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span></span>
<span>    <span class='st'>"extdata"</span>, </span>
<span>    <span class='st'>"PlasmoDB-51_Pfalciparum3D7"</span>,</span>
<span>    <span class='st'>"PlasmoDB-51_Pfalciparum3D7.gff"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>txdb_columns</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html'>columns</a></span><span class='op'>(</span><span class='va'>txdb</span><span class='op'>)</span></span>
<span><span class='va'>txdb_columns</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/setNames.html'>setNames</a></span><span class='op'>(</span><span class='va'>txdb_columns</span>, <span class='va'>txdb_columns</span><span class='op'>)</span></span>
<span><span class='va'>txdb_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html'>DataFrame</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>txdb_columns</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>column</span><span class='op'>)</span> <span class='op'>{</span></span>
<span>    <span class='fu'><a href='https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html'>mapIds</a></span><span class='op'>(</span></span>
<span>      x <span class='op'>=</span> <span class='va'>txdb</span>, </span>
<span>      <span class='co'># NOTE: Need to remove gene version number prior to lookup.</span></span>
<span>      keys <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/grep.html'>gsub</a></span><span class='op'>(</span><span class='st'>"\\.[0-9]+$"</span>, <span class='st'>""</span>, <span class='va'>ids</span><span class='op'>)</span>,</span>
<span>      keytype <span class='op'>=</span> <span class='st'>"GENEID"</span>,</span>
<span>      column <span class='op'>=</span> <span class='va'>column</span>,</span>
<span>      multiVals <span class='op'>=</span> <span class='st'>"CharacterList"</span><span class='op'>)</span></span>
<span>  <span class='op'>}</span><span class='op'>)</span>,</span>
<span>  row.names <span class='op'>=</span> <span class='va'>ids</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>rowData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='va'>txdb_df</span></span></code></pre>
</div>
</details>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Some useful gene sets</span></span>
<span><span class='va'>mito_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/any.html'>any</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>rowData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>$</span><span class='va'>TXCHROM</span> <span class='op'>==</span> <span class='st'>"Pf3D7_MIT_v3"</span><span class='op'>)</span><span class='op'>]</span></span>
<span><span class='va'>pseudogene_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/base/any.html'>any</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>rowData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>$</span><span class='va'>TXTYPE</span> <span class='op'>==</span> <span class='st'>"pseudogenic_transcript"</span><span class='op'>)</span><span class='op'>]</span></span>
<span><span class='va'>ribo_set</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/sets.html'>union</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/any.html'>any</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>rowData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>$</span><span class='va'>TXTYPE</span> <span class='op'>==</span> <span class='st'>"rRNA"</span><span class='op'>)</span><span class='op'>]</span>,</span>
<span>  <span class='co'># NOTE: Include mRNA that are ribosomal protein genes.</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span></span>
<span>    <span class='fu'><a href='https://rdrr.io/r/base/any.html'>any</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/grep.html'>grepl</a></span><span class='op'>(</span><span class='st'>"ribosomal"</span>, <span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>rowData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>$</span><span class='va'>GENENAME</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>&amp;</span></span>
<span>    <span class='fu'><a href='https://rdrr.io/r/base/any.html'>any</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/grep.html'>grepl</a></span><span class='op'>(</span><span class='st'>"protein"</span>, <span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>rowData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>$</span><span class='va'>GENENAME</span><span class='op'>)</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<h1 id="read-mapping-metrics">Read mapping metrics</h1>
<h2 id="motivation">Motivation</h2>
<p>We aimed to sequence approximately 5 million reads per technical replicate by sequencing these 240 mini-bulk libraries with 1.2 billion reads (<span class="math inline">\(\frac{1.2 \times 10^9}{240} = 5 \times 10^6\)</span>).
We further hoped that the majority of these reads would be exonic.</p>
<p>In fact, we obtained <span class="math inline">\(1.362 \times 10^9\)</span> reads, with approximately <span class="math inline">\(77.29\%\)</span> having a match<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a> to the 376 CEL-Seq2 well barcodes for a total of <span class="math inline">\(1.053 \times 10^9\)</span> usable reads.
A further <span class="math inline">\(21.03\%\)</span> of reads had no barcode match nor aligned to the reference genome<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a>.</p>
<p>Further QC metrics are available in <a href="../output/scPipe/RPI-43.scPipe_QC_report.nb.html"><code>output/scPipe/RPI-43.scPipe_QC_report.nb.html</code></a></p>
<h2 id="analysis">Analysis</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/LuyiTian/scPipe'>scPipe</a></span><span class='op'>)</span></span>
<span><span class='va'>sce</span><span class='op'>$</span><span class='va'>nreads</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/apply.html'>apply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/scPipe/man/QC_metrics.html'>QC_metrics</a></span><span class='op'>(</span><span class='va'>sce_deduped</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>, <span class='op'>]</span>, <span class='fl'>1</span>, <span class='va'>sum</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:nreads">1</a> - <a href="#fig:intronic">3</a> shows that we generated a median of 3.1 million reads for each technical replicate with a median 75% of these exonic and a median 9% of these intronic.</p>
<div class="panelset">
<section id="number-of-reads" class="panel">
<h3>Number of reads</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>, </span>
<span>  y <span class='op'>=</span> <span class='st'>"nreads"</span>,</span>
<span>  x <span class='op'>=</span> <span class='st'>"biological_replicate"</span>,</span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  other_fields <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cell_line"</span>, <span class='st'>"timepoint"</span><span class='op'>)</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  theme_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='cn'>NA</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span> <span class='op'>~</span> <span class='va'>cell_line</span>, scales <span class='op'>=</span> <span class='st'>"free_x"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span></span>
<span>    yintercept <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>nreads</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_type</span> <span class='op'>!=</span> <span class='st'>"empty"</span><span class='op'>]</span><span class='op'>)</span>, </span>
<span>    lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span></span>
<span>    yintercept <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>nreads</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_type</span> <span class='op'>==</span> <span class='st'>"empty"</span><span class='op'>]</span><span class='op'>)</span>, </span>
<span>    lty <span class='op'>=</span> <span class='fl'>2</span>,</span>
<span>    col <span class='op'>=</span> <span class='st'>"red"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/annotation_logticks.html'>annotation_logticks</a></span><span class='op'>(</span></span>
<span>    sides <span class='op'>=</span> <span class='st'>"l"</span>,</span>
<span>    short <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.03</span>, <span class='st'>"cm"</span><span class='op'>)</span>,</span>
<span>    mid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.06</span>, <span class='st'>"cm"</span><span class='op'>)</span>,</span>
<span>    long <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.09</span>, <span class='st'>"cm"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:nreads"></span>
<img src="G000396_Danu.preprocess_files/nreads-1.png" alt="Number of reads generated per technical replicate. The black   dashed horizontal line denotes the median across all samples and the red   dashed horizontal lines denotes the maximum across all the 'empty' wells." width="624" />
<p class="caption">
Figure 1: Number of reads generated per technical replicate. The black dashed horizontal line denotes the median across all samples and the red dashed horizontal lines denotes the maximum across all the empty wells.
</p>
</div>
</div>
</section>
<section id="exonic" class="panel">
<h3>Exonic %</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>, </span>
<span>  y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/AsIs.html'>I</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>mapped_to_exon</span> <span class='op'>/</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>nreads</span><span class='op'>)</span>, </span>
<span>  x <span class='op'>=</span> <span class='st'>"biological_replicate"</span>,</span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  other_fields <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cell_line"</span>, <span class='st'>"timepoint"</span><span class='op'>)</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  theme_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Percentage exonic"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span></span>
<span>    yintercept <span class='op'>=</span> </span>
<span>      <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>mapped_to_exon</span> <span class='op'>/</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>nreads</span><span class='op'>)</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_type</span> <span class='op'>!=</span> <span class='st'>"empty"</span><span class='op'>]</span><span class='op'>)</span>,</span>
<span>    lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span> <span class='op'>~</span> <span class='va'>cell_line</span>, scales <span class='op'>=</span> <span class='st'>"free_x"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_continuous</a></span><span class='op'>(</span>labels <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='va'><a href='https://scales.r-lib.org/reference/percent_format.html'>percent</a></span>, limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:exonic"></span>
<img src="G000396_Danu.preprocess_files/exonic-1.png" alt="Percentage of reads that are exonic per sample. The dashed   horizontal line denotes the median across all samples." width="624" />
<p class="caption">
Figure 2: Percentage of reads that are exonic per sample. The dashed horizontal line denotes the median across all samples.
</p>
</div>
</div>
</section>
<section id="intronic" class="panel">
<h3>Intronic %</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>, </span>
<span>  y <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/AsIs.html'>I</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>mapped_to_intron</span> <span class='op'>/</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>nreads</span><span class='op'>)</span>, </span>
<span>  x <span class='op'>=</span> <span class='st'>"biological_replicate"</span>,</span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  other_fields <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cell_line"</span>, <span class='st'>"timepoint"</span><span class='op'>)</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  theme_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Percentage intronic"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span></span>
<span>    yintercept <span class='op'>=</span> </span>
<span>      <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>mapped_to_intron</span> <span class='op'>/</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>nreads</span><span class='op'>)</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_type</span> <span class='op'>!=</span> <span class='st'>"empty"</span><span class='op'>]</span><span class='op'>)</span>,</span>
<span>    lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span> <span class='op'>~</span> <span class='va'>cell_line</span>, scales <span class='op'>=</span> <span class='st'>"free_x"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_continuous</a></span><span class='op'>(</span>labels <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='va'><a href='https://scales.r-lib.org/reference/percent_format.html'>percent</a></span>, limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:intronic"></span>
<img src="G000396_Danu.preprocess_files/intronic-1.png" alt="Percentage of reads that are intronic per sample. The dashed   horizontal line denotes the median across all samples." width="624" />
<p class="caption">
Figure 3: Percentage of reads that are intronic per sample. The dashed horizontal line denotes the median across all samples.
</p>
</div>
</div>
</section>
</div>
<p>We observe that there are 16 replicates, listed below, with fewer reads than the maximum number of reads from an empty well (nreads = 312,601).
The majority of these are from <code>GID2KO</code>, particularly biological replicates <code>1</code> and <code>2</code>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/filter.html'>filter</a></span><span class='op'>(</span></span>
<span>    <span class='va'>nreads</span> <span class='op'>&lt;</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>nreads</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_type</span> <span class='op'>==</span> <span class='st'>"empty"</span><span class='op'>]</span><span class='op'>)</span>,</span>
<span>    <span class='va'>sample_type</span> <span class='op'>!=</span> <span class='st'>"empty"</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span></span>
<span>    <span class='va'>sample</span>, <span class='va'>technical_replicate</span>, <span class='va'>well_position</span>, <span class='va'>nreads</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/arrange.html'>arrange</a></span><span class='op'>(</span><span class='va'>sample</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span></span>
<span>    caption <span class='op'>=</span> <span class='st'>"Metadata of samples with `nreads` less than the maximim number of of reads from an 'empty' well."</span>,</span>
<span>    format.args <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>big.mark <span class='op'>=</span> <span class='st'>","</span><span class='op'>)</span>,</span>
<span>    row.names <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-12">Table 4: </span>Metadata of samples with <code>nreads</code> less than the maximim number of of reads from an empty well.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">sample</th>
<th style="text-align: left;">technical_replicate</th>
<th style="text-align: left;">well_position</th>
<th style="text-align: right;">nreads</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">GID2KO.Day_3.1</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">D1</td>
<td style="text-align: right;">183,053</td>
</tr>
<tr class="even">
<td style="text-align: left;">GID2KO.Day_3.1</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">E1</td>
<td style="text-align: right;">48,030</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GID2KO.Day_3.2</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">E3</td>
<td style="text-align: right;">84,179</td>
</tr>
<tr class="even">
<td style="text-align: left;">GID2KO.Day_6.1</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">M1</td>
<td style="text-align: right;">111,283</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GID2KO.Day_6.2</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">L3</td>
<td style="text-align: right;">256,956</td>
</tr>
<tr class="even">
<td style="text-align: left;">GID2KO.Day_9.1</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">D2</td>
<td style="text-align: right;">145,661</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GID2KO.Day_9.2</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">E4</td>
<td style="text-align: right;">28,048</td>
</tr>
<tr class="even">
<td style="text-align: left;">GID2KO.Day_9.4</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">D8</td>
<td style="text-align: right;">38,192</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GID2KO.Day_12.1</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">L2</td>
<td style="text-align: right;">298,184</td>
</tr>
<tr class="even">
<td style="text-align: left;">GID8KO.Day_3.1</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">F1</td>
<td style="text-align: right;">162,145</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GID8KO.Day_3.1</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">G1</td>
<td style="text-align: right;">51,027</td>
</tr>
<tr class="even">
<td style="text-align: left;">GID8KO.Day_9.1</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">F2</td>
<td style="text-align: right;">100,783</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GID8KO.Day_9.1</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">G2</td>
<td style="text-align: right;">32,021</td>
</tr>
<tr class="even">
<td style="text-align: left;">GID8KO.Day_9.2</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">G4</td>
<td style="text-align: right;">201,457</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GID9KO.Day_6.2</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">N15</td>
<td style="text-align: right;">62,057</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT.Day_6.1</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">J1</td>
<td style="text-align: right;">37,751</td>
</tr>
</tbody>
</table>
</div>
<p>Figure <a href="#fig:plate-map">4</a> highlights on the 384-well plate layout which replicates have fewer reads than the maximum number of reads from an empty well (nreads = 312,601), showing that the these cluster on the leftmost side of the plate (columns 1-2 in particular).</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotPlatePosition.html'>plotPlatePosition</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>,</span>
<span>  <span class='fu'><a href='https://rdrr.io/r/base/character.html'>as.character</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>well_position</span><span class='op'>)</span>,</span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  point_size <span class='op'>=</span> <span class='fl'>5</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/AsIs.html'>I</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>nreads</span> <span class='op'>&lt;=</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>nreads</span><span class='op'>[</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_type</span> <span class='op'>==</span> <span class='st'>"empty"</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  shape_by <span class='op'>=</span> <span class='st'>"sample_type"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span></span>
<span>    colour <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_legend.html'>guide_legend</a></span><span class='op'>(</span></span>
<span>      override.aes <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>3</span>, alpha <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>, </span>
<span>      title <span class='op'>=</span> <span class='st'>"nreads &lt;= cutoff"</span><span class='op'>)</span>,</span>
<span>    shape <span class='op'>=</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guide_legend.html'>guide_legend</a></span><span class='op'>(</span></span>
<span>      override.aes <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='fl'>3</span>, alpha <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>,</span>
<span>      title <span class='op'>=</span> <span class='st'>"sample_type"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>""</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>""</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:plate-map"></span>
<img src="G000396_Danu.preprocess_files/plate-map-1.png" alt="384-well plate layout highlighting wells with fewer reads than   the maximum number of reads from an 'empty' well." width="624" />
<p class="caption">
Figure 4: 384-well plate layout highlighting wells with fewer reads than the maximum number of reads from an empty well.
</p>
</div>
</div>
<h2 id="summary">Summary</h2>
<p>We achieved our aim of roughly 5 million reads/replicate <em>pre-alignment</em> with 3.1 reads/replicate <em>post-alignment</em>.
There is some notable variation in the number of reads generated, but 91% replicates achieved at least 1 million reads/replicate.</p>
<p><strong>Henceforth, we only use the exonic reads.</strong>
Although the intronic reads are potentially useful, to do so would require substantial changes to the data analysis workflow and additional analysis.</p>
<h1 id="analysis-of-empty-wells">Analysis of empty wells</h1>
<h2 id="background">Background</h2>
<p>As seen in Figure <a href="#fig:nreads">1</a>, the empty wells have reads with a median of 53,063 reads, but as many as 312,601 reads.
Wed obviously hope/expect that empty wells should not have any reads assigned to them, but this has been observed in previous CEL-Seq2 experiments.</p>
<p>Here, we explore the reads from these empty wells to see what we can learn from them.</p>
<h2 id="analysis-1">Analysis</h2>
<p>We estimate the ambient profile from the empty wells and look at the most highly expressed genes in the profile.
Almost a third of the reads counts in the ambient profile come from <em>PF3D7_0726000.1</em>, an rRNA.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>reads_ap</span> <span class='op'>&lt;-</span> <span class='fu'>DropletUtils</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DropletUtils/man/ambientProfileEmpty.html'>ambientProfileEmpty</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span>,</span>
<span>  known.empty <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/which.html'>which</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_type</span> <span class='op'>==</span> <span class='st'>"empty"</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>Contribution <span class='op'>=</span> <span class='fu'><a href='https://scales.r-lib.org/reference/percent_format.html'>percent</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/sort.html'>sort</a></span><span class='op'>(</span><span class='va'>reads_ap</span>, decreasing <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  caption <span class='op'>=</span> <span class='st'>"Most highly expressed genes (using read counts) in the ambient profile and the corresponding contribution of counts from that gene to the profile."</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-14">Table 5: </span>Most highly expressed genes (using read counts) in the ambient profile and the corresponding contribution of counts from that gene to the profile.</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Contribution</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">PF3D7_0726000.1</td>
<td style="text-align: left;">30.64%</td>
</tr>
<tr class="even">
<td style="text-align: left;">PF3D7_0532000.1</td>
<td style="text-align: left;">6.98%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PF3D7_0406200.1</td>
<td style="text-align: left;">2.20%</td>
</tr>
<tr class="even">
<td style="text-align: left;">PF3D7_0830200.1</td>
<td style="text-align: left;">1.73%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PF3D7_1038400.1</td>
<td style="text-align: left;">1.61%</td>
</tr>
<tr class="even">
<td style="text-align: left;">PF3D7_0112700.1</td>
<td style="text-align: left;">1.47%</td>
</tr>
</tbody>
</table>
</div>
<h2 id="summary-1">Summary</h2>
<p>We defer further investigation of the reads from the empty wells and remove the empty wells from the dataset.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='va'>sce</span><span class='op'>[</span>, <span class='va'>sce</span><span class='op'>$</span><span class='va'>sample_type</span> <span class='op'>!=</span> <span class='st'>"empty"</span><span class='op'>]</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/droplevels.html'>droplevels</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<h1 id="have-we-sequenced-enough">Have we sequenced enough?</h1>
<h2 id="motivation-1">Motivation</h2>
<p>SCOREs mini-bulk protocol is still being finalised and some technical questions remain.
In particular, we want to know if we have sequenced enough to answer the biological questions of interest.</p>
<p>One way of estimating if we have sequenced enough uses a metric called sequencing saturation.
10x Genomics have some useful information about sequencing saturation available <a href="https://kb.10xgenomics.com/hc/en-us/articles/115005062366-What-is-sequencing-saturation-">here</a>.
In particular,</p>
<blockquote>
<p>Sequencing saturation is a measure of the fraction of library complexity that was sequenced in a given experiment. The inverse of the sequencing saturation can be interpreted as the number of additional reads it would take to detect a new transcript.  Sequencing saturation is dependent on the library complexity and sequencing depth. Different cell types will have different amounts of RNA and thus will differ in the total number of different transcripts in the final library (also known as library complexity).  As sequencing depth increases, more genes are detected, but this reaches saturation at different sequencing depths depending on cell type.  Sequencing depth also affects sequencing saturation; generally, the more sequencing reads, the more additional unique transcripts you can detect. However, this is limited by the library complexity.</p>
</blockquote>
<blockquote>
<p>This naturally leads to the question of <a href="https://kb.10xgenomics.com/hc/en-us/articles/115002474263-How-much-sequencing-saturation-should-I-aim-for-">how much sequencing saturation should I aim for?</a>, which very much depends on the goals of the experiment.</p>
</blockquote>
<p>For this experiment, where we wish to assess differential expression between experimental groups, we will likely want quite high sequencing saturation.</p>
<h2 id="analysis-2">Analysis</h2>
<p>To begin, we simply examine the number of reads sequenced/sample.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>reads_ls</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log10</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>colSums</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:read-library-size">5</a> plots the distribution of the number of reads mapped to exons in each sample<a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a>.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>reads_ls</span><span class='op'>)</span>,</span>
<span>  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>log</span><span class='op'>[</span><span class='fl'>10</span><span class='op'>]</span><span class='op'>(</span><span class='va'>n</span><span class='op'>[</span><span class='va'>reads</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  ylab <span class='op'>=</span> <span class='st'>"density"</span>,</span>
<span>  xlim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>reads_ls</span><span class='op'>)</span>,</span>
<span>  main <span class='op'>=</span> <span class='st'>""</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:read-library-size"></span>
<img src="G000396_Danu.preprocess_files/read-library-size-1.png" alt="Distribution of the number of reads mapped to exons in each   sample." width="624" />
<p class="caption">
Figure 5: Distribution of the number of reads mapped to exons in each sample.
</p>
</div>
</div>
<p>We estimate the sequencing saturation using <a href="https://kb.10xgenomics.com/hc/en-us/articles/115003646912-How-is-sequencing-saturation-calculated-">10x Genomics formula</a>,</p>
<p><span class="math display">\[
1 - (n_{deduped reads} / n_{reads})
\]</span>
where:</p>
<ul>
<li><span class="math inline">\(n_{deduped reads}\)</span> = Number of unique (valid cell-barcode, valid UMI, gene)-combinations among confidently mapped reads</li>
<li><span class="math inline">\(n_{reads}\)</span> = Total number of confidently mapped, valid cell-barcode, valid UMI reads</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>saturation</span> <span class='op'>&lt;-</span> <span class='fl'>1</span> <span class='op'>-</span> </span>
<span>  <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>colSums</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>colSums</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:saturation">6</a> shows that almost all samples are sequenced close to saturation, with a median of 89.7% saturation.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span></span>
<span>    <span class='va'>saturation</span>,</span>
<span>    from <span class='op'>=</span> <span class='fl'>0</span>,</span>
<span>    to <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>,</span>
<span>  xlab <span class='op'>=</span> <span class='st'>"saturation"</span>,</span>
<span>  ylab <span class='op'>=</span> <span class='st'>"density"</span>,</span>
<span>  xlim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>1</span><span class='op'>)</span>,</span>
<span>  main <span class='op'>=</span> <span class='st'>"Sequencing saturation"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:saturation"></span>
<img src="G000396_Danu.preprocess_files/saturation-1.png" alt="Distribution of the sequencing saturation." width="624" />
<p class="caption">
Figure 6: Distribution of the sequencing saturation.
</p>
</div>
</div>
<h2 id="summary-2">Summary</h2>
<p>These results show that the libraries are sequenced to near-saturation.</p>
<p>One way to think of sequencing saturation is <a href="https://kb.10xgenomics.com/hc/en-us/articles/115002474263-How-much-sequencing-saturation-should-I-aim-for-">provided by 10x Genomics</a>:</p>
<blockquote>
<p>The inverse of the sequencing saturation can be interpreted as roughly the number of new transcripts you expect to find with one new read. If sequencing saturation is at 50%, it means that every 2 new reads will result in 1 new UMI count (unique transcript) detected. In contrast, 90% sequencing saturation means that 10 new reads are necessary to obtain one new UMI count. If the sequencing saturation is high, additional sequencing would not recover much new information for the library.</p>
</blockquote>
<p>At this stage, it is not necessary to perform additional sequencing of the library.</p>
<h1 id="umi-counts-vs.-read-counts">UMI counts vs.read counts</h1>
<h2 id="motivation-2">Motivation</h2>
<p>A unique feature of SCOREs mini-bulk RNA-sequencing protocol is the incorporation of unique molecular identifiers (UMIs) into the library preparation; an ordinary bulk RNA-seq protocol does not incorporate UMIs.</p>
<p>UMIs are commonly used in single-cell RNA-sequencing protocols, such as the CEL-Seq2 protocol from which the mini-bulk protocol is adapted.
UMIs are molecular tags that are used to detect and quantify unique mRNA transcripts.
In this method, mRNA libraries are generated by fragmentation and reverse-transcribed to cDNA.
Oligo(dT) primers with specific sequencing linkers are added to the cDNA.
Another sequencing linker with a 6 bp random label and an index sequence is added to the 5 end of the template, which is amplified and sequenced.</p>
<aside>
This explanation is adapted from <a href="https://sapac.illumina.com/science/sequencing-method-explorer/kits-and-arrays/umi.html">https://sapac.illumina.com/science/sequencing-method-explorer/kits-and-arrays/umi.html</a>.
</aside>
<p>The UMI count is always less than or equal to the read count and a UMI count of zero also implies a read count of zero.</p>
<p>Pros of using UMIs:</p>
<ul>
<li>Can sequence unique mRNA transcripts</li>
<li>Can detect transcripts occurring at low frequencies</li>
<li>Transcripts can be quantified based on sequencing reads specific to each barcode</li>
</ul>
<p>Cons of using UMIs:</p>
<ul>
<li>SCOREs mini-bulk protocol uses a <span class="math inline">\(k=6\)</span> bp UMI<a href="#fn7" class="footnote-ref" id="fnref7" role="doc-noteref"><sup>7</sup></a> and so (theoretically) the maximum UMI count for each gene is <span class="math inline">\(4^6 = 4096\)</span>.</li>
<li>There is therefore a risk that highly expressed genes may have their expression measurements artificially truncated, particularly if the sample is deeply sequenced.</li>
</ul>
<p>To investigate whether expression measurements are being truncated in this dataset, and decide whether to use the UMI counts or the read counts, we generated and now analyse two count matrices:</p>
<ol type="1">
<li>UMI counts (i.e.deduplicating UMI counts)</li>
<li>Read counts (i.e.not deduplicating UMI counts, equivalent to ignoring the UMIs)</li>
</ol>
<h2 id="analysis-3">Analysis</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>n</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/apply.html'>apply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span>, <span class='fl'>1</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'><a href='https://rdrr.io/r/base/sum.html'>sum</a></span><span class='op'>(</span><span class='va'>x</span> <span class='op'>&gt;</span> <span class='fl'>4</span> <span class='op'>^</span> <span class='fl'>6</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>g</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>n</span><span class='op'>[</span><span class='va'>n</span> <span class='op'>&gt;</span> <span class='fl'>0</span><span class='op'>]</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>Figure <a href="#fig:read-distribution">7</a> plots the distribution of the logarithm of the read counts.
From this we see that the vast majority of read counts are small.
In fact:</p>
<ul>
<li>15.4% of read counts are 0</li>
<li>95% of the read counts are no greater than 1,228</li>
<li>99% of the read counts are no greater than 4,735</li>
<li>99.9% of the read counts are no greater than 39,893</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/as.vector.html'>as.vector</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1L</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>2</span><span class='op'>)</span>, cex <span class='op'>=</span> <span class='fl'>0.5</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>, from <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>min</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span>, to <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>log</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span><span class='op'>(</span><span class='va'>reads</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  main <span class='op'>=</span> <span class='st'>""</span>,</span>
<span>  ylab <span class='op'>=</span> <span class='st'>"density"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>v <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='fl'>4</span> <span class='op'>^</span> <span class='fl'>6</span><span class='op'>)</span>, col <span class='op'>=</span> <span class='st'>"orange"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/stats/ecdf.html'>ecdf</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span>, </span>
<span>  xlim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  main <span class='op'>=</span> <span class='st'>""</span>,</span>
<span>  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>log</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span><span class='op'>(</span><span class='va'>reads</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  ylab <span class='op'>=</span> <span class='st'>"cumulative density"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>v <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='fl'>4</span> <span class='op'>^</span> <span class='fl'>6</span><span class='op'>)</span>, col <span class='op'>=</span> <span class='st'>"orange"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:read-distribution"></span>
<img src="G000396_Danu.preprocess_files/read-distribution-1.png" alt="Distribution of the logarithm of the read counts as density plot   (left) and empirical cumulative distribution function (right). A count of   1 is added to each UMI count to avoid taking the logarithm of zero. The   orange vertical line denotes $4^6 = 4096$." width="624" />
<p class="caption">
Figure 7: Distribution of the logarithm of the read counts as density plot (left) and empirical cumulative distribution function (right). A count of 1 is added to each UMI count to avoid taking the logarithm of zero. The orange vertical line denotes <span class="math inline">\(4^6 = 4096\)</span>.
</p>
</div>
</div>
<p>Only 1.19% of read counts exceed the <span class="math inline">\(4^6 = 4096\)</span> threshold.
However there are 1,207 genes for which at least one sample exceeds the <span class="math inline">\(4^6=4096\)</span> threshold; these genes are shown in the table below.
If we were to use the UMI counts, the genes in this list may have their expression measurements artificially truncated in those samples where the number of copies of the transcript exceed the <span class="math inline">\(4^6=4096\)</span> threshold.</p>
<p>This table shows that many of these genes are mitochondrial and ribosomal protein genes.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>read_e</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span><span class='op'>[</span><span class='va'>g</span>, <span class='op'>]</span><span class='op'>)</span></span>
<span><span class='va'>umi_e</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span><span class='op'>[</span><span class='va'>g</span>, <span class='op'>]</span><span class='op'>)</span></span>
<span><span class='va'>high_count_tbl</span> <span class='op'>&lt;-</span> <span class='fu'>tibble</span><span class='fu'>::</span><span class='fu'><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></span><span class='op'>(</span></span>
<span>  gene <span class='op'>=</span> <span class='va'>g</span>,</span>
<span>  read_median_count <span class='op'>=</span> <span class='fu'>matrixStats</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/matrixStats/man/rowMedians.html'>rowMedians</a></span><span class='op'>(</span><span class='va'>read_e</span><span class='op'>)</span>,</span>
<span>  read_exceed_sum <span class='op'>=</span> <span class='fu'>matrixStats</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/matrixStats/man/rowSums2.html'>rowSums2</a></span><span class='op'>(</span><span class='va'>read_e</span> <span class='op'>&gt;</span> <span class='fl'>4</span> <span class='op'>^</span> <span class='fl'>6</span><span class='op'>)</span>,</span>
<span>  read_exceed_percent <span class='op'>=</span> <span class='va'>read_exceed_sum</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>,</span>
<span>  umi_median_count <span class='op'>=</span> <span class='fu'>matrixStats</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/matrixStats/man/rowMedians.html'>rowMedians</a></span><span class='op'>(</span><span class='va'>umi_e</span><span class='op'>)</span>,</span>
<span>  umi_exceed_sum <span class='op'>=</span> <span class='fu'>matrixStats</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/matrixStats/man/rowSums2.html'>rowSums2</a></span><span class='op'>(</span><span class='va'>umi_e</span> <span class='op'>&gt;</span> <span class='fl'>4</span> <span class='op'>^</span> <span class='fl'>6</span><span class='op'>)</span>,</span>
<span>  umi_exceed_percent <span class='op'>=</span> <span class='va'>umi_exceed_sum</span> <span class='op'>/</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/arrange.html'>arrange</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/IRanges/man/MaskCollection-class.html'>desc</a></span><span class='op'>(</span><span class='va'>read_exceed_sum</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>sketch</span> <span class='op'>&lt;-</span> <span class='fu'>htmltools</span><span class='fu'>::</span><span class='fu'><a href='https://rstudio.github.io/htmltools/reference/withTags.html'>withTags</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/table.html'>table</a></span><span class='op'>(</span></span>
<span>    class <span class='op'>=</span> <span class='st'>"display"</span>,</span>
<span>    <span class='fu'>thead</span><span class='op'>(</span></span>
<span>      <span class='fu'>tr</span><span class='op'>(</span></span>
<span>        <span class='fu'>th</span><span class='op'>(</span>rowspan <span class='op'>=</span> <span class='fl'>2</span>, <span class='st'>"gene"</span><span class='op'>)</span>,</span>
<span>        <span class='fu'>th</span><span class='op'>(</span>colspan <span class='op'>=</span> <span class='fl'>3</span>, <span class='st'>"reads"</span><span class='op'>)</span>,</span>
<span>        <span class='fu'>th</span><span class='op'>(</span>colspan <span class='op'>=</span> <span class='fl'>3</span>, <span class='st'>"UMIs"</span><span class='op'>)</span></span>
<span>      <span class='op'>)</span>,</span>
<span>      <span class='fu'>tr</span><span class='op'>(</span></span>
<span>        <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/lapply.html'>lapply</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"median count"</span>, <span class='st'>"N &gt; 4096"</span>, <span class='st'>"% &gt; 4096"</span><span class='op'>)</span>, <span class='fl'>2</span><span class='op'>)</span>, <span class='va'>th</span><span class='op'>)</span></span>
<span>      <span class='op'>)</span></span>
<span>    <span class='op'>)</span></span>
<span>  <span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/datatable.html'>datatable</a></span><span class='op'>(</span></span>
<span>  <span class='va'>high_count_tbl</span>, </span>
<span>  caption <span class='op'>=</span> <span class='st'>"Genes for which any sample has a read count exceeding the 4096 threshold and summaries of the corresponding read and UMI counts, namely: the median count ('median count'); the number of samples for which the counts exceed the threshold ('N &gt; 4096'); and the percentage of samples for which the counts exceed the threshold ('% &gt; 4096')."</span>,</span>
<span>  container <span class='op'>=</span> <span class='va'>sketch</span>,</span>
<span>  rownames <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/formatCurrency.html'>formatPercentage</a></span><span class='op'>(</span></span>
<span>    columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"read_exceed_percent"</span>, <span class='st'>"umi_exceed_percent"</span><span class='op'>)</span>,</span>
<span>    digits <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'><a href='https://magrittr.tidyverse.org/reference/pipe.html'>%&gt;%</a></span></span>
<span>  <span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/formatCurrency.html'>formatRound</a></span><span class='op'>(</span></span>
<span>    columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"read_median_count"</span>, <span class='st'>"umi_median_count"</span><span class='op'>)</span>,</span>
<span>    digits <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-f0535d693ffd2c26631b" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f0535d693ffd2c26631b">{"x":{"filter":"none","vertical":false,"class":"display","caption":"<caption>Genes for which any sample has a read count exceeding the 4096 threshold and summaries of the corresponding read and UMI counts, namely: the median count ('median count'); the number of samples for which the counts exceed the threshold ('N &gt; 4096'); and the percentage of samples for which the counts exceed the threshold ('% &gt; 4096').<\/caption>","data":[["PF3D7_0726000.1","PF3D7_0532000.1","PF3D7_0830200.1","PF3D7_MIT04000.1","PF3D7_0112700.1","PF3D7_0725600.1","PF3D7_1021900.1","PF3D7_0617900.1","PF3D7_0818900.1","PF3D7_0617800.1","PF3D7_1105100.1","PF3D7_1105000.1","PF3D7_1148640.1","PF3D7_0406200.1","PF3D7_0708400.1","PF3D7_1038400.1","PF3D7_1327300.1","PF3D7_1024800.1","PF3D7_1008700.1","PF3D7_0725800.1","PF3D7_0114000.1","PF3D7_0531600.1","PF3D7_0501400.1","PF3D7_1103100.1","PF3D7_1371300.1","PF3D7_1302100.1","PF3D7_1016300.1","PF3D7_0209000.1","PF3D7_1246200.1","PF3D7_1023900.1","PF3D7_0801900.1","PF3D7_0827900.1","PF3D7_1124900.1","PF3D7_1466800.1","PF3D7_MIT01600.1","PF3D7_MIT02100.1","PF3D7_MIT02200.1","PF3D7_1146800.1","PF3D7_0513600.1","PF3D7_1233600.1","PF3D7_1027300.1","PF3D7_1309400.1","PF3D7_1459800.1","PF3D7_1468100.1","PF3D7_1351600.1","PF3D7_1106800.1","PF3D7_0608800.1","PF3D7_0930300.1","PF3D7_1016900.1","PF3D7_1033900.1","PF3D7_1035200.1","PF3D7_0619400.1","PF3D7_1011800.1","PF3D7_1448500.1","PF3D7_1202600.1","PF3D7_0207600.1","PF3D7_1216500.1","PF3D7_0704600.1","PF3D7_0216000.1","PF3D7_1338800.1","PF3D7_1362600.1","PF3D7_0723400.1","PF3D7_0814200.1","PF3D7_1228800.1","PF3D7_0110000.1","PF3D7_1212700.1","PF3D7_0320900.1","PF3D7_0804500.1","PF3D7_1240600.1","PF3D7_1348400.1","PF3D7_1453700.1","PF3D7_0511500.1","PF3D7_0823200.1","PF3D7_1221400.1","PF3D7_0418300.1","PF3D7_1414300.1","PF3D7_1145200.1","PF3D7_0715200.1","PF3D7_0826100.1","PF3D7_1250100.1","PF3D7_0421300.1","PF3D7_1107300.1","PF3D7_0411000.1","PF3D7_1429300.1","PF3D7_0726500.1","PF3D7_0607700.1","PF3D7_1202900.1","PF3D7_1467600.1","PF3D7_1247800.1","PF3D7_0803400.1","PF3D7_0924600.1","PF3D7_0804700.1","PF3D7_1248700.1","PF3D7_0215000.1","PF3D7_1430200.1","PF3D7_0705500.1","PF3D7_0624600.1","PF3D7_1110400.1","PF3D7_0313600.1","PF3D7_0519500.1","PF3D7_0724900.1","PF3D7_MIT01000.1","PF3D7_0304100.1","PF3D7_0710200.1","PF3D7_1025900.1","PF3D7_1358600.1","PF3D7_1303800.1","PF3D7_0604500.1","PF3D7_0314700.1","PF3D7_0628100.1","PF3D7_0108300.1","PF3D7_1346400.1","PF3D7_1103800.1","PF3D7_1435200.1","PF3D7_0504700.1","PF3D7_1121600.1","PF3D7_0723300.1","PF3D7_1462800.1","PF3D7_0405400.1","PF3D7_0827800.1","PF3D7_1248500.1","PF3D7_1431100.1","PF3D7_1219000.1","PF3D7_0419600.1","PF3D7_0605600.1","PF3D7_1014300.1","PF3D7_1444800.1","PF3D7_1349500.1","PF3D7_0609000.1","PF3D7_1021800.1","PF3D7_0707700.1","PF3D7_1424100.1","PF3D7_0619300.1","PF3D7_0105700.1","PF3D7_0704100.1","PF3D7_0506900.1","PF3D7_1234900.1","PF3D7_1433400.1","PF3D7_0724100.1","PF3D7_0728600.1","PF3D7_0506500.1","PF3D7_0210200.1","PF3D7_1133200.1","PF3D7_1026800.1","PF3D7_1321700.1","PF3D7_1438800.1","PF3D7_0214900.1","PF3D7_0629700.1","PF3D7_0503000.1","PF3D7_0621400.1","PF3D7_1318700.1","PF3D7_0415300.1","PF3D7_1461300.1","PF3D7_1107900.1","PF3D7_0422200.1","PF3D7_1461800.1","PF3D7_1466200.1","PF3D7_0500800.1","PF3D7_0110700.1","PF3D7_0516900.1","PF3D7_1020900.1","PF3D7_MIT02300.1","PF3D7_0110500.1","PF3D7_1136500.1","PF3D7_1362100.1","PF3D7_0918100.1","PF3D7_0519200.1","PF3D7_1234700.1","PF3D7_0202000.1","PF3D7_0309600.1","PF3D7_1361300.1","PF3D7_0602000.1","PF3D7_0504900.1","PF3D7_1206500.1","PF3D7_0613800.1","PF3D7_1306500.1","PF3D7_0508900.1","PF3D7_0623600.1","PF3D7_0530800.1","PF3D7_0719900.1","PF3D7_1235300.1","PF3D7_1033100.1","PF3D7_0522400.1","PF3D7_0418000.1","PF3D7_1240400.1","PF3D7_1009400.1","PF3D7_0802100.1","PF3D7_0423700.1","PF3D7_1410400.1","PF3D7_1362700.1","PF3D7_0516800.1","PF3D7_1474000.1","PF3D7_1147800.1","PF3D7_1144800.1","PF3D7_0720700.1","PF3D7_0934700.1","PF3D7_1249800.1","PF3D7_1474400.1","PF3D7_1347500.1","PF3D7_1302500.1","PF3D7_1205600.1","PF3D7_0507100.1","PF3D7_0709700.1","PF3D7_0628200.1","PF3D7_1220900.1","PF3D7_1434500.1","PF3D7_1123400.1","PF3D7_0214300.1","PF3D7_1020200.1","PF3D7_1319000.1","PF3D7_1230100.1","PF3D7_1115200.1","PF3D7_0312400.1","PF3D7_0816800.1","PF3D7_0317700.1","PF3D7_0309200.1","PF3D7_0802000.1","PF3D7_0919900.1","PF3D7_1250400.1","PF3D7_0917900.1","PF3D7_0828600.1","PF3D7_0417400.1","PF3D7_0205900.1","PF3D7_1147200.1","PF3D7_1408600.1","PF3D7_1135900.1","PF3D7_1130200.1","PF3D7_0612200.1","PF3D7_0518700.1","PF3D7_0612900.1","PF3D7_1322200.1","PF3D7_0814000.1","PF3D7_1221100.1","PF3D7_1227500.1","PF3D7_1334000.1","PF3D7_0305100.1","PF3D7_1210200.1","PF3D7_1116100.1","PF3D7_0903800.1","PF3D7_1343800.1","PF3D7_1412400.1","PF3D7_0406500.1","PF3D7_1123100.1","PF3D7_0609700.1","PF3D7_1129600.1","PF3D7_0825000.1","PF3D7_1471700.1","PF3D7_1010100.1","PF3D7_0813300.1","PF3D7_0611800.1","PF3D7_1236200.1","PF3D7_1033200.1","PF3D7_0621200.1","PF3D7_1319600.1","PF3D7_1337500.1","PF3D7_1435600.1","PF3D7_0527500.1","PF3D7_0710600.1","PF3D7_1148700.1","PF3D7_1014600.1","PF3D7_1222700.1","PF3D7_1417200.1","PF3D7_1120700.1","PF3D7_1211000.1","PF3D7_1416200.1","PF3D7_0706500.1","PF3D7_1451100.1","PF3D7_1437200.1","PF3D7_1356800.1","PF3D7_1038000.1","PF3D7_0527000.1","PF3D7_0831300.1","PF3D7_0904200.1","PF3D7_0311400.1","PF3D7_1126700.1","PF3D7_1034900.1","PF3D7_1471600.1","PF3D7_0903300.1","PF3D7_1361900.1","PF3D7_0719400.1","PF3D7_0311300.1","PF3D7_1457000.1","PF3D7_1431700.1","PF3D7_1431500.1","PF3D7_0422500.1","PF3D7_1006200.1","PF3D7_0515500.1","PF3D7_1105400.1","PF3D7_0911900.1","PF3D7_1126800.1","PF3D7_0607000.1","PF3D7_0208100.1","PF3D7_1323700.1","PF3D7_1029900.1","PF3D7_0412700.1","PF3D7_0615900.1","PF3D7_0919800.1","PF3D7_1228600.1","PF3D7_1019100.1","PF3D7_1021700.1","PF3D7_1113100.1","PF3D7_0606600.1","PF3D7_1143100.1","PF3D7_1207700.1","PF3D7_1222300.1","PF3D7_0614700.1","PF3D7_0525000.1","PF3D7_0319400.1","PF3D7_0109400.1","PF3D7_1410300.1","PF3D7_1202100.2","PF3D7_1322900.1","PF3D7_0103400.1","PF3D7_1201600.1","PF3D7_1117700.1","PF3D7_0926400.1","PF3D7_0904000.1","PF3D7_1403100.1","PF3D7_1347200.1","PF3D7_1342000.1","PF3D7_1023100.1","PF3D7_0624300.1","PF3D7_1008400.1","PF3D7_1341800.1","PF3D7_0707300.1","PF3D7_1008100.1","PF3D7_0103900.1","PF3D7_1420700.1","PF3D7_1239800.1","PF3D7_1330500.1","PF3D7_0510100.1","PF3D7_1319400.1","PF3D7_0929200.1","PF3D7_1222400.1","PF3D7_1037300.1","PF3D7_1252100.1","PF3D7_0204100.1","PF3D7_0107500.1","PF3D7_1103500.1","PF3D7_1223400.1","PF3D7_0517400.1","PF3D7_1421800.1","PF3D7_1325400.1","PF3D7_1323100.1","PF3D7_0310200.1","PF3D7_0111000.1","PF3D7_0704300.1","PF3D7_1365900.1","PF3D7_0416400.1","PF3D7_0203100.1","PF3D7_0613500.1","PF3D7_0208400.1","PF3D7_0630300.1","PF3D7_1211400.1","PF3D7_1312800.1","PF3D7_0827200.1","PF3D7_0725000.1","PF3D7_1141900.1","PF3D7_1203000.1","PF3D7_0608600.1","PF3D7_1214800.1","PF3D7_0716200.1","PF3D7_1437900.1","PF3D7_1409100.1","PF3D7_0815800.1","PF3D7_0825800.1","PF3D7_1012300.1","PF3D7_1027800.1","PF3D7_1406200.1","PF3D7_1304600.1","PF3D7_1350100.1","PF3D7_0909000.1","PF3D7_0817500.1","PF3D7_0801100.1","PF3D7_0104300.1","PF3D7_0927200.1","PF3D7_1141100.1","PF3D7_1361800.1","PF3D7_0404600.1","PF3D7_1211300.1","PF3D7_0721000.1","PF3D7_1413200.1","PF3D7_1414500.1","PF3D7_0420000.1","PF3D7_0203800.1","PF3D7_1447000.1","PF3D7_1417600.1","PF3D7_1343300.1","PF3D7_1122900.1","PF3D7_0420300.1","PF3D7_0518600.1","PF3D7_0810600.1","PF3D7_0729100.1","PF3D7_1423700.1","PF3D7_1467900.1","PF3D7_1408700.1","PF3D7_1118300.1","PF3D7_0708800.1","PF3D7_0922800.1","PF3D7_0812400.1","PF3D7_0112300.1","PF3D7_1123000.1","PF3D7_1369300.1","PF3D7_MIT00900.1","PF3D7_1472200.1","PF3D7_1020000.1","PF3D7_0210900.1","PF3D7_0714000.1","PF3D7_0408500.1","PF3D7_0829500.1","PF3D7_0110600.1","PF3D7_1240900.1","PF3D7_0728100.1","PF3D7_0414000.1","PF3D7_0616200.1","PF3D7_1124000.1","PF3D7_0712600.1","PF3D7_0109100.1","PF3D7_1306400.1","PF3D7_1202300.1","PF3D7_0718000.1","PF3D7_0320800.1","PF3D7_1014900.1","PF3D7_0905300.1","PF3D7_1409900.1","PF3D7_1208900.1","PF3D7_1451800.1","PF3D7_0406900.1","PF3D7_1227700.1","PF3D7_0801800.1","PF3D7_0928100.1","PF3D7_1341300.1","PF3D7_1324300.1","PF3D7_1148000.1","PF3D7_1133700.1","PF3D7_0931400.1","PF3D7_0723800.1","PF3D7_1223500.1","PF3D7_0212500.1","PF3D7_1329100.1","PF3D7_1468800.1","PF3D7_1005500.1","PF3D7_1446500.1","PF3D7_1407300.1","PF3D7_0305200.1","PF3D7_0935200.1","PF3D7_0414500.1","PF3D7_0930800.1","PF3D7_0820000.1","PF3D7_0110400.1","PF3D7_1471500.1","PF3D7_1432700.1","PF3D7_1469900.1","PF3D7_1430300.1","PF3D7_0529400.2","PF3D7_0907500.1","PF3D7_1142100.1","PF3D7_0411900.1","PF3D7_0318500.1","PF3D7_0623100.1","PF3D7_0718100.1","PF3D7_0316500.1","PF3D7_0206300.1","PF3D7_1303500.1","PF3D7_0504800.1","PF3D7_0530900.1","PF3D7_1033600.1","PF3D7_0317200.1","PF3D7_1450400.1","PF3D7_1359400.1","PF3D7_1456500.1","PF3D7_0923400.1","PF3D7_1002400.2","PF3D7_1416600.1","PF3D7_1445700.1","PF3D7_1104900.1","PF3D7_1308900.1","PF3D7_0303200.1","PF3D7_0927600.1","PF3D7_0307900.1","PF3D7_0709100.1","PF3D7_1436300.1","PF3D7_1218800.1","PF3D7_1359600.1","PF3D7_0627100.1","PF3D7_1130100.1","PF3D7_0716000.1","PF3D7_1235700.1","PF3D7_1206200.1","PF3D7_0924400.1","PF3D7_0217100.1","PF3D7_1036900.1","PF3D7_1447200.1","PF3D7_1419700.1","PF3D7_1012400.1","PF3D7_1326300.1","PF3D7_1455000.1","PF3D7_1356100.1","PF3D7_0708500.1","PF3D7_1243900.1","PF3D7_1345400.1","PF3D7_0603600.1","PF3D7_1340300.1","PF3D7_1239200.1","PF3D7_0925800.1","PF3D7_1108600.1","PF3D7_1003400.1","PF3D7_1138400.1","PF3D7_1361200.1","PF3D7_1342900.1","PF3D7_1130700.1","PF3D7_1210600.1","PF3D7_1247400.1","PF3D7_1428400.1","PF3D7_0517200.1","PF3D7_1204300.1","PF3D7_1139300.1","PF3D7_1128300.1","PF3D7_0615500.1","PF3D7_0214400.1","PF3D7_0722800.1","PF3D7_1438900.1","PF3D7_1120600.1","PF3D7_1330800.1","PF3D7_1452900.1","PF3D7_1021600.1","PF3D7_1444100.1","PF3D7_0904600.1","PF3D7_1140700.1","PF3D7_1326600.1","PF3D7_0813400.1","PF3D7_MIT01400.1","PF3D7_1230000.1","PF3D7_1345000.1","PF3D7_1329700.1","PF3D7_0924500.1","PF3D7_0415900.1","PF3D7_0626400.1","PF3D7_0730500.1","PF3D7_1439300.1","PF3D7_1122600.1","PF3D7_1248400.1","PF3D7_0615400.1","PF3D7_1405900.1","PF3D7_1134000.1","PF3D7_1325200.1","PF3D7_0410700.1","PF3D7_1130800.1","PF3D7_0616800.1","PF3D7_1136300.1","PF3D7_0610900.1","PF3D7_0726400.1","PF3D7_1108000.1","PF3D7_1205500.1","PF3D7_0807800.1","PF3D7_MIT01500.1","PF3D7_0816600.1","PF3D7_1006000.1","PF3D7_0601900.1","PF3D7_1366400.1","PF3D7_0306900.1","PF3D7_1416400.1","PF3D7_0729900.1","PF3D7_0806200.1","PF3D7_1346700.1","PF3D7_MIT04200.1","PF3D7_0322700.1","PF3D7_0808300.1","PF3D7_1006700.1","PF3D7_1118200.1","PF3D7_1447100.1","PF3D7_0918500.1","PF3D7_0320200.1","PF3D7_0826500.1","PF3D7_0915100.1","PF3D7_1225600.1","PF3D7_0417800.1","PF3D7_1431200.1","PF3D7_1004400.1","PF3D7_0302100.1","PF3D7_1311900.1","PF3D7_1308400.1","PF3D7_1372100.1","PF3D7_0903900.1","PF3D7_1132200.1","PF3D7_1234800.1","PF3D7_MIT03200.1","PF3D7_0525900.1","PF3D7_1330600.1","PF3D7_1010400.1","PF3D7_1246100.1","PF3D7_1424400.1","PF3D7_0507800.1","PF3D7_1409400.1","PF3D7_0718500.1","PF3D7_0817700.1","PF3D7_0509800.1","PF3D7_1216900.1","PF3D7_0723900.1","PF3D7_0521900.1","PF3D7_0621100.1","PF3D7_0715400.1","PF3D7_1461600.1","PF3D7_1318400.1","PF3D7_1203700.1","PF3D7_1471100.1","PF3D7_1149000.1","PF3D7_0818500.1","PF3D7_0703500.1","PF3D7_1366300.1","PF3D7_1320700.1","PF3D7_1022500.1","PF3D7_1370800.1","PF3D7_0305600.1","PF3D7_1349300.1","PF3D7_0923900.1","PF3D7_0614300.1","PF3D7_1448400.1","PF3D7_1208800.1","PF3D7_1338300.1","PF3D7_1104500.1","PF3D7_0520900.1","PF3D7_1102500.1","PF3D7_0516100.1","PF3D7_1246400.1","PF3D7_0410300.1","PF3D7_0624400.1","PF3D7_1309100.1","PF3D7_0926500.1","PF3D7_0104400.1","PF3D7_0307700.1","PF3D7_0607100.1","PF3D7_0931700.1","PF3D7_0317300.1","PF3D7_1111100.1","PF3D7_0915400.1","PF3D7_1407800.1","PF3D7_API04900.1","PF3D7_1342600.1","PF3D7_1025000.1","PF3D7_0628300.1","PF3D7_0806800.1","PF3D7_1019000.1","PF3D7_0406100.1","PF3D7_0508200.1","PF3D7_0728700.1","PF3D7_1473200.1","PF3D7_1445900.1","PF3D7_1142300.1","PF3D7_1010700.1","PF3D7_0219600.1","PF3D7_1353300.1","PF3D7_1132000.1","PF3D7_0212800.1","PF3D7_1014700.1","PF3D7_1446600.1","PF3D7_0526600.1","PF3D7_1037000.1","PF3D7_1107800.1","PF3D7_0406000.1","PF3D7_0503200.1","PF3D7_1127500.1","PF3D7_0719800.1","PF3D7_0517700.1","PF3D7_0818200.1","PF3D7_0209800.1","PF3D7_0610800.1","PF3D7_1114900.1","PF3D7_1324800.1","PF3D7_0622200.1","PF3D7_0517800.1","PF3D7_1227800.1","PF3D7_1475400.1","PF3D7_1408100.1","PF3D7_0423600.1","PF3D7_0926300.1","PF3D7_1464600.1","PF3D7_0311700.1","PF3D7_0412900.1","PF3D7_0508700.1","PF3D7_0210000.1","PF3D7_0407800.1","PF3D7_0108600.1","PF3D7_1418000.1","PF3D7_0528400.1","PF3D7_1227600.1","PF3D7_0214100.1","PF3D7_0803800.1","PF3D7_0322300.1","PF3D7_0517500.1","PF3D7_1467200.1","PF3D7_0512000.1","PF3D7_0804000.1","PF3D7_0706000.1","PF3D7_0413100.1","PF3D7_0818700.1","PF3D7_1430800.1","PF3D7_1011700.1","PF3D7_0932200.1","PF3D7_0704800.1","PF3D7_0909500.1","PF3D7_1006900.1","PF3D7_0911000.1","PF3D7_1343000.1","PF3D7_0507600.1","PF3D7_0211800.1","PF3D7_0716800.1","PF3D7_1108700.1","PF3D7_1316000.1","PF3D7_0703800.1","PF3D7_1004900.1","PF3D7_0505800.1","PF3D7_0903500.1","PF3D7_0213100.1","PF3D7_1360800.1","PF3D7_0606700.1","PF3D7_0206800.1","PF3D7_1209200.1","PF3D7_1454400.1","PF3D7_1469600.1","PF3D7_1105700.1","PF3D7_0717600.1","PF3D7_0322000.1","PF3D7_1426700.1","PF3D7_0907600.1","PF3D7_0627300.1","PF3D7_1309000.1","PF3D7_0110800.1","PF3D7_0515600.1","PF3D7_0607200.1","PF3D7_0105800.1","PF3D7_0912100.1","PF3D7_0914800.1","PF3D7_0918000.1","PF3D7_0914100.1","PF3D7_0913200.1","PF3D7_0106700.1","PF3D7_0303500.1","PF3D7_0308200.1","PF3D7_1028700.1","PF3D7_0403900.1","PF3D7_1243600.1","PF3D7_0903600.1","PF3D7_0819600.1","PF3D7_0727000.1","PF3D7_0802600.1","PF3D7_1466500.1","PF3D7_0722700.1","PF3D7_0515300.1","PF3D7_0801300.1","PF3D7_1030100.1","PF3D7_1025500.1","PF3D7_1344300.1","PF3D7_1238800.1","PF3D7_1473400.1","PF3D7_0532100.1","PF3D7_1035400.1","PF3D7_1331400.1","PF3D7_1426500.1","PF3D7_0520000.1","PF3D7_0409100.1","PF3D7_1357000.1","PF3D7_1023600.1","PF3D7_1012100.1","PF3D7_0936500.1","PF3D7_0304600.1","PF3D7_0305800.1","PF3D7_0508100.1","PF3D7_0904900.1","PF3D7_0213600.1","PF3D7_0929400.1","PF3D7_0509100.1","PF3D7_0817600.1","PF3D7_1435300.1","PF3D7_0321800.1","PF3D7_0406700.1","PF3D7_1221300.1","PF3D7_1120000.1","PF3D7_1132400.1","PF3D7_1316500.1","PF3D7_0614500.1","PF3D7_1452700.1","PF3D7_1229500.1","PF3D7_1472900.1","PF3D7_0111800.1","PF3D7_0519800.1","PF3D7_0711700.1","PF3D7_1234100.1","PF3D7_0807600.1","PF3D7_0528500.1","PF3D7_1229600.1","PF3D7_1317100.1","PF3D7_0514100.1","PF3D7_0928200.1","PF3D7_1453500.1","PF3D7_0217500.1","PF3D7_1019700.1","PF3D7_0803100.1","PF3D7_1451600.1","PF3D7_1472700.1","PF3D7_1231600.1","PF3D7_1310700.1","PF3D7_1021200.1","PF3D7_0409800.1","PF3D7_0717500.1","PF3D7_0422300.1","PF3D7_1368100.1","PF3D7_1407100.1","PF3D7_1134600.1","PF3D7_0212100.1","PF3D7_1108100.1","PF3D7_1316100.1","PF3D7_0922200.1","PF3D7_1004300.1","PF3D7_0502300.1","PF3D7_1026100.1","PF3D7_1338100.1","PF3D7_1457300.1","PF3D7_1228900.1","PF3D7_1015900.1","PF3D7_0826700.1","PF3D7_1202100.1","PF3D7_0515700.1","PF3D7_0418200.1","PF3D7_0311600.1","PF3D7_1423600.1","PF3D7_1125000.1","PF3D7_1420000.1","PF3D7_0806700.1","PF3D7_0501800.1","PF3D7_0614900.1","PF3D7_0106000.1","PF3D7_0316600.1","PF3D7_0624200.1","PF3D7_1309300.1","PF3D7_1338900.1","PF3D7_1031500.1","PF3D7_0935900.1","PF3D7_1446900.1","PF3D7_0809600.1","PF3D7_1438700.1","PF3D7_0310400.1","PF3D7_0206000.1","PF3D7_1434300.1","PF3D7_0513000.1","PF3D7_1138000.1","PF3D7_0412000.1","PF3D7_0603800.1","PF3D7_1019300.1","PF3D7_0812500.1","PF3D7_1414000.1","PF3D7_0703900.1","PF3D7_1473700.1","PF3D7_0800700.1","PF3D7_1448300.1","PF3D7_1317200.1","PF3D7_0809900.1","PF3D7_0418800.1","PF3D7_1366900.1","PF3D7_1031600.1","PF3D7_1215100.1","PF3D7_1141300.1","PF3D7_0302500.1","PF3D7_1466600.1","PF3D7_0931500.1","PF3D7_0514700.1","PF3D7_0420700.1","PF3D7_0627500.1","PF3D7_1207100.1","PF3D7_1129900.1","PF3D7_1451700.1","PF3D7_0310500.1","PF3D7_0617100.1","PF3D7_1231400.1","PF3D7_1019500.1","PF3D7_1102800.1","PF3D7_0218200.1","PF3D7_0934400.1","PF3D7_1461700.1","PF3D7_0624900.1","PF3D7_1352500.1","PF3D7_1237700.1","PF3D7_0323100.1","PF3D7_0615600.1","PF3D7_1215400.1","PF3D7_1311100.1","PF3D7_1132800.1","PF3D7_0803600.1","PF3D7_0301600.1","PF3D7_1018200.1","PF3D7_1209000.1","PF3D7_1118800.1","PF3D7_0931900.1","PF3D7_1231000.1","PF3D7_0319700.1","PF3D7_1425800.1","PF3D7_0508600.1","PF3D7_1032100.1","PF3D7_0416100.1","PF3D7_1119100.1","PF3D7_1450700.1","PF3D7_1318300.1","PF3D7_1368400.1","PF3D7_1417400.1","PF3D7_1143500.1","PF3D7_1429200.1","PF3D7_1356300.1","PF3D7_1441300.1","PF3D7_1419300.1","PF3D7_1423400.1","PF3D7_0316200.1","PF3D7_1003700.1","PF3D7_0617200.1","PF3D7_1465000.1","PF3D7_1250800.1","PF3D7_1428500.1","PF3D7_1314700.1","PF3D7_1026400.1","PF3D7_1110200.1","PF3D7_0808600.1","PF3D7_1236100.1","PF3D7_1477700.1","PF3D7_0318200.1","PF3D7_1117900.1","PF3D7_0605100.1","PF3D7_0831800.1","PF3D7_1113000.1","PF3D7_0522900.1","PF3D7_1436100.1","PF3D7_0812100.1","PF3D7_1024900.1","PF3D7_0106500.1","PF3D7_1114700.1","PF3D7_1309800.1","PF3D7_1009700.1","PF3D7_1313300.1","PF3D7_0320300.1","PF3D7_1412900.1","PF3D7_0417600.1","PF3D7_1328000.1","PF3D7_1355900.1","PF3D7_0103200.1","PF3D7_1224300.1","PF3D7_1362900.1","PF3D7_1143400.1","PF3D7_1001600.1","PF3D7_0220000.1","PF3D7_1410100.1","PF3D7_0604100.1","PF3D7_1457100.1","PF3D7_0608100.1","PF3D7_1022200.1","PF3D7_1205900.1","PF3D7_0914700.1","PF3D7_0420900.1","PF3D7_0219400.1","PF3D7_1241400.1","PF3D7_0830100.1","PF3D7_0528200.1","PF3D7_1237200.1","PF3D7_0310300.1","PF3D7_0905400.1","PF3D7_0811300.1","PF3D7_0908500.1","PF3D7_1004600.1","PF3D7_1104300.1","PF3D7_1107700.1","PF3D7_1414400.1","PF3D7_1241900.1","PF3D7_0809200.1","PF3D7_0207800.1","PF3D7_0510500.1","PF3D7_0703200.1","PF3D7_1442900.1","PF3D7_1433500.1","PF3D7_1217200.1","PF3D7_0618300.1","PF3D7_0316800.1","PF3D7_1405100.1","PF3D7_1442700.1","PF3D7_1341900.1","PF3D7_1405800.1","PF3D7_1325900.1","PF3D7_1309700.1","PF3D7_1309200.1","PF3D7_0828200.1","PF3D7_1325800.1","PF3D7_0505700.1","PF3D7_1002200.1","PF3D7_1331700.1","PF3D7_0525600.1","PF3D7_1200600.1","PF3D7_1308700.1","PF3D7_0315600.1","PF3D7_0724700.1","PF3D7_0913700.1","PF3D7_1336700.1","PF3D7_1228300.1","PF3D7_1403800.1","PF3D7_0911100.1","PF3D7_1250600.1","PF3D7_1427900.1","PF3D7_1432400.1","PF3D7_1335900.1","PF3D7_0405700.1","PF3D7_0909600.1","PF3D7_1026300.1","PF3D7_1440500.1","PF3D7_0529800.1","PF3D7_1109400.1","PF3D7_1109100.1","PF3D7_0820700.1","PF3D7_1347100.1","PF3D7_0509700.1","PF3D7_0818000.1","PF3D7_1129100.1","PF3D7_1407000.1","PF3D7_1205800.1","PF3D7_1245800.1","PF3D7_1131800.1","PF3D7_1453800.1","PF3D7_1356000.1","PF3D7_1460400.1","PF3D7_1236900.1","PF3D7_1327000.1","PF3D7_1005600.1","PF3D7_0719300.1","PF3D7_1322100.1","PF3D7_0717200.1","PF3D7_0515400.1","PF3D7_1311400.1","PF3D7_0717800.1","PF3D7_0826300.1","PF3D7_1228700.1","PF3D7_1476600.1","PF3D7_1125300.1","PF3D7_1018600.1","PF3D7_0926900.1","PF3D7_1212900.1","PF3D7_0909900.1","PF3D7_1004200.1","PF3D7_1402200.1","PF3D7_1430100.1","PF3D7_1438000.1","PF3D7_0926100.1","PF3D7_0711500.1","PF3D7_1018400.1","PF3D7_1433900.1","PF3D7_1031900.1","PF3D7_1459000.1","PF3D7_0731800.1","PF3D7_0620700.1","PF3D7_1003600.1","PF3D7_1134300.1","PF3D7_0521000.1","PF3D7_0417300.1","PF3D7_0518800.1","PF3D7_1117800.1","PF3D7_1457900.1","PF3D7_0904100.1","PF3D7_0408700.1","PF3D7_0218700.1","PF3D7_0207100.1","PF3D7_1348800.1","PF3D7_1207600.1","PF3D7_0410800.1","PF3D7_0626800.1","PF3D7_1123500.1","PF3D7_0411200.1","PF3D7_1232200.1","PF3D7_0207700.1","PF3D7_0410900.1","PF3D7_1343700.1","PF3D7_1219100.1","PF3D7_0403400.1","PF3D7_1360500.1","PF3D7_1350900.1","PF3D7_1347700.1","PF3D7_0416900.1","PF3D7_1312900.1","PF3D7_0927700.1","PF3D7_0213900.1","PF3D7_1336200.1","PF3D7_1336800.1","PF3D7_0819800.1","PF3D7_1003100.1","PF3D7_0725200.1","PF3D7_1428300.1","PF3D7_0528100.1","PF3D7_1229800.1","PF3D7_0202500.1","PF3D7_1017000.1","PF3D7_1207000.1","PF3D7_1206300.1","PF3D7_1429700.1","PF3D7_0323800.1","PF3D7_1358800.1","PF3D7_1324600.1","PF3D7_1477300.1","PF3D7_0317500.1","PF3D7_1140600.1","PF3D7_0517300.1","PF3D7_0308000.1","PF3D7_1205100.1","PF3D7_1351000.1","PF3D7_0815600.1","PF3D7_0801700.1","PF3D7_0510000.1","PF3D7_1124600.1","PF3D7_0509400.1","PF3D7_1118700.1","PF3D7_1206000.1","PF3D7_1017400.1","PF3D7_1324200.1","PF3D7_1031000.1","PF3D7_0926800.1","PF3D7_1008000.1","PF3D7_0924000.1","PF3D7_0922100.1","PF3D7_1104000.1","PF3D7_0623800.1","PF3D7_0518500.1","PF3D7_0516600.1","PF3D7_0303900.1","PF3D7_0215400.1","PF3D7_0303700.1","PF3D7_1216700.1","PF3D7_0108700.1","PF3D7_1475500.1","PF3D7_0203000.1","PF3D7_1206700.1","PF3D7_1427100.1","PF3D7_1357400.1","PF3D7_0404900.1","PF3D7_0903400.1","PF3D7_1367300.1","PF3D7_0909700.1","PF3D7_0407700.1","PF3D7_1470950.1","PF3D7_1245600.1","PF3D7_0619000.1","PF3D7_0314000.1","PF3D7_0321600.1","PF3D7_0111400.1","PF3D7_1305300.1","PF3D7_1315400.1","PF3D7_1359900.1","PF3D7_1465800.1","PF3D7_0602100.1","PF3D7_1136000.1","PF3D7_0404300.1","PF3D7_1133800.1","PF3D7_0511000.1","PF3D7_0305500.1","PF3D7_1303400.1","PF3D7_1321000.1","PF3D7_1407600.1","PF3D7_1116800.1","PF3D7_0929600.1","PF3D7_1442600.1","PF3D7_0203700.1","PF3D7_0611900.1","PF3D7_0809700.1","PF3D7_0802800.1","PF3D7_0319500.1","PF3D7_0208600.1","PF3D7_0933600.1","PF3D7_0523400.1","PF3D7_0912200.1","PF3D7_1223100.1","PF3D7_1233800.1","PF3D7_0618900.1","PF3D7_1144200.1","PF3D7_0207400.1","PF3D7_1360900.1","PF3D7_1112100.1","PF3D7_0505100.1","PF3D7_1422400.1","PF3D7_1451200.1","PF3D7_1127900.1","PF3D7_1147100.1","PF3D7_1125200.1","PF3D7_0623900.1","PF3D7_1405600.1","PF3D7_1211800.1","PF3D7_0805300.1","PF3D7_1032000.1","PF3D7_1449500.1","PF3D7_0421100.1","PF3D7_0605800.1","PF3D7_0705300.1","PF3D7_0930000.1","PF3D7_1462400.1","PF3D7_1218200.1","PF3D7_1237500.1","PF3D7_1214500.1","PF3D7_0403700.1","PF3D7_0915800.1","PF3D7_1409300.1","PF3D7_1209300.1","PF3D7_0103600.1","PF3D7_1222800.1","PF3D7_0719100.1","PF3D7_0802200.1","PF3D7_1031200.1","PF3D7_1127100.1","PF3D7_1038100.1"],[845159,188001,53033.5,21628,32972.5,15815,15089,14721,19152.5,13284.5,8079.5,8910,6763,24077.5,7255.5,39552.5,8237.5,6121.5,5902,4958.5,5834.5,4920,4605.5,4797,4386.5,3835.5,3653.5,3252.5,3319,3514,3231.5,3145,2801,2754,2215,2407.5,2851.5,2599,2713,2656,2301.5,1832,2088,2531,977.5,2446.5,2238.5,943.5,2352,1899.5,679.5,2098,2075.5,2615,1950.5,252,1494,2099.5,1199.5,1265.5,937.5,2092,1964.5,1910,1609,1905,2198,1785.5,1064,1166,1649.5,1705.5,1936,731,1919.5,1541.5,1299,1325.5,1780.5,729,1322.5,1704,805,918.5,1646.5,976.5,1439.5,1720.5,925.5,1155,1017.5,1078.5,1593,1407,1184.5,1346.5,1338.5,1611,1326,1596.5,753,1586,817,1225.5,1289.5,678.5,1124.5,1298.5,1046.5,1307,1386,1218,1366,823,1300.5,1386.5,717.5,1021.5,1363,1032.5,1113,1069,1253,1155.5,928,1087,1280,1035.5,1096.5,1203,1336,952.5,956,1026.5,985,891,1101,1089,1288,1078,1145.5,1173,1181,1298,1232.5,833.5,956.5,1066.5,806,468.5,528,1091,982,589,810,835.5,936.5,1085.5,969.5,1035,1048.5,727,1037.5,956.5,693,904,793,525,425,1087,402,688.5,629.5,855.5,1049,764.5,1118,1179,429.5,1161.5,1065.5,750,985.5,814.5,762.5,1242.5,982,259,324.5,864,820.5,819.5,799.5,468,954,972,880.5,845,900.5,872,632.5,834,554,1023.5,760,867,669.5,871,309,659.5,470,599,889,437,848.5,887,943,915,409.5,727.5,504,838.5,798,449,736,673.5,676.5,848,618.5,980,908.5,736.5,631,824.5,737.5,911.5,319,822.5,307.5,817,689.5,1008.5,880,789,524,515,301.5,659.5,840.5,747,364,619,807,566,710,509,829,872.5,680,869.5,1071,798,348.5,424.5,820,838,642,447.5,605,545.5,786,643,454,632.5,668.5,619.5,324.5,749.5,406,560,791,563,835.5,626,835.5,657,439.5,636.5,725.5,574.5,744,799.5,765.5,831.5,231.5,684.5,543,139,444.5,775.5,603.5,700,403.5,327.5,823.5,665.5,786.5,615.5,484,699,381,299.5,452,127,534.5,280,632,493.5,452,486.5,418,324,635.5,310,463.5,754,634.5,730.5,750.5,431.5,518,548.5,651.5,676,624.5,124,739,569.5,290.5,481,731.5,306.5,651.5,421,698,624.5,580,809,600,508.5,649,579,586.5,625.5,511.5,156.5,674.5,213,567.5,362.5,227.5,495.5,537,606.5,560,229.5,646.5,773.5,701,628.5,606,338,828,672.5,638.5,432,317,411.5,488.5,731,666,474,542,656,266,486,662.5,596.5,416.5,640.5,395,398.5,674,737,669.5,576.5,514,685.5,486.5,567.5,697,622,486.5,843.5,522,640,555,326.5,698,539.5,594.5,561,509,481.5,540.5,653,544.5,519.5,318,431.5,374.5,341.5,304,379.5,449,610,345.5,421.5,658,474.5,508,799.5,438.5,553.5,452.5,564,605.5,443,624.5,629.5,522,430.5,663.5,635.5,377,430,621,552,471.5,700.5,223.5,503.5,194.5,657,692.5,388.5,530,484,469,572.5,444.5,442.5,494.5,537.5,532.5,624,777,510.5,492.5,575,360.5,344.5,590,566.5,477.5,532,496.5,498.5,296,328.5,516.5,461.5,328.5,360,452.5,359.5,543.5,270,544,450.5,335,463.5,465.5,461.5,191,620.5,305.5,365,535.5,535.5,487,423.5,344.5,457,461.5,320,470,438,450.5,448,398.5,433.5,572,442,466.5,388.5,345,286.5,359.5,497.5,288,519.5,390,551,204,273,435,356.5,607.5,472,487,304.5,342.5,367,191,510.5,422.5,400,492,296,413.5,174.5,519.5,469,473.5,380.5,353.5,410,344,422,451.5,429.5,397,396,367.5,448.5,376.5,166.5,456.5,652,530.5,302,402.5,253,180.5,676,309.5,482,473,391,196,625,170,436.5,451,435,508,362.5,505.5,407.5,253.5,467.5,240.5,625.5,281,343.5,681.5,270.5,283,320,360,450,379.5,241,408,336,379,459,421.5,384,297.5,182.5,386,315.5,457,239,669,343,502.5,501.5,387.5,284.5,226,310,319,636.5,376.5,402,356.5,315,246,225,202,225,210.5,350.5,315,332.5,389.5,463.5,617.5,332.5,370,475,299,343.5,204.5,675.5,241.5,380.5,411.5,444.5,567,277.5,191,278,290.5,307.5,361,286,187,454.5,272,444,165,255.5,500.5,350,416,309,297,180,231.5,410,319.5,310,192.5,298,224.5,424,176.5,233,356,277.5,358.5,221.5,359,333,122,444.5,99,407,327.5,148.5,232,331.5,768.5,289.5,290,344,278.5,386.5,396.5,256,157,355,286.5,342,411,386.5,242.5,285.5,173.5,159.5,290.5,316,253,192,191,285,252,271.5,477,338,310,366,66,267.5,177.5,339,206.5,284.5,333.5,267,250.5,355,111.5,363,497.5,267.5,264.5,241,240.5,250,291,306,387,378,254.5,434,320,272,258,246.5,427,426,132,216,508,116.5,344.5,301.5,327,297.5,242.5,135.5,104,116,327,268,282.5,464,135.5,263,74,229.5,266,367.5,438.5,256,303,497.5,355.5,294,395,352,194.5,362.5,224,248.5,454,587,272.5,139,162,437,167.5,417.5,318.5,174,416,573,458,159,142,294,403.5,294.5,248.5,255.5,258.5,202.5,354,236,347.5,387.5,332.5,208.5,404.5,689,297.5,345.5,219,333,94,206.5,463.5,128.5,261,252.5,173,97,144,171,130.5,88,271.5,66,290,245,278.5,165.5,61,256.5,305,190.5,301,209,205.5,273.5,192,257.5,227,261,235,365,260.5,366.5,204,360,282.5,222.5,406.5,151.5,396.5,267,261.5,68,498,517.5,218.5,405,37,185,178,205,360,124.5,347,170,135.5,329,219,309.5,165,244,221,399,195.5,166,249,218.5,274,280.5,189.5,182.5,168.5,207.5,107,396,154,227,164.5,216.5,199.5,371,252.5,307,232.5,201.5,275,219,263.5,261,383,333,107.5,154.5,79.5,389.5,449,221,344,182.5,388.5,287.5,311.5,140,357.5,373,274,207,454.5,503,514.5,151,248.5,247,256.5,204,169.5,159,279,156.5,224.5,128,111,181.5,209.5,225,157,162.5,485.5,128.5,242.5,22.5,50.5,207,173.5,118,298,59.5,195,134.5,287.5,70,250.5,67,246,200,230.5,88.5,389,226.5,194.5,340.5,272,198.5,227.5,126.5,28,241,318,317,342,270.5,375,157.5,175.5,282,352.5,170.5,311,338.5,249,214.5,220.5,300.5,179,295.5,249,302,198.5,247.5,254.5,429,344,355,223.5,190.5,196,182,137,137,152.5,228.5,201.5,133.5,387,174.5,95.5,232.5,236,118.5,463.5,155,147,442,306,217,103.5,95.5,98,141,150,286.5,273,249,296.5,192.5,351.5,210,126,185.5,269.5,236,307,165,273.5,316,303.5,266,108.5,293,236,225.5,59.5,168,145,251.5,110.5,143,29,290,318,347.5,91,155,403.5,307.5,102,231,290.5,120.5,172,216,86.5,177,253,139.5,18,260.5,165.5,254.5,297.5,164.5,218.5,136,360.5,332,169,267,294,206,208.5,127,74,226,329.5,278,98,222.5,321.5,312,194,234.5,279.5,300,23,297.5,130.5,303.5,170,111.5,248.5,140,256,288.5,284,319.5,234.5,54.5,74.5,468.5,57.5,246,421.5,376.5,305,117.5,313.5,249,331.5,64,228,358.5,147,371,109,348.5,331.5,231,196.5,80.5,229.5,154.5,215.5,260,326.5,180.5,327,233.5,259,215,284.5,149,277.5,156.5,158,233.5,335.5,180,184.5,274,248.5,50,107,403,170,225.5,226,109,129,198,353,255.5,401,244,41,306,169.5,109,149.5,39,122,320.5,425,167.5,374,351.5,33,109,115,169,68,75.5,152.5,166,149,287.5,254.5,34.5,298,212.5,267,70.5,219,121.5,230.5,555,287,53,132.5,355,152,280,249],[233,225,223,222,221,219,219,218,216,213,202,200,199,185,171,170,161,156,153,146,145,139,136,135,132,115,111,104,104,103,98,90,88,87,82,82,80,80,80,74,72,71,70,69,68,67,67,66,66,65,62,61,60,59,58,57,54,53,52,51,49,49,49,49,48,48,48,47,47,46,46,46,46,46,46,45,45,45,44,43,43,43,43,42,42,41,41,40,40,40,40,40,40,40,39,39,37,36,36,36,36,36,35,35,35,35,34,34,34,33,33,33,32,32,32,32,31,31,30,30,30,30,30,30,30,29,29,29,29,29,28,28,28,28,28,28,27,27,27,26,26,26,26,26,26,25,25,25,25,25,25,25,24,24,24,24,24,24,24,24,23,23,23,23,23,22,22,22,22,22,21,21,21,21,21,21,21,20,20,20,20,20,20,20,20,19,19,19,19,19,19,19,19,19,18,18,18,18,18,18,17,17,17,17,17,17,17,17,17,17,17,16,16,16,16,16,16,16,16,16,16,15,15,15,15,15,15,15,15,15,15,14,14,14,14,14,14,14,14,14,14,14,14,14,14,14,14,14,14,14,13,13,13,13,13,13,13,13,13,13,13,13,13,13,12,12,12,12,12,12,12,12,12,12,12,12,12,12,12,12,12,12,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],[0.9708333333333333,0.9375,0.9291666666666667,0.925,0.9208333333333333,0.9125,0.9125,0.9083333333333333,0.9,0.8875,0.8416666666666667,0.8333333333333334,0.8291666666666667,0.7708333333333334,0.7125,0.7083333333333334,0.6708333333333333,0.65,0.6375,0.6083333333333333,0.6041666666666666,0.5791666666666667,0.5666666666666667,0.5625,0.55,0.4791666666666667,0.4625,0.4333333333333333,0.4333333333333333,0.4291666666666666,0.4083333333333333,0.375,0.3666666666666666,0.3625,0.3416666666666667,0.3416666666666667,0.3333333333333333,0.3333333333333333,0.3333333333333333,0.3083333333333333,0.3,0.2958333333333333,0.2916666666666667,0.2875,0.2833333333333333,0.2791666666666667,0.2791666666666667,0.275,0.275,0.2708333333333333,0.2583333333333334,0.2541666666666667,0.25,0.2458333333333333,0.2416666666666667,0.2375,0.225,0.2208333333333333,0.2166666666666667,0.2125,0.2041666666666667,0.2041666666666667,0.2041666666666667,0.2041666666666667,0.2,0.2,0.2,0.1958333333333333,0.1958333333333333,0.1916666666666667,0.1916666666666667,0.1916666666666667,0.1916666666666667,0.1916666666666667,0.1916666666666667,0.1875,0.1875,0.1875,0.1833333333333333,0.1791666666666667,0.1791666666666667,0.1791666666666667,0.1791666666666667,0.175,0.175,0.1708333333333333,0.1708333333333333,0.1666666666666667,0.1666666666666667,0.1666666666666667,0.1666666666666667,0.1666666666666667,0.1666666666666667,0.1666666666666667,0.1625,0.1625,0.1541666666666667,0.15,0.15,0.15,0.15,0.15,0.1458333333333333,0.1458333333333333,0.1458333333333333,0.1458333333333333,0.1416666666666667,0.1416666666666667,0.1416666666666667,0.1375,0.1375,0.1375,0.1333333333333333,0.1333333333333333,0.1333333333333333,0.1333333333333333,0.1291666666666667,0.1291666666666667,0.125,0.125,0.125,0.125,0.125,0.125,0.125,0.1208333333333333,0.1208333333333333,0.1208333333333333,0.1208333333333333,0.1208333333333333,0.1166666666666667,0.1166666666666667,0.1166666666666667,0.1166666666666667,0.1166666666666667,0.1166666666666667,0.1125,0.1125,0.1125,0.1083333333333333,0.1083333333333333,0.1083333333333333,0.1083333333333333,0.1083333333333333,0.1083333333333333,0.1041666666666667,0.1041666666666667,0.1041666666666667,0.1041666666666667,0.1041666666666667,0.1041666666666667,0.1041666666666667,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.09583333333333334,0.09583333333333334,0.09583333333333334,0.09583333333333334,0.09583333333333334,0.09166666666666666,0.09166666666666666,0.09166666666666666,0.09166666666666666,0.09166666666666666,0.08749999999999999,0.08749999999999999,0.08749999999999999,0.08749999999999999,0.08749999999999999,0.08749999999999999,0.08749999999999999,0.08333333333333333,0.08333333333333333,0.08333333333333333,0.08333333333333333,0.08333333333333333,0.08333333333333333,0.08333333333333333,0.08333333333333333,0.07916666666666666,0.07916666666666666,0.07916666666666666,0.07916666666666666,0.07916666666666666,0.07916666666666666,0.07916666666666666,0.07916666666666666,0.07916666666666666,0.075,0.075,0.075,0.075,0.075,0.075,0.07083333333333333,0.07083333333333333,0.07083333333333333,0.07083333333333333,0.07083333333333333,0.07083333333333333,0.07083333333333333,0.07083333333333333,0.07083333333333333,0.07083333333333333,0.07083333333333333,0.06666666666666667,0.06666666666666667,0.06666666666666667,0.06666666666666667,0.06666666666666667,0.06666666666666667,0.06666666666666667,0.06666666666666667,0.06666666666666667,0.06666666666666667,0.0625,0.0625,0.0625,0.0625,0.0625,0.0625,0.0625,0.0625,0.0625,0.0625,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05833333333333333,0.05416666666666667,0.05416666666666667,0.05416666666666667,0.05416666666666667,0.05416666666666667,0.05416666666666667,0.05416666666666667,0.05416666666666667,0.05416666666666667,0.05416666666666667,0.05416666666666667,0.05416666666666667,0.05416666666666667,0.05416666666666667,0.05,0.05,0.05,0.05,0.05,0.05,0.05,0.05,0.05,0.05,0.05,0.05,0.05,0.05,0.05,0.05,0.05,0.05,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04583333333333333,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.04166666666666666,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.0375,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.03333333333333333,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.02916666666666667,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.025,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.02083333333333333,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.01666666666666667,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.0125,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.008333333333333333,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667,0.004166666666666667],[536,569,388,416,458,370.5,284,316,363,400,380.5,325.5,423,410,356,322.5,318,331.5,299.5,354.5,354,342.5,179.5,299.5,387.5,307.5,209,303,268.5,287,235.5,227.5,201,254,215.5,270,334.5,247.5,233,265,249.5,214.5,155.5,243,121.5,110,265,143.5,295,206.5,147,188,264.5,180,235.5,72,265,265,186,149,158,166,268.5,243,221.5,244.5,175.5,237,142,159.5,227,255,108.5,84,252,163,211.5,217.5,247,94.5,106,243.5,147.5,91,245,185.5,185,233.5,107,198,193.5,189.5,237.5,226.5,148.5,166.5,218,232,224.5,167.5,151,199,137.5,215.5,171.5,162,199.5,213.5,195,201.5,215,222.5,230,132.5,221.5,167,125.5,202,154,201.5,168.5,191.5,185.5,178.5,135.5,194.5,199,185.5,191,218,167,136,173.5,208.5,189.5,140.5,129.5,207.5,226.5,203,202.5,190.5,177.5,175,182.5,176,83,200,149,100,95,187.5,151.5,102.5,173,187.5,172.5,204,155,174,177.5,149,169,159.5,67.5,135.5,107,118,65.5,189,62.5,172,130.5,198.5,183,158,219,105,67,151.5,204,143.5,184.5,159,79,66.5,179,86,55,188,183.5,183.5,151.5,103.5,173,188,141.5,155,161,127,134,145.5,103.5,192,139,171,94,170.5,95,98.5,79,144,140.5,91,175,178.5,186,192,119.5,164,75.5,184,108.5,137,132,76.5,140,185,133.5,148.5,189,136,142.5,173,83,165.5,88,139,61.5,149,146,194.5,184.5,149,118.5,127,85,60.5,173,161,103.5,138,106.5,106,173.5,136.5,121,127.5,112,148,150,174.5,114,100,94,157.5,124,88,130,94.5,168.5,166.5,122,143,147,73,96.5,175,83,146.5,159,117,142,125,165,94,43,130.5,154.5,132.5,174,157,135.5,162.5,55,144,125.5,53.5,122,175,123.5,119.5,113.5,94,172,75,113.5,130,100.5,153,105.5,50,117.5,44,129.5,61,150.5,119.5,90.5,117,98.5,75,116.5,35,132,141,86.5,155,166.5,111,138,142,85.5,125.5,129,38,143,117,74,126.5,147.5,96,153.5,98.5,154,115.5,131,145.5,127,120,102,110.5,117,110,139.5,58.5,118,65,120.5,101,63.5,92,121,124.5,146.5,81,106,138,158.5,46,78,85,129,170.5,153.5,112.5,101,109,111,107.5,138,142.5,149.5,158.5,62,102,147,137,115.5,132,106.5,87,146,171.5,166.5,164.5,56.5,149.5,126.5,120,163.5,145,121,231.5,137,100.5,107,91,119.5,98,131.5,98,125,126.5,129.5,131.5,88,94.5,84,123,102,93.5,86,107.5,100,130,93,92.5,129.5,98,121.5,141,101.5,102,110.5,128,154,112,154,116,130,122.5,134.5,132,97,114.5,109,103,115,70.5,46,93,73.5,137.5,159.5,96,143.5,101,114,120,118.5,86.5,123,119,134,120.5,126,106,98,108.5,102.5,69.5,101,121,115,138,117.5,128,95.5,108,107,109,74,96.5,121,86,131.5,56,135,131,72.5,142.5,127,110.5,65,94,77.5,114,115,122.5,133,113,76.5,129,102.5,73,109,103,109.5,110.5,105,109.5,92.5,121,54,87.5,97.5,86,89,113.5,80.5,130.5,103,152,57,59,124,89,121,125,72,108.5,82.5,77,58.5,111.5,122,90,123.5,74,93,51,144,108,101,57,87,18,54,107.5,119,107.5,116,113.5,90.5,137.5,87.5,58,34,91,98,90.5,93.5,51,61,195,72,106.5,132.5,66,36.5,155.5,60.5,106,61,120,94,90,101,99.5,55,127.5,85,84.5,79,63,209,63.5,68.5,73,54.5,102.5,98,79.5,91.5,87,103.5,122.5,100,115,66,45,97,90,135,53.5,128,108.5,124.5,131.5,83.5,52,80.5,80.5,78,144,80,82.5,92.5,69.5,40,57.5,73,55.5,58.5,71,72,111,115.5,88.5,128,51,91.5,140.5,66,71,53.5,144.5,68.5,112.5,49,114,126.5,74,43,87.5,87,87.5,92.5,75.5,45.5,115,62,69,51,82,104.5,77.5,113.5,40,79.5,49,45.5,94.5,58,92,48,63,39,66.5,48.5,41.5,93,71.5,117,65,108,83.5,35,131,19,96,57,35,56,80,87,93,60,72,27,98.5,111,86.5,31.5,103.5,91.5,90.5,89.5,61.5,22,59,30,57,59,68,47.5,59,54,38,71,74,70.5,74,88,88.5,20.5,32,42,89,30.5,92.5,99.5,76,64,85,22,63,105.5,60,65,48,45.5,69.5,94,78.5,99.5,108.5,66,88.5,91,91,76.5,80,88.5,96.5,48,45,94,21.5,81,93,85,79.5,53,26,28.5,29,88,77,70,73.5,46,41,28,20,88.5,87,114.5,90,91,132,84.5,66.5,107,82,51.5,113,59.5,67.5,105,88.5,58,31,38.5,115,31.5,127.5,85,44,102.5,91,100,39,25,72.5,116,100.5,71.5,50.5,81.5,71,85,69,90,69.5,51.5,41,102.5,120.5,88.5,99,70.5,102,37,69.5,87,42,75,60,51.5,37,40,44.5,35.5,25,62.5,8,87,76.5,83,43,14,75,80.5,59.5,89.5,42.5,51.5,88.5,46.5,77.5,72.5,77,66,94,77.5,73.5,65,92,80.5,80.5,108.5,47.5,110,89,66,25,142,100,75.5,94,14,56,52.5,39,71,38.5,102.5,50.5,28.5,95,54,61,42,52,59.5,84,59.5,58,65,74,83,58,58.5,65,40,60,36.5,100,30.5,63,39,74,74,94.5,71,105,64,68,86.5,81,83,76,100.5,98.5,18,54,22.5,79,115,70.5,118,42.5,93,89,77,38,86,53,84,74,97.5,111.5,87,37.5,70,84,68.5,66,60.5,50,88.5,43,46.5,20.5,42.5,46.5,57,69.5,52,50.5,102,53.5,56,9,12,52.5,49.5,46.5,79,11,68,45.5,75,23,80.5,22.5,58,57,67,35,115.5,68,49.5,100.5,62,64,71.5,47.5,7.5,67.5,96,96,85,77,52,51.5,52.5,90.5,54.5,54,99.5,54.5,67,67,76.5,96,49.5,68,60,70.5,67.5,58.5,77.5,78.5,87.5,91.5,80,67,58,61,38.5,18.5,53.5,55.5,55.5,44,116,57.5,45,60.5,64,41.5,83,50,53,120,75.5,69.5,29,40.5,30.5,44.5,56.5,75,70,77.5,82,57,74,55.5,38.5,49,74.5,66.5,84.5,40,92,59,95,74.5,40,95,87,66,24,50,47,63,39,54,8,94.5,96.5,81,33,44,96,89,30,71.5,92.5,41,49.5,78.5,21,46.5,62,38.5,5,70.5,41,73.5,79,69,60.5,50,98,111.5,43,69.5,55.5,52,71.5,24,25.5,40.5,59,79,30.5,61,113,106,72.5,77.5,94.5,83,9,81,49.5,82,44,32,90,35,85,91.5,91,88,77.5,13,23,124,23,87,116,115.5,110.5,28.5,78,66.5,86.5,22,59.5,67,54,92,38,102.5,89.5,64,64.5,17.5,65,35.5,74,86,104,63.5,92,67.5,79,66,88,48,89.5,57,50,77,109,57,43,100,54.5,24,39,88,52.5,76,70,33.5,41.5,68,86.5,79,93,64,9,79,41,41,33,13.5,20,100,105.5,58,116,53.5,10.5,31,43,63.5,13.5,32.5,46,57,47,84,85,14,89,61.5,80.5,26,76.5,44.5,71.5,115,90.5,27,47.5,82,43,52,79],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th rowspan=\"2\">gene<\/th>\n      <th colspan=\"3\">reads<\/th>\n      <th colspan=\"3\">UMIs<\/th>\n    <\/tr>\n    <tr>\n      <th>median count<\/th>\n      <th>N &gt; 4096<\/th>\n      <th>% &gt; 4096<\/th>\n      <th>median count<\/th>\n      <th>N &gt; 4096<\/th>\n      <th>% &gt; 4096<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 0, 3, \",\", \".\", null);\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 0, 3, \",\", \".\", null);\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 1, 3, \",\", \".\", null);\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 1, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[1,2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render"],"jsHooks":[]}</script>
</div>
<p>Figure <a href="#fig:scatterplot-and-ma-plot">8</a> plots the relationship between UMI counts and read counts for all genes and samples.
There are a few notable points about this figure:</p>
<ol type="1">
<li>The UMI counts and read counts are very strongly associated.</li>
<li>The log-fold change between the read counts and the UMI counts can be large, indicating the UMI counts are likely removing substantial PCR amplification noise.</li>
<li>The log-fold changes between the read counts and the UMI counts are larger on average for more highly expressed genes. The variation in the log-fold change is larger for the lowly expressed genes.</li>
</ol>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/as.vector.html'>as.vector</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span></span>
<span><span class='va'>y</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log2</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/as.vector.html'>as.vector</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span></span>
<span><span class='va'>m</span> <span class='op'>&lt;-</span> <span class='op'>(</span><span class='va'>x</span> <span class='op'>+</span> <span class='va'>y</span><span class='op'>)</span> <span class='op'>/</span> <span class='fl'>2</span></span>
<span><span class='va'>d</span> <span class='op'>&lt;-</span> <span class='va'>y</span> <span class='op'>-</span> <span class='va'>x</span></span>
<span><span class='va'>g</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='co'># NOTE: Only showing unique points for plotting speed.</span></span>
<span><span class='va'>i</span> <span class='op'>&lt;-</span> <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='st'>"."</span>, <span class='va'>y</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>X</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='va'>Y</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='va'>M</span> <span class='op'>&lt;-</span> <span class='op'>(</span><span class='va'>X</span> <span class='op'>+</span> <span class='va'>Y</span><span class='op'>)</span> <span class='op'>/</span> <span class='fl'>2</span></span>
<span><span class='va'>D</span> <span class='op'>&lt;-</span> <span class='va'>y</span> <span class='op'>-</span> <span class='va'>x</span></span>
<span><span class='va'>G</span> <span class='op'>&lt;-</span> <span class='va'>g</span></span>
<span><span class='co'># NOTE: Only showing unique points for plotting speed.</span></span>
<span><span class='va'>I</span> <span class='op'>&lt;-</span> <span class='op'>!</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/duplicated.html'>duplicated</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='st'>"."</span>, <span class='va'>y</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>2</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotWithHighlights.html'>plotWithHighlights</a></span><span class='op'>(</span></span>
<span>  x <span class='op'>=</span> <span class='va'>x</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>,</span>
<span>  y <span class='op'>=</span> <span class='va'>y</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>,</span>
<span>  status <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>ribo_set</span>, <span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>mito_set</span>, <span class='op'>-</span><span class='fl'>1</span>, <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  legend <span class='op'>=</span> <span class='cn'>FALSE</span>,</span>
<span>  bg.pch <span class='op'>=</span> <span class='st'>"."</span>,</span>
<span>  hl.pch <span class='op'>=</span> <span class='st'>"."</span>,</span>
<span>  hl.col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"orange"</span>, <span class='st'>"dodgerBlue"</span><span class='op'>)</span>,</span>
<span>  hl.cex <span class='op'>=</span> <span class='fl'>3</span>,</span>
<span>  sub <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"r = "</span>, <span class='fu'><a href='https://scales.r-lib.org/reference/number.html'>number</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/cor.html'>cor</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>y</span><span class='op'>)</span>, accuracy <span class='op'>=</span> <span class='fl'>0.001</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>log</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span><span class='op'>(</span><span class='va'>UMIs</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  ylab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>log</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span><span class='op'>(</span><span class='va'>reads</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>a <span class='op'>=</span> <span class='fl'>0</span>, b <span class='op'>=</span> <span class='fl'>1</span>, lty <span class='op'>=</span> <span class='fl'>2</span>, col <span class='op'>=</span> <span class='st'>"grey"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotWithHighlights.html'>plotWithHighlights</a></span><span class='op'>(</span></span>
<span>  x <span class='op'>=</span> <span class='va'>m</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span> <span class='op'>/</span> <span class='fl'>2</span>,</span>
<span>  y <span class='op'>=</span> <span class='va'>d</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>,</span>
<span>  pch <span class='op'>=</span> <span class='st'>"."</span>,</span>
<span>  status <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>ribo_set</span>, <span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>mito_set</span>, <span class='op'>-</span><span class='fl'>1</span>, <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  legend <span class='op'>=</span> <span class='cn'>FALSE</span>,</span>
<span>  bg.pch <span class='op'>=</span> <span class='st'>"."</span>,</span>
<span>  hl.pch <span class='op'>=</span> <span class='st'>"."</span>,</span>
<span>  hl.col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"orange"</span>, <span class='st'>"dodgerBlue"</span><span class='op'>)</span>,</span>
<span>  hl.cex <span class='op'>=</span> <span class='fl'>3</span>,</span>
<span>  xlab <span class='op'>=</span> <span class='st'>"Average log-(count+1)"</span>,</span>
<span>  ylab <span class='op'>=</span> <span class='st'>"logFC (reads vs. UMIs)"</span>,</span>
<span>  ylim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='op'>-</span><span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>abs</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>d</span>, <span class='va'>D</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>abs</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>d</span>, <span class='va'>D</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, lty <span class='op'>=</span> <span class='fl'>2</span>, col <span class='op'>=</span> <span class='st'>"grey"</span><span class='op'>)</span></span>
<span><span class='va'>l</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/lowess.html'>lowess</a></span><span class='op'>(</span><span class='va'>m</span>, <span class='va'>d</span>, f <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/lines.html'>lines</a></span><span class='op'>(</span><span class='va'>l</span>, col <span class='op'>=</span> <span class='st'>"red"</span>, lwd <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotWithHighlights.html'>plotWithHighlights</a></span><span class='op'>(</span></span>
<span>  x <span class='op'>=</span> <span class='va'>X</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span>,</span>
<span>  y <span class='op'>=</span> <span class='va'>Y</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span>,</span>
<span>  status <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>G</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>ribo_set</span>, <span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>G</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>mito_set</span>, <span class='op'>-</span><span class='fl'>1</span>, <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  legend <span class='op'>=</span> <span class='cn'>FALSE</span>,</span>
<span>  bg.pch <span class='op'>=</span> <span class='st'>"."</span>,</span>
<span>  hl.pch <span class='op'>=</span> <span class='st'>"."</span>,</span>
<span>  hl.col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"orange"</span>, <span class='st'>"dodgerBlue"</span><span class='op'>)</span>,</span>
<span>  hl.cex <span class='op'>=</span> <span class='fl'>3</span>,</span>
<span>  sub <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span></span>
<span>    <span class='st'>"r = "</span>, </span>
<span>    <span class='fu'><a href='https://scales.r-lib.org/reference/number.html'>number</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/cor.html'>cor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/as.vector.html'>as.vector</a></span><span class='op'>(</span><span class='va'>X</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/as.vector.html'>as.vector</a></span><span class='op'>(</span><span class='va'>Y</span><span class='op'>)</span><span class='op'>)</span>, accuracy <span class='op'>=</span> <span class='fl'>0.001</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='fu'>logCPM</span><span class='op'>(</span><span class='va'>UMIs</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  ylab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='fu'>logCPM</span><span class='op'>(</span><span class='va'>reads</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>a <span class='op'>=</span> <span class='fl'>0</span>, b <span class='op'>=</span> <span class='fl'>1</span>, lty <span class='op'>=</span> <span class='fl'>2</span>, col <span class='op'>=</span> <span class='st'>"grey"</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/limma/man/plotWithHighlights.html'>plotWithHighlights</a></span><span class='op'>(</span></span>
<span>  x <span class='op'>=</span> <span class='va'>M</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span>,</span>
<span>  y <span class='op'>=</span> <span class='va'>D</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>,</span>
<span>  status <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>G</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>ribo_set</span>, <span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>G</span><span class='op'>[</span><span class='va'>I</span><span class='op'>]</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>mito_set</span>, <span class='op'>-</span><span class='fl'>1</span>, <span class='fl'>0</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  legend <span class='op'>=</span> <span class='cn'>FALSE</span>,</span>
<span>  bg.pch <span class='op'>=</span> <span class='st'>"."</span>,</span>
<span>  hl.pch <span class='op'>=</span> <span class='st'>"."</span>,</span>
<span>  hl.col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"orange"</span>, <span class='st'>"dodgerBlue"</span><span class='op'>)</span>,</span>
<span>  hl.cex <span class='op'>=</span> <span class='fl'>3</span>,</span>
<span>  xlab <span class='op'>=</span> <span class='st'>"Average logCPM"</span>,</span>
<span>  ylab <span class='op'>=</span> <span class='st'>"logFC (reads vs. UMIs)"</span>,</span>
<span>  ylim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='op'>-</span><span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>abs</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>d</span>, <span class='va'>D</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>abs</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>d</span>, <span class='va'>D</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>h <span class='op'>=</span> <span class='fl'>0</span>, lty <span class='op'>=</span> <span class='fl'>2</span>, col <span class='op'>=</span> <span class='st'>"grey"</span><span class='op'>)</span></span>
<span><span class='va'>l</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/stats/lowess.html'>lowess</a></span><span class='op'>(</span><span class='va'>M</span>, <span class='va'>D</span>, f <span class='op'>=</span> <span class='fl'>0.3</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/lines.html'>lines</a></span><span class='op'>(</span><span class='va'>l</span>, col <span class='op'>=</span> <span class='st'>"red"</span>, lwd <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:scatterplot-and-ma-plot"></span>
<img src="G000396_Danu.preprocess_files/scatterplot-and-ma-plot-1.png" alt="Scatterplot of the logarithm of the UMI and read counts (top)   and logCPM of the UMI and read counts (bottom) for all genes and samples   (left) and mean-difference plot of the same data (right). The correlation   in the scatterplot is reported below the x-axis label. Genes that are   ribosomal protein genes subunits are highlighted in orange and   mitochondrial genes in blue. The red line is a trend fitted to the   mean-differences." width="624" />
<p class="caption">
Figure 8: Scatterplot of the logarithm of the UMI and read counts (top) and logCPM of the UMI and read counts (bottom) for all genes and samples (left) and mean-difference plot of the same data (right). The correlation in the scatterplot is reported below the x-axis label. Genes that are ribosomal protein genes subunits are highlighted in orange and mitochondrial genes in blue. The red line is a trend fitted to the mean-differences.
</p>
</div>
</div>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>X</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"UMI_counts"</span><span class='op'>)</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='va'>Y</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='va'>cor_j</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/diag.html'>diag</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/cor.html'>cor</a></span><span class='op'>(</span><span class='va'>X</span>, <span class='va'>Y</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='co'># NOTE: Splitting up the cor() computations is a hack to improve speed whilst </span></span>
<span><span class='co'>#       keeping memory usage somewhat under control.</span></span>
<span><span class='va'>b</span> <span class='op'>&lt;-</span> <span class='fl'>1000</span></span>
<span><span class='va'>s</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>1</span>, by <span class='op'>=</span> <span class='va'>b</span>, to <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>e</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/Extremes.html'>pmin</a></span><span class='op'>(</span><span class='va'>s</span> <span class='op'>+</span> <span class='va'>b</span> <span class='op'>-</span> <span class='fl'>1</span>, <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/nrow.html'>nrow</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>)</span></span>
<span><span class='va'>cor_i</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/unlist.html'>unlist</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/funprog.html'>Map</a></span><span class='op'>(</span></span>
<span>    f <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>ss</span>, <span class='va'>ee</span> <span class='op'>)</span> <span class='fu'><a href='https://rdrr.io/r/base/diag.html'>diag</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/cor.html'>cor</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/t.html'>t</a></span><span class='op'>(</span><span class='va'>X</span><span class='op'>[</span><span class='va'>ss</span><span class='op'>:</span><span class='va'>ee</span>, <span class='op'>]</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/t.html'>t</a></span><span class='op'>(</span><span class='va'>Y</span><span class='op'>[</span><span class='va'>ss</span><span class='op'>:</span><span class='va'>ee</span>, <span class='op'>]</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>    ss <span class='op'>=</span> <span class='va'>s</span>,</span>
<span>    ee <span class='op'>=</span> <span class='va'>e</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>Rather than looking at all samples and all genes at once, we can also examine the relationship between UMI counts and read counts within a single sample across genes (Figure <a href="#fig:within-sample-cors">9</a>) and within a single gene across samples (Figure <a href="#fig:within-gene-cors">10</a>).
These figures show that the UMI counts and read counts are highly correlated across both samples (median = 0.980, median absolute deviation = 0.002) and genes (median = 0.940, median absolute deviation = 0.026).</p>
<div class="panelset">
<section id="within-sample" class="panel">
<h3>Within-sample</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>3</span>, <span class='fl'>3</span><span class='op'>)</span>, cex <span class='op'>=</span> <span class='fl'>0.5</span><span class='op'>)</span></span>
<span><span class='va'>jj</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/sort.html'>sort</a></span><span class='op'>(</span><span class='va'>cor_j</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>3</span>, <span class='fl'>119</span><span class='op'>:</span><span class='fl'>120</span>, <span class='fl'>238</span><span class='op'>:</span><span class='fl'>240</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='kw'>for</span> <span class='op'>(</span><span class='va'>j</span> <span class='kw'>in</span> <span class='va'>jj</span><span class='op'>)</span> <span class='op'>{</span></span>
<span>  <span class='va'>g</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span></span>
<span>    x <span class='op'>=</span> <span class='va'>X</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span>,</span>
<span>    y <span class='op'>=</span> <span class='va'>Y</span><span class='op'>[</span>, <span class='va'>j</span><span class='op'>]</span>,</span>
<span>    pch <span class='op'>=</span> <span class='st'>"."</span>,</span>
<span>    cex <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>ribo_set</span>, <span class='va'>mito_set</span><span class='op'>)</span>, <span class='fl'>5</span>, <span class='fl'>1</span><span class='op'>)</span>,</span>
<span>    col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span></span>
<span>      <span class='va'>g</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>ribo_set</span>, <span class='st'>"orange"</span>,</span>
<span>      <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>mito_set</span>, <span class='st'>"dodgerBlue"</span>, <span class='st'>"black"</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>    sub <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"r = "</span>, <span class='fu'><a href='https://scales.r-lib.org/reference/number.html'>number</a></span><span class='op'>(</span><span class='va'>cor_j</span><span class='op'>[</span><span class='va'>j</span><span class='op'>]</span>, accuracy <span class='op'>=</span> <span class='fl'>0.001</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>    xlab <span class='op'>=</span> <span class='st'>"logCPM(UMIs)"</span>,</span>
<span>    ylab <span class='op'>=</span> <span class='st'>"logCPM(reads)"</span>,</span>
<span>    main <span class='op'>=</span> <span class='va'>j</span>,</span>
<span>    xlim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>,</span>
<span>    ylim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span><span class='op'>)</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>a <span class='op'>=</span> <span class='fl'>0</span>, b <span class='op'>=</span> <span class='fl'>1</span>, col <span class='op'>=</span> <span class='st'>"grey"</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>cor_j</span>, na.rm <span class='op'>=</span> <span class='cn'>TRUE</span>, from <span class='op'>=</span> <span class='op'>-</span><span class='fl'>1</span>, to <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>,</span>
<span>  main <span class='op'>=</span> <span class='st'>"All samples"</span>, </span>
<span>  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>r</span><span class='op'>[</span><span class='va'>p</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:within-sample-cors"></span>
<img src="G000396_Danu.preprocess_files/within-sample-cors-1.png" alt="Association between UMI counts and read counts for each gene   within each sample. The first 8 panels are scatter plots of the UMI   counts and read counts for each gene for the samples with the 3 lowest   correlations, 2 middlemost correlations, and 3 highest correlations. The   correlation in the scatterplot is reported below each panel. Genes that   are ribosomal protein genes subunits are highlighted in orange and   mitochondrial genes in blue. The final panel is the distribution of these   correlations across all samples." width="624" />
<p class="caption">
Figure 9: Association between UMI counts and read counts for each gene within each sample. The first 8 panels are scatter plots of the UMI counts and read counts for each gene for the samples with the 3 lowest correlations, 2 middlemost correlations, and 3 highest correlations. The correlation in the scatterplot is reported below each panel. Genes that are ribosomal protein genes subunits are highlighted in orange and mitochondrial genes in blue. The final panel is the distribution of these correlations across all samples.
</p>
</div>
</div>
</section>
<section id="within-gene" class="panel">
<h3>Within-gene</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span><span class='op'>(</span>mfrow <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>3</span>, <span class='fl'>3</span><span class='op'>)</span>, cex <span class='op'>=</span> <span class='fl'>0.5</span><span class='op'>)</span></span>
<span><span class='va'>ii</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/sort.html'>sort</a></span><span class='op'>(</span><span class='va'>cor_i</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>3</span>, <span class='fl'>2861</span><span class='op'>:</span><span class='fl'>2862</span>, <span class='fl'>5722</span><span class='op'>:</span><span class='fl'>5724</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='va'>ii</span><span class='op'>)</span> <span class='op'>{</span></span>
<span>  <span class='va'>g</span> <span class='op'>&lt;-</span> <span class='va'>i</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span></span>
<span>    x <span class='op'>=</span> <span class='va'>X</span><span class='op'>[</span><span class='va'>i</span>, <span class='op'>]</span>,</span>
<span>    y <span class='op'>=</span> <span class='va'>Y</span><span class='op'>[</span><span class='va'>i</span>, <span class='op'>]</span>,</span>
<span>    pch <span class='op'>=</span> <span class='fl'>16</span>,</span>
<span>    col <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span></span>
<span>      <span class='va'>g</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>ribo_set</span>, <span class='st'>"orange"</span>,</span>
<span>      <span class='fu'><a href='https://rdrr.io/r/base/ifelse.html'>ifelse</a></span><span class='op'>(</span><span class='va'>g</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>mito_set</span>, <span class='st'>"dodgerBlue"</span>, <span class='st'>"black"</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>    sub <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"r = "</span>, <span class='fu'><a href='https://scales.r-lib.org/reference/number.html'>number</a></span><span class='op'>(</span><span class='va'>cor_i</span><span class='op'>[</span><span class='va'>i</span><span class='op'>]</span>, accuracy <span class='op'>=</span> <span class='fl'>0.001</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>    xlab <span class='op'>=</span> <span class='st'>"logCPM(UMIs)"</span>,</span>
<span>    ylab <span class='op'>=</span> <span class='st'>"logCPM(reads)"</span>,</span>
<span>    main <span class='op'>=</span> <span class='va'>i</span>,</span>
<span>    xlim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>X</span><span class='op'>)</span>,</span>
<span>    ylim <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/range.html'>range</a></span><span class='op'>(</span><span class='va'>Y</span><span class='op'>)</span><span class='op'>)</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span><span class='op'>(</span>a <span class='op'>=</span> <span class='fl'>0</span>, b <span class='op'>=</span> <span class='fl'>1</span>, col <span class='op'>=</span> <span class='st'>"grey"</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span></span>
<span><span class='op'>}</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/graphics/plot.default.html'>plot</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/stats/density.html'>density</a></span><span class='op'>(</span><span class='va'>cor_i</span>, na.rm <span class='op'>=</span> <span class='cn'>TRUE</span>, from <span class='op'>=</span> <span class='op'>-</span><span class='fl'>1</span>, to <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>,</span>
<span>  main <span class='op'>=</span> <span class='st'>"All genes"</span>, </span>
<span>  xlab <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span><span class='op'>(</span><span class='va'>r</span><span class='op'>[</span><span class='va'>p</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:within-gene-cors"></span>
<img src="G000396_Danu.preprocess_files/within-gene-cors-1.png" alt="Association between UMI counts and read counts for each sample   within each gene. The first 8 panels are scatter plots of the UMI counts   and read counts for each sample for the genes with the 3 lowest   correlations, 2 middlemost correlations, and 3 highest correlations. The   correlation in the scatterplot is reported below each panel. Genes that   are ribosomal protein genes subunits are highlighted in orange and   mitochondrial genes in blue. The final panel is the distribution of   these correlations across all genes." width="624" />
<p class="caption">
Figure 10: Association between UMI counts and read counts for each sample within each gene. The first 8 panels are scatter plots of the UMI counts and read counts for each sample for the genes with the 3 lowest correlations, 2 middlemost correlations, and 3 highest correlations. The correlation in the scatterplot is reported below each panel. Genes that are ribosomal protein genes subunits are highlighted in orange and mitochondrial genes in blue. The final panel is the distribution of these correlations across all genes.
</p>
</div>
</div>
</section>
</div>
<h2 id="summary-3">Summary</h2>
<p>The ultimate aim of this analysis is to perform a differential expression (DE) analysis.
On the one hand, a DE analysis is more powerful when there are more counts available, which is where the read counts are favoured.
On the other hand, a DE analysis may be biased if these counts arise through PCR duplicates, which is where the deduplicated UMI counts are favoured.
The choice between using the read counts or the UMI counts in the DE analysis is therefore a trade-off between these two choices.</p>
<p><strong>For now, we defer our choice of read counts vs.UMI counts, but the remainder of this preprocessing report uses (exonic) read counts</strong>.</p>
<h1 id="quality-control-of-cells">Quality control of cells</h1>
<h2 id="defining-the-quality-control-metrics">Defining the quality control metrics</h2>
<p>Low-quality cells need to be removed to ensure that technical effects do not distort downstream analysis results.
We use several quality control (QC) metrics to measure the quality of the cells:</p>
<ul>
<li><code>sum</code>: This measures the library size of the samples, which is the total sum of counts across genes. We want samples to have high library sizes as this means more RNA has been successfully captured during library preparation.</li>
<li><code>detected</code>: This is the number of expressed features<a href="#fn8" class="footnote-ref" id="fnref8" role="doc-noteref"><sup>8</sup></a> in each cell. Cells with few expressed features are likely to be of poor quality, as the diverse transcript population has not been successful captured.</li>
<li><code>altexps_ERCC_detected</code>: This measures the proportion of reads which are mapped to spike-in transcripts relative to the library size of each cell. High proportions are indicative of poor-quality cells, where endogenous RNA has been lost during processing (e.g., due to cell lysis or RNA degradation).
The same amount of spike-in RNA to each cell, so an enrichment in spike-in counts is symptomatic of loss of endogenous RNA.</li>
<li><code>subsets_Mito_percent</code>: This measures the proportion of reads which are mapped to mitochondrial RNA. If there is a higher than expected proportion of mitochondrial RNA this is often symptomatic of a cell which is under stress and is therefore of low quality and will not be used for the analysis.</li>
<li><code>subsets_Ribo_percent</code>: This measures the proportion of UMIs which are mapped to ribosomal protein genes. If there is a higher than expected proportion of ribosomal protein gene expression this is often symptomatic of a cell which is of compromised quality and we may want to exclude it from the analysis.</li>
</ul>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>is_mito</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>mito_set</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>is_mito</span><span class='op'>)</span></span>
<span><span class='va'>is_ribo</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>ribo_set</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>is_ribo</span><span class='op'>)</span></span>
<span><span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/addPerCellQCMetrics.html'>addPerCellQC</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>, </span>
<span>  subsets <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>Mito <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/which.html'>which</a></span><span class='op'>(</span><span class='va'>is_mito</span><span class='op'>)</span>, Ribo <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/which.html'>which</a></span><span class='op'>(</span><span class='va'>is_ribo</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  assay.type <span class='op'>=</span> <span class='st'>"read_counts"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<h2 id="visualizing-the-qc-metrics">Visualizing the QC metrics</h2>
<h3 id="individual-technical-replicates">Individual technical replicates</h3>
<p>Figures <a href="#fig:qcplot-by-sample-gex-sum">11</a> - <a href="#fig:qcplot-by-sample-gex-ribo">16</a> compare the QC metrics of the mini-bulk samples.
These figures show that the vast majority of mini-bulk samples are good-quality with:</p>
<ul>
<li>The median library size is around 2,365,000.</li>
<li>The median number of genes detected is around 5,100.</li>
<li>The median percentage of reads that are mapped to mitochondrial RNA is around 2%.</li>
</ul>
<p>Samples with a low number of reads also have a low number of detected genes, which is to be expected.
All samples have a low percentage of mitochondrial RNA, but there are a few replicates that are outliers.</p>
<p>The median percentage of reads that are mapped to ribosomal protein genes is quite high, at around 50%.
Such high percentages seems to be intrinsic to these samples.</p>
<div class="panelset">
<section id="sum" class="panel">
<h4>Sum</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p_sum</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>, </span>
<span>  y <span class='op'>=</span> <span class='st'>"sum"</span>,</span>
<span>  x <span class='op'>=</span> <span class='st'>"biological_replicate"</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  other_fields <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cell_line"</span>, <span class='st'>"timepoint"</span><span class='op'>)</span>,</span>
<span>  theme_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sum</span><span class='op'>)</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span> <span class='op'>~</span> <span class='va'>cell_line</span>, scales <span class='op'>=</span> <span class='st'>"free_x"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_continuous</a></span><span class='op'>(</span>labels <span class='op'>=</span> <span class='va'>comma</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p_sum</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"GEX: Library size"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:qcplot-by-sample-gex-sum"></span>
<img src="G000396_Danu.preprocess_files/qcplot-by-sample-gex-sum-1.png" alt="Distribution of a QC metric (given in plot title) for each   technical replicate in the dataset. Each point represents a technical   replicate and is coloured according to the legend." width="624" />
<p class="caption">
Figure 11: Distribution of a QC metric (given in plot title) for each technical replicate in the dataset. Each point represents a technical replicate and is coloured according to the legend.
</p>
</div>
</div>
</section>
<section id="sum-log-scale" class="panel">
<h4>Sum (log scale)</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p_sum_log</span> <span class='op'>&lt;-</span> <span class='va'>p_sum</span> <span class='op'>+</span> </span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='cn'>NA</span><span class='op'>)</span>, labels <span class='op'>=</span> <span class='va'>comma</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/annotation_logticks.html'>annotation_logticks</a></span><span class='op'>(</span></span>
<span>    sides <span class='op'>=</span> <span class='st'>"l"</span>,</span>
<span>    short <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.03</span>, <span class='st'>"cm"</span><span class='op'>)</span>,</span>
<span>    mid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.06</span>, <span class='st'>"cm"</span><span class='op'>)</span>,</span>
<span>    long <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.09</span>, <span class='st'>"cm"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p_sum_log</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"GEX: Library size (log scale)"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:qcplot-by-sample-gex-log-sum"></span>
<img src="G000396_Danu.preprocess_files/qcplot-by-sample-gex-log-sum-1.png" alt="Distribution of a QC metric (given in plot title) for each   technical replicate in the dataset. Each point represents a technical   replicate and is coloured according to the legend." width="624" />
<p class="caption">
Figure 12: Distribution of a QC metric (given in plot title) for each technical replicate in the dataset. Each point represents a technical replicate and is coloured according to the legend.
</p>
</div>
</div>
</section>
<section id="detected" class="panel">
<h4>Detected</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p_detected</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>, </span>
<span>  y <span class='op'>=</span> <span class='st'>"detected"</span>,</span>
<span>  x <span class='op'>=</span> <span class='st'>"biological_replicate"</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  other_fields <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cell_line"</span>, <span class='st'>"timepoint"</span><span class='op'>)</span>,</span>
<span>  theme_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>detected</span><span class='op'>)</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span> <span class='op'>~</span> <span class='va'>cell_line</span>, scales <span class='op'>=</span> <span class='st'>"free_x"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_continuous</a></span><span class='op'>(</span>labels <span class='op'>=</span> <span class='va'>comma</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p_detected</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"GEX: Number of genes detected"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:qcplot-by-sample-gex-detected"></span>
<img src="G000396_Danu.preprocess_files/qcplot-by-sample-gex-detected-1.png" alt="Distribution of a QC metric (given in plot title) for each   technical replicate in the dataset. Each point represents a technical   replicate and is coloured according to the legend." width="624" />
<p class="caption">
Figure 13: Distribution of a QC metric (given in plot title) for each technical replicate in the dataset. Each point represents a technical replicate and is coloured according to the legend.
</p>
</div>
</div>
</section>
<section id="detected-log-scale" class="panel">
<h4>Detected (log scale)</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p_detected_log</span> <span class='op'>&lt;-</span> <span class='va'>p_detected</span> <span class='op'>+</span> </span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='cn'>NA</span><span class='op'>)</span>, labels <span class='op'>=</span> <span class='va'>comma</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/annotation_logticks.html'>annotation_logticks</a></span><span class='op'>(</span></span>
<span>    sides <span class='op'>=</span> <span class='st'>"l"</span>,</span>
<span>    short <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.03</span>, <span class='st'>"cm"</span><span class='op'>)</span>,</span>
<span>    mid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.06</span>, <span class='st'>"cm"</span><span class='op'>)</span>,</span>
<span>    long <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.09</span>, <span class='st'>"cm"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p_detected_log</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"GEX: Number of genes detected (log scale)"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:qcplot-by-sample-gex-log-detected"></span>
<img src="G000396_Danu.preprocess_files/qcplot-by-sample-gex-log-detected-1.png" alt="Distribution of a QC metric (given in plot title) for each   technical replicate in the dataset. Each point represents a technical   replicate and is coloured according to the legend." width="624" />
<p class="caption">
Figure 14: Distribution of a QC metric (given in plot title) for each technical replicate in the dataset. Each point represents a technical replicate and is coloured according to the legend.
</p>
</div>
</div>
</section>
<section id="mitochondrial" class="panel">
<h4>Mitochondrial %</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p_mito</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>, </span>
<span>  y <span class='op'>=</span> <span class='st'>"subsets_Mito_percent"</span>,</span>
<span>  x <span class='op'>=</span> <span class='st'>"biological_replicate"</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  other_fields <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cell_line"</span>, <span class='st'>"timepoint"</span><span class='op'>)</span>,</span>
<span>  theme_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>subsets_Mito_percent</span><span class='op'>)</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span> <span class='op'>~</span> <span class='va'>cell_line</span>, scales <span class='op'>=</span> <span class='st'>"free_x"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_continuous</a></span><span class='op'>(</span></span>
<span>    labels <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'><a href='https://scales.r-lib.org/reference/percent_format.html'>percent</a></span><span class='op'>(</span><span class='va'>x</span>, scale <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>,</span>
<span>    limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>100</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p_mito</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"GEX: Percentage mitochondrial expression"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:qcplot-by-sample-gex-mito"></span>
<img src="G000396_Danu.preprocess_files/qcplot-by-sample-gex-mito-1.png" alt="Distribution of a QC metric (given in plot title) for each   technical replicate in the dataset. Each point represents a technical   replicate and is coloured according to the legend." width="624" />
<p class="caption">
Figure 15: Distribution of a QC metric (given in plot title) for each technical replicate in the dataset. Each point represents a technical replicate and is coloured according to the legend.
</p>
</div>
</div>
</section>
<section id="ribosomal" class="panel">
<h4>Ribosomal %</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p_ribo</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>, </span>
<span>  y <span class='op'>=</span> <span class='st'>"subsets_Ribo_percent"</span>,</span>
<span>  x <span class='op'>=</span> <span class='st'>"biological_replicate"</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  other_fields <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"cell_line"</span>, <span class='st'>"timepoint"</span><span class='op'>)</span>,</span>
<span>  theme_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>subsets_Ribo_percent</span><span class='op'>)</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span> <span class='op'>~</span> <span class='va'>cell_line</span>, scales <span class='op'>=</span> <span class='st'>"free_x"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_continuous</a></span><span class='op'>(</span></span>
<span>    labels <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'><a href='https://scales.r-lib.org/reference/percent_format.html'>percent</a></span><span class='op'>(</span><span class='va'>x</span>, scale <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>, </span>
<span>    limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>100</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p_ribo</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"GEX: Percentage ribosomal expression"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:qcplot-by-sample-gex-ribo"></span>
<img src="G000396_Danu.preprocess_files/qcplot-by-sample-gex-ribo-1.png" alt="Distribution of a QC metric (given in plot title) for each   technical replicate in the dataset. Each point represents a technical   replicate and is coloured according to the legend." width="624" />
<p class="caption">
Figure 16: Distribution of a QC metric (given in plot title) for each technical replicate in the dataset. Each point represents a technical replicate and is coloured according to the legend.
</p>
</div>
</div>
</section>
</div>
<h3 id="aggregated-technical-replicates">Aggregated technical replicates</h3>
<p>We will perform the differential expression analysis of the data by aggregating the technical duplicates by summing their counts.
Hence, we also look at at the QC metrics after this aggregation.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>sce_aggregated</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/aggregateAcrossCells.html'>aggregateAcrossCells</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>, </span>
<span>  <span class='va'>sce</span><span class='op'>$</span><span class='va'>sample</span>, </span>
<span>  use.assay.type <span class='op'>=</span> <span class='st'>"read_counts"</span><span class='op'>)</span></span>
<span><span class='va'>sce_aggregated</span><span class='op'>$</span><span class='va'>sum</span> <span class='op'>&lt;-</span> </span>
<span>  <span class='va'>sce_aggregated</span><span class='op'>$</span><span class='va'>detected</span> <span class='op'>&lt;-</span> </span>
<span>  <span class='va'>sce_aggregated</span><span class='op'>$</span><span class='va'>subsets_Mito_percent</span> <span class='op'>&lt;-</span></span>
<span>  <span class='va'>sce_aggregated</span><span class='op'>$</span><span class='va'>subsets_Ribo_percent</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span></span>
<span></span>
<span><span class='va'>is_mito</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce_aggregated</span><span class='op'>)</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>mito_set</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>is_mito</span><span class='op'>)</span></span>
<span><span class='va'>is_ribo</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce_aggregated</span><span class='op'>)</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='va'>ribo_set</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/summary.html'>summary</a></span><span class='op'>(</span><span class='va'>is_ribo</span><span class='op'>)</span></span>
<span><span class='va'>sce_aggregated</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/addPerCellQCMetrics.html'>addPerCellQC</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce_aggregated</span>, </span>
<span>  subsets <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>Mito <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/which.html'>which</a></span><span class='op'>(</span><span class='va'>is_mito</span><span class='op'>)</span>, Ribo <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/which.html'>which</a></span><span class='op'>(</span><span class='va'>is_ribo</span><span class='op'>)</span><span class='op'>)</span>,</span>
<span>  assay.type <span class='op'>=</span> <span class='st'>"read_counts"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>Figures <a href="#fig:qcplot-by-sample-gex-aggregated-sum">17</a> - <a href="#fig:qcplot-by-sample-gex-aggregated-ribo">22</a> compare the QC metrics of the mini-bulk samples.
These figures show that the vast majority of aggregatved mini-bulk samples are good-quality with:</p>
<ul>
<li>The median library size is around 5,175,000.</li>
<li>The median number of genes detected is around 5,300.</li>
<li>The median percentage of reads that are mapped to mitochondrial RNA is around 2%.</li>
</ul>
<p>Samples with a low number of reads also have a low number of detected genes, which is to be expected.
All samples have a low percentage of mitochondrial RNA, but there are a few replicates that are outliers.</p>
<p>The median percentage of reads that are mapped to ribosomal protein genes is quite high, at around 47%.
Such high percentages seems to be intrinsic to these samples.</p>
<div class="panelset">
<section id="sum-1" class="panel">
<h4>Sum</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p_sum</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce_aggregated</span>, </span>
<span>  y <span class='op'>=</span> <span class='st'>"sum"</span>,</span>
<span>  x <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  other_fields <span class='op'>=</span>  <span class='st'>"timepoint"</span>,</span>
<span>  theme_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sum</span><span class='op'>)</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span> <span class='op'>~</span> <span class='va'>.</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_continuous</a></span><span class='op'>(</span>labels <span class='op'>=</span> <span class='va'>comma</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p_sum</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"GEX: Library size"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:qcplot-by-sample-gex-aggregated-sum"></span>
<img src="G000396_Danu.preprocess_files/qcplot-by-sample-gex-aggregated-sum-1.png" alt="Distribution of a QC metric (given in plot title) for each   biological replicate in the dataset. Each point represents a biological   replicate and is coloured according to the legend." width="624" />
<p class="caption">
Figure 17: Distribution of a QC metric (given in plot title) for each biological replicate in the dataset. Each point represents a biological replicate and is coloured according to the legend.
</p>
</div>
</div>
</section>
<section id="sum-log-scale-1" class="panel">
<h4>Sum (log scale)</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p_sum_log</span> <span class='op'>&lt;-</span> <span class='va'>p_sum</span> <span class='op'>+</span> </span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='cn'>NA</span><span class='op'>)</span>, labels <span class='op'>=</span> <span class='va'>comma</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/annotation_logticks.html'>annotation_logticks</a></span><span class='op'>(</span></span>
<span>    sides <span class='op'>=</span> <span class='st'>"l"</span>,</span>
<span>    short <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.03</span>, <span class='st'>"cm"</span><span class='op'>)</span>,</span>
<span>    mid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.06</span>, <span class='st'>"cm"</span><span class='op'>)</span>,</span>
<span>    long <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.09</span>, <span class='st'>"cm"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p_sum_log</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"GEX: Library size (log scale)"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:qcplot-by-sample-gex-aggregated-log-sum"></span>
<img src="G000396_Danu.preprocess_files/qcplot-by-sample-gex-aggregated-log-sum-1.png" alt="Distribution of a QC metric (given in plot title) for each   biological replicate in the dataset. Each point represents a biological   replicate and is coloured according to the legend." width="624" />
<p class="caption">
Figure 18: Distribution of a QC metric (given in plot title) for each biological replicate in the dataset. Each point represents a biological replicate and is coloured according to the legend.
</p>
</div>
</div>
</section>
<section id="detected-1" class="panel">
<h4>Detected</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p_detected</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce_aggregated</span>, </span>
<span>  y <span class='op'>=</span> <span class='st'>"detected"</span>,</span>
<span>  x <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  other_fields <span class='op'>=</span>  <span class='st'>"timepoint"</span>,</span>
<span>  theme_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>detected</span><span class='op'>)</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span> <span class='op'>~</span> <span class='va'>.</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_continuous</a></span><span class='op'>(</span>labels <span class='op'>=</span> <span class='va'>comma</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p_detected</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"GEX: Number of genes detected"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:qcplot-by-sample-gex-aggregated-detected"></span>
<img src="G000396_Danu.preprocess_files/qcplot-by-sample-gex-aggregated-detected-1.png" alt="Distribution of a QC metric (given in plot title) for each   biological replicate in the dataset. Each point represents a biological   replicate and is coloured according to the legend." width="624" />
<p class="caption">
Figure 19: Distribution of a QC metric (given in plot title) for each biological replicate in the dataset. Each point represents a biological replicate and is coloured according to the legend.
</p>
</div>
</div>
</section>
<section id="detected-log-scale-1" class="panel">
<h4>Detected (log scale)</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p_detected_log</span> <span class='op'>&lt;-</span> <span class='va'>p_detected</span> <span class='op'>+</span> </span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_log10</a></span><span class='op'>(</span>limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='cn'>NA</span><span class='op'>)</span>, labels <span class='op'>=</span> <span class='va'>comma</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/annotation_logticks.html'>annotation_logticks</a></span><span class='op'>(</span></span>
<span>    sides <span class='op'>=</span> <span class='st'>"l"</span>,</span>
<span>    short <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.03</span>, <span class='st'>"cm"</span><span class='op'>)</span>,</span>
<span>    mid <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.06</span>, <span class='st'>"cm"</span><span class='op'>)</span>,</span>
<span>    long <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/grid/unit.html'>unit</a></span><span class='op'>(</span><span class='fl'>0.09</span>, <span class='st'>"cm"</span><span class='op'>)</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p_detected_log</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"GEX: Number of genes detected (log scale)"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:qcplot-by-sample-gex-aggregated-log-detected"></span>
<img src="G000396_Danu.preprocess_files/qcplot-by-sample-gex-aggregated-log-detected-1.png" alt="Distribution of a QC metric (given in plot title) for each   biological replicate in the dataset. Each point represents a biological   replicate and is coloured according to the legend." width="624" />
<p class="caption">
Figure 20: Distribution of a QC metric (given in plot title) for each biological replicate in the dataset. Each point represents a biological replicate and is coloured according to the legend.
</p>
</div>
</div>
</section>
<section id="mitochondrial-1" class="panel">
<h4>Mitochondrial %</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p_mito</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce_aggregated</span>, </span>
<span>  y <span class='op'>=</span> <span class='st'>"subsets_Mito_percent"</span>,</span>
<span>  x <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  other_fields <span class='op'>=</span>  <span class='st'>"timepoint"</span>,</span>
<span>  theme_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>subsets_Mito_percent</span><span class='op'>)</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span> <span class='op'>~</span> <span class='va'>.</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_continuous</a></span><span class='op'>(</span></span>
<span>    labels <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'><a href='https://scales.r-lib.org/reference/percent_format.html'>percent</a></span><span class='op'>(</span><span class='va'>x</span>, scale <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>,</span>
<span>    limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>100</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p_mito</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"GEX: Percentage mitochondrial expression"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:qcplot-by-sample-gex-aggregated-mito"></span>
<img src="G000396_Danu.preprocess_files/qcplot-by-sample-gex-aggregated-mito-1.png" alt="Distribution of a QC metric (given in plot title) for each   biological replicate in the dataset. Each point represents a biological   replicate and is coloured according to the legend." width="624" />
<p class="caption">
Figure 21: Distribution of a QC metric (given in plot title) for each biological replicate in the dataset. Each point represents a biological replicate and is coloured according to the legend.
</p>
</div>
</div>
</section>
<section id="ribosomal-1" class="panel">
<h4>Ribosomal %</h4>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>p_ribo</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotColData.html'>plotColData</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce_aggregated</span>, </span>
<span>  y <span class='op'>=</span> <span class='st'>"subsets_Ribo_percent"</span>,</span>
<span>  x <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='st'>"cell_line"</span>,</span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  other_fields <span class='op'>=</span>  <span class='st'>"timepoint"</span>,</span>
<span>  theme_size <span class='op'>=</span> <span class='fl'>8</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_abline.html'>geom_hline</a></span><span class='op'>(</span>yintercept <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>subsets_Ribo_percent</span><span class='op'>)</span>, lty <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span> <span class='op'>~</span> <span class='va'>.</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/scale_continuous.html'>scale_y_continuous</a></span><span class='op'>(</span></span>
<span>    labels <span class='op'>=</span> <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='fu'><a href='https://scales.r-lib.org/reference/percent_format.html'>percent</a></span><span class='op'>(</span><span class='va'>x</span>, scale <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>,</span>
<span>    limits <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>100</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span></span>
<span><span class='va'>p_ribo</span> <span class='op'>+</span> <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"GEX: Percentage ribosomal expression"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:qcplot-by-sample-gex-aggregated-ribo"></span>
<img src="G000396_Danu.preprocess_files/qcplot-by-sample-gex-aggregated-ribo-1.png" alt="Distribution of a QC metric (given in plot title) for each   biological replicate in the dataset. Each point represents a biological   replicate and is coloured according to the legend." width="624" />
<p class="caption">
Figure 22: Distribution of a QC metric (given in plot title) for each biological replicate in the dataset. Each point represents a biological replicate and is coloured according to the legend.
</p>
</div>
</div>
</section>
</div>
<h2 id="identifying-outliers-by-each-metric">Identifying outliers by each metric</h2>
<p>Outliers are defined based on the median absolute deviation (MADs) from the median value of each metric across all cells.
We identify small outliers for the library size and the number of expressed features, and large outliers for the mitochondrial and ribosomal gene expression proportions.
Identification of low-quality cells is then performed by combining the filters for all of the metrics.</p>
<p>Note that we do this at the level of the unaggregated technical replicates.</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>libsize_drop</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/isOutlier.html'>isOutlier</a></span><span class='op'>(</span></span>
<span>  metric <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>sum</span>, </span>
<span>  nmads <span class='op'>=</span> <span class='fl'>3</span>,</span>
<span>  type <span class='op'>=</span> <span class='st'>"lower"</span>, </span>
<span>  log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='va'>feature_drop</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/isOutlier.html'>isOutlier</a></span><span class='op'>(</span></span>
<span>  metric <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>detected</span>,</span>
<span>  nmads <span class='op'>=</span> <span class='fl'>3</span>, </span>
<span>  type <span class='op'>=</span> <span class='st'>"lower"</span>, </span>
<span>  log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='va'>mito_drop</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/isOutlier.html'>isOutlier</a></span><span class='op'>(</span></span>
<span>  metric <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>subsets_Mito_percent</span>, </span>
<span>  nmads <span class='op'>=</span> <span class='fl'>3</span>, </span>
<span>  type <span class='op'>=</span> <span class='st'>"higher"</span><span class='op'>)</span></span>
<span><span class='va'>ribo_drop</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/isOutlier.html'>isOutlier</a></span><span class='op'>(</span></span>
<span>  metric <span class='op'>=</span> <span class='va'>sce</span><span class='op'>$</span><span class='va'>subsets_Ribo_percent</span>, </span>
<span>  nmads <span class='op'>=</span> <span class='fl'>3</span>, </span>
<span>  type <span class='op'>=</span> <span class='st'>"higher"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>The following table summarises the QC cutoffs:</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>libsize_drop_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span>  cutoff <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/attributes.html'>attributes</a></span><span class='op'>(</span><span class='va'>libsize_drop</span><span class='op'>)</span><span class='op'>$</span><span class='va'>thresholds</span><span class='op'>[</span><span class='st'>"lower"</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='va'>feature_drop_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span>  cutoff <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/attributes.html'>attributes</a></span><span class='op'>(</span><span class='va'>feature_drop</span><span class='op'>)</span><span class='op'>$</span><span class='va'>thresholds</span><span class='op'>[</span><span class='st'>"lower"</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='va'>mito_drop_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span>  cutoff <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/attributes.html'>attributes</a></span><span class='op'>(</span><span class='va'>mito_drop</span><span class='op'>)</span><span class='op'>$</span><span class='va'>thresholds</span><span class='op'>[</span><span class='st'>"higher"</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='va'>ribo_drop_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span>  cutoff <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/attributes.html'>attributes</a></span><span class='op'>(</span><span class='va'>ribo_drop</span><span class='op'>)</span><span class='op'>$</span><span class='va'>thresholds</span><span class='op'>[</span><span class='st'>"higher"</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='va'>qc_cutoffs_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/cbind.html'>cbind</a></span><span class='op'>(</span></span>
<span>    <span class='va'>libsize_drop_df</span>,</span>
<span>    <span class='va'>feature_drop_df</span>, </span>
<span>    <span class='va'>mito_drop_df</span>, </span>
<span>    <span class='va'>ribo_drop_df</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>qc_cutoffs_df</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span></span>
<span>  <span class='st'>"total counts"</span>, <span class='st'>"total features"</span>, <span class='st'>"%mito"</span>, <span class='st'>"%ribo"</span><span class='op'>)</span></span>
<span><span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='va'>qc_cutoffs_df</span>, caption <span class='op'>=</span> <span class='st'>"QC cutoffs"</span>, digits <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<table>
<caption><span id="tab:unnamed-chunk-42">Table 6: </span>QC cutoffs</caption>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">total counts</th>
<th style="text-align: right;">total features</th>
<th style="text-align: right;">%mito</th>
<th style="text-align: right;">%ribo</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">lower</td>
<td style="text-align: right;">301300.1</td>
<td style="text-align: right;">4365.3</td>
<td style="text-align: right;">2.8</td>
<td style="text-align: right;">90.7</td>
</tr>
</tbody>
</table>
</div>
<p><strong>We would typically remove outliers (e.g., small outliers for the library size and the number of expressed features, large outliers for mitochondrial or ribosomal gene expression percentage) at this stage.</strong>
However, we opt to retain all replicates for the time being.
The following table highlights samples that would typically exclude as outliers by this criteria:</p>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='va'>keep</span> <span class='op'>&lt;-</span> <span class='op'>!</span><span class='op'>(</span><span class='va'>libsize_drop</span> <span class='op'>|</span> <span class='va'>feature_drop</span> <span class='op'>|</span> <span class='va'>mito_drop</span> <span class='op'>|</span> <span class='va'>ribo_drop</span><span class='op'>)</span></span>
<span><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/cbind.html'>cbind</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>sample <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span>, keep <span class='op'>=</span> <span class='va'>keep</span><span class='op'>)</span>,</span>
<span>  <span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span><span class='op'>[</span></span>
<span>    ,</span>
<span>    <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"sample"</span>, <span class='st'>"sum"</span>, <span class='st'>"detected"</span>, <span class='st'>"subsets_Mito_percent"</span>, <span class='st'>"subsets_Ribo_percent"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/row.names.html'>row.names</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span></span>
<span><span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/datatable.html'>datatable</a></span><span class='op'>(</span></span>
<span>  <span class='va'>x</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/order.html'>order</a></span><span class='op'>(</span><span class='va'>keep</span><span class='op'>)</span>, <span class='op'>]</span>,</span>
<span>  caption <span class='op'>=</span> <span class='st'>"Quality control metrics including library size (`sum`) and number of genes detected (`keep`) for each replicate. Each replicate is annotated by whether it would be kept or excluded by the typical quality control procedures (`keep`)."</span>,</span>
<span>  rownames <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>|&gt;</span></span>
<span>  <span class='fu'>DT</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DT/man/formatCurrency.html'>formatRound</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"subsets_Mito_percent"</span>, <span class='st'>"subsets_Ribo_percent"</span><span class='op'>)</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-e017a11de23f723e399b" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e017a11de23f723e399b">{"x":{"filter":"none","vertical":false,"caption":"<caption>Quality control metrics including library size (`sum`) and number of genes detected (`keep`) for each replicate. Each replicate is annotated by whether it would be kept or excluded by the typical quality control procedures (`keep`).<\/caption>","data":[["Danu_MB_1_B5","Danu_MB_1_C14","Danu_MB_1_D1","Danu_MB_1_D2","Danu_MB_1_D8","Danu_MB_1_E1","Danu_MB_1_E2","Danu_MB_1_E3","Danu_MB_1_E4","Danu_MB_1_E5","Danu_MB_1_F1","Danu_MB_1_F2","Danu_MB_1_F5","Danu_MB_1_G1","Danu_MB_1_G2","Danu_MB_1_G4","Danu_MB_1_J1","Danu_MB_1_K13","Danu_MB_1_L2","Danu_MB_1_L3","Danu_MB_1_L9","Danu_MB_1_M1","Danu_MB_1_N15","Danu_MB_1_O17","Danu_MB_1_B1","Danu_MB_1_B10","Danu_MB_1_B13","Danu_MB_1_B14","Danu_MB_1_B15","Danu_MB_1_B16","Danu_MB_1_B17","Danu_MB_1_B18","Danu_MB_1_B19","Danu_MB_1_B2","Danu_MB_1_B20","Danu_MB_1_B21","Danu_MB_1_B22","Danu_MB_1_B3","Danu_MB_1_B4","Danu_MB_1_B6","Danu_MB_1_B7","Danu_MB_1_B8","Danu_MB_1_B9","Danu_MB_1_C1","Danu_MB_1_C10","Danu_MB_1_C13","Danu_MB_1_C15","Danu_MB_1_C16","Danu_MB_1_C17","Danu_MB_1_C18","Danu_MB_1_C19","Danu_MB_1_C2","Danu_MB_1_C20","Danu_MB_1_C21","Danu_MB_1_C22","Danu_MB_1_C3","Danu_MB_1_C4","Danu_MB_1_C5","Danu_MB_1_C6","Danu_MB_1_C7","Danu_MB_1_C8","Danu_MB_1_C9","Danu_MB_1_D10","Danu_MB_1_D13","Danu_MB_1_D14","Danu_MB_1_D15","Danu_MB_1_D16","Danu_MB_1_D17","Danu_MB_1_D18","Danu_MB_1_D19","Danu_MB_1_D20","Danu_MB_1_D21","Danu_MB_1_D22","Danu_MB_1_D3","Danu_MB_1_D4","Danu_MB_1_D5","Danu_MB_1_D6","Danu_MB_1_D7","Danu_MB_1_D9","Danu_MB_1_E10","Danu_MB_1_E13","Danu_MB_1_E14","Danu_MB_1_E15","Danu_MB_1_E16","Danu_MB_1_E17","Danu_MB_1_E18","Danu_MB_1_E19","Danu_MB_1_E20","Danu_MB_1_E21","Danu_MB_1_E22","Danu_MB_1_E6","Danu_MB_1_E7","Danu_MB_1_E8","Danu_MB_1_E9","Danu_MB_1_F10","Danu_MB_1_F13","Danu_MB_1_F14","Danu_MB_1_F15","Danu_MB_1_F16","Danu_MB_1_F17","Danu_MB_1_F18","Danu_MB_1_F19","Danu_MB_1_F20","Danu_MB_1_F21","Danu_MB_1_F22","Danu_MB_1_F3","Danu_MB_1_F4","Danu_MB_1_F6","Danu_MB_1_F7","Danu_MB_1_F8","Danu_MB_1_F9","Danu_MB_1_G10","Danu_MB_1_G13","Danu_MB_1_G14","Danu_MB_1_G15","Danu_MB_1_G16","Danu_MB_1_G17","Danu_MB_1_G18","Danu_MB_1_G19","Danu_MB_1_G20","Danu_MB_1_G21","Danu_MB_1_G22","Danu_MB_1_G3","Danu_MB_1_G5","Danu_MB_1_G6","Danu_MB_1_G7","Danu_MB_1_G8","Danu_MB_1_G9","Danu_MB_1_J10","Danu_MB_1_J13","Danu_MB_1_J14","Danu_MB_1_J15","Danu_MB_1_J16","Danu_MB_1_J17","Danu_MB_1_J18","Danu_MB_1_J19","Danu_MB_1_J2","Danu_MB_1_J20","Danu_MB_1_J21","Danu_MB_1_J22","Danu_MB_1_J3","Danu_MB_1_J4","Danu_MB_1_J5","Danu_MB_1_J6","Danu_MB_1_J7","Danu_MB_1_J8","Danu_MB_1_J9","Danu_MB_1_K1","Danu_MB_1_K10","Danu_MB_1_K14","Danu_MB_1_K15","Danu_MB_1_K16","Danu_MB_1_K17","Danu_MB_1_K18","Danu_MB_1_K19","Danu_MB_1_K2","Danu_MB_1_K20","Danu_MB_1_K21","Danu_MB_1_K22","Danu_MB_1_K3","Danu_MB_1_K4","Danu_MB_1_K5","Danu_MB_1_K6","Danu_MB_1_K7","Danu_MB_1_K8","Danu_MB_1_K9","Danu_MB_1_L1","Danu_MB_1_L10","Danu_MB_1_L13","Danu_MB_1_L14","Danu_MB_1_L15","Danu_MB_1_L16","Danu_MB_1_L17","Danu_MB_1_L18","Danu_MB_1_L19","Danu_MB_1_L20","Danu_MB_1_L21","Danu_MB_1_L22","Danu_MB_1_L4","Danu_MB_1_L5","Danu_MB_1_L6","Danu_MB_1_L7","Danu_MB_1_L8","Danu_MB_1_M10","Danu_MB_1_M13","Danu_MB_1_M14","Danu_MB_1_M15","Danu_MB_1_M16","Danu_MB_1_M17","Danu_MB_1_M18","Danu_MB_1_M19","Danu_MB_1_M2","Danu_MB_1_M20","Danu_MB_1_M21","Danu_MB_1_M22","Danu_MB_1_M3","Danu_MB_1_M4","Danu_MB_1_M5","Danu_MB_1_M6","Danu_MB_1_M7","Danu_MB_1_M8","Danu_MB_1_M9","Danu_MB_1_N1","Danu_MB_1_N10","Danu_MB_1_N13","Danu_MB_1_N14","Danu_MB_1_N16","Danu_MB_1_N17","Danu_MB_1_N18","Danu_MB_1_N19","Danu_MB_1_N2","Danu_MB_1_N20","Danu_MB_1_N21","Danu_MB_1_N22","Danu_MB_1_N3","Danu_MB_1_N4","Danu_MB_1_N5","Danu_MB_1_N6","Danu_MB_1_N7","Danu_MB_1_N8","Danu_MB_1_N9","Danu_MB_1_O1","Danu_MB_1_O10","Danu_MB_1_O13","Danu_MB_1_O14","Danu_MB_1_O15","Danu_MB_1_O16","Danu_MB_1_O18","Danu_MB_1_O19","Danu_MB_1_O2","Danu_MB_1_O20","Danu_MB_1_O21","Danu_MB_1_O22","Danu_MB_1_O3","Danu_MB_1_O4","Danu_MB_1_O5","Danu_MB_1_O6","Danu_MB_1_O7","Danu_MB_1_O8","Danu_MB_1_O9"],[false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true],["WT.Day_3.3","GID1KO.Day_9.1","GID2KO.Day_3.1","GID2KO.Day_9.1","GID2KO.Day_9.4","GID2KO.Day_3.1","GID2KO.Day_9.1","GID2KO.Day_3.2","GID2KO.Day_9.2","GID2KO.Day_3.3","GID8KO.Day_3.1","GID8KO.Day_9.1","GID8KO.Day_3.3","GID8KO.Day_3.1","GID8KO.Day_9.1","GID8KO.Day_9.2","WT.Day_6.1","GID1KO.Day_6.1","GID2KO.Day_12.1","GID2KO.Day_6.2","GID2KO.Day_6.5","GID2KO.Day_6.1","GID9KO.Day_6.2","GID9KO.Day_6.3","WT.Day_3.1","WT.Day_9.5","GID1KO.Day_3.1","GID1KO.Day_9.1","GID1KO.Day_3.2","GID1KO.Day_9.2","GID1KO.Day_3.3","GID1KO.Day_9.3","GID1KO.Day_3.4","WT.Day_9.1","GID1KO.Day_9.4","GID1KO.Day_3.5","GID1KO.Day_9.5","WT.Day_3.2","WT.Day_9.2","WT.Day_9.3","WT.Day_3.4","WT.Day_9.4","WT.Day_3.5","WT.Day_3.1","WT.Day_9.5","GID1KO.Day_3.1","GID1KO.Day_3.2","GID1KO.Day_9.2","GID1KO.Day_3.3","GID1KO.Day_9.3","GID1KO.Day_3.4","WT.Day_9.1","GID1KO.Day_9.4","GID1KO.Day_3.5","GID1KO.Day_9.5","WT.Day_3.2","WT.Day_9.2","WT.Day_3.3","WT.Day_9.3","WT.Day_3.4","WT.Day_9.4","WT.Day_3.5","GID2KO.Day_9.5","GID7KO.Day_3.1","GID7KO.Day_9.1","GID7KO.Day_3.2","GID7KO.Day_9.2","GID7KO.Day_3.3","GID7KO.Day_9.3","GID7KO.Day_3.4","GID7KO.Day_9.4","GID7KO.Day_3.5","GID7KO.Day_9.5","GID2KO.Day_3.2","GID2KO.Day_9.2","GID2KO.Day_3.3","GID2KO.Day_9.3","GID2KO.Day_3.4","GID2KO.Day_3.5","GID2KO.Day_9.5","GID7KO.Day_3.1","GID7KO.Day_9.1","GID7KO.Day_3.2","GID7KO.Day_9.2","GID7KO.Day_3.3","GID7KO.Day_9.3","GID7KO.Day_3.4","GID7KO.Day_9.4","GID7KO.Day_3.5","GID7KO.Day_9.5","GID2KO.Day_9.3","GID2KO.Day_3.4","GID2KO.Day_9.4","GID2KO.Day_3.5","GID8KO.Day_9.5","GID9KO.Day_3.1","GID9KO.Day_9.1","GID9KO.Day_3.2","GID9KO.Day_9.2","GID9KO.Day_3.3","GID9KO.Day_9.3","GID9KO.Day_3.4","GID9KO.Day_9.4","GID9KO.Day_3.5","GID9KO.Day_9.5","GID8KO.Day_3.2","GID8KO.Day_9.2","GID8KO.Day_9.3","GID8KO.Day_3.4","GID8KO.Day_9.4","GID8KO.Day_3.5","GID8KO.Day_9.5","GID9KO.Day_3.1","GID9KO.Day_9.1","GID9KO.Day_3.2","GID9KO.Day_9.2","GID9KO.Day_3.3","GID9KO.Day_9.3","GID9KO.Day_3.4","GID9KO.Day_9.4","GID9KO.Day_3.5","GID9KO.Day_9.5","GID8KO.Day_3.2","GID8KO.Day_3.3","GID8KO.Day_9.3","GID8KO.Day_3.4","GID8KO.Day_9.4","GID8KO.Day_3.5","WT.Day_12.5","GID1KO.Day_6.1","GID1KO.Day_12.1","GID1KO.Day_6.2","GID1KO.Day_12.2","GID1KO.Day_6.3","GID1KO.Day_12.3","GID1KO.Day_6.4","WT.Day_12.1","GID1KO.Day_12.4","GID1KO.Day_6.5","GID1KO.Day_12.5","WT.Day_6.2","WT.Day_12.2","WT.Day_6.3","WT.Day_12.3","WT.Day_6.4","WT.Day_12.4","WT.Day_6.5","WT.Day_6.1","WT.Day_12.5","GID1KO.Day_12.1","GID1KO.Day_6.2","GID1KO.Day_12.2","GID1KO.Day_6.3","GID1KO.Day_12.3","GID1KO.Day_6.4","WT.Day_12.1","GID1KO.Day_12.4","GID1KO.Day_6.5","GID1KO.Day_12.5","WT.Day_6.2","WT.Day_12.2","WT.Day_6.3","WT.Day_12.3","WT.Day_6.4","WT.Day_12.4","WT.Day_6.5","GID2KO.Day_6.1","GID2KO.Day_12.5","GID7KO.Day_6.1","GID7KO.Day_12.1","GID7KO.Day_6.2","GID7KO.Day_12.2","GID7KO.Day_6.3","GID7KO.Day_12.3","GID7KO.Day_6.4","GID7KO.Day_12.4","GID7KO.Day_6.5","GID7KO.Day_12.5","GID2KO.Day_12.2","GID2KO.Day_6.3","GID2KO.Day_12.3","GID2KO.Day_6.4","GID2KO.Day_12.4","GID2KO.Day_12.5","GID7KO.Day_6.1","GID7KO.Day_12.1","GID7KO.Day_6.2","GID7KO.Day_12.2","GID7KO.Day_6.3","GID7KO.Day_12.3","GID7KO.Day_6.4","GID2KO.Day_12.1","GID7KO.Day_12.4","GID7KO.Day_6.5","GID7KO.Day_12.5","GID2KO.Day_6.2","GID2KO.Day_12.2","GID2KO.Day_6.3","GID2KO.Day_12.3","GID2KO.Day_6.4","GID2KO.Day_12.4","GID2KO.Day_6.5","GID8KO.Day_6.1","GID8KO.Day_12.5","GID9KO.Day_6.1","GID9KO.Day_12.1","GID9KO.Day_12.2","GID9KO.Day_6.3","GID9KO.Day_12.3","GID9KO.Day_6.4","GID8KO.Day_12.1","GID9KO.Day_12.4","GID9KO.Day_6.5","GID9KO.Day_12.5","GID8KO.Day_6.2","GID8KO.Day_12.2","GID8KO.Day_6.3","GID8KO.Day_12.3","GID8KO.Day_6.4","GID8KO.Day_12.4","GID8KO.Day_6.5","GID8KO.Day_6.1","GID8KO.Day_12.5","GID9KO.Day_6.1","GID9KO.Day_12.1","GID9KO.Day_6.2","GID9KO.Day_12.2","GID9KO.Day_12.3","GID9KO.Day_6.4","GID8KO.Day_12.1","GID9KO.Day_12.4","GID9KO.Day_6.5","GID9KO.Day_12.5","GID8KO.Day_6.2","GID8KO.Day_12.2","GID8KO.Day_6.3","GID8KO.Day_12.3","GID8KO.Day_6.4","GID8KO.Day_12.4","GID8KO.Day_6.5"],[618709,494157,38686,25102,2114,3076,87199,23246,9597,516724,25216,12660,384181,10623,10617,65564,4480,767680,62005,163474,16125424,64364,8962,16545066,1573243,3427374,2657532,27227738,2367527,5836341,3266888,2828035,1711920,3497314,15588557,1857539,5994358,4017638,5783605,3455798,1670240,3564499,1781956,1270615,3614040,1808339,6239471,3102127,1763618,1819303,1211397,8370219,8079414,1549539,3184058,1417346,10193858,777771,3130661,1220789,9389420,1859532,4221548,5064181,6731847,4141401,2148204,2026152,3841437,5095293,3359411,1921979,2374850,3287622,3996499,3529557,6182244,2279588,1330333,2509885,1319399,7269341,1177705,1092396,1363777,1624825,2191641,847943,1048155,1774473,2832150,1400052,9765664,5519063,935901,1529043,1238440,1353111,3668022,2426663,3022377,2029381,3634497,1316216,2080343,3774393,4271983,1491271,1352269,2743639,1451394,4623876,1307376,4185398,1099614,2696766,2301400,1464644,1394014,2130848,1175105,1896745,5460854,1162439,983010,1108817,1303418,3473657,1613228,6166541,4386421,2371124,3264318,2188632,4870232,1999048,9889988,2812798,3101805,3129512,1498389,5211924,2362700,4917293,1657312,4263457,2367954,2526229,1760937,2676307,1437333,1470051,3481940,1328802,1844558,2951586,1498614,1723112,1059388,7401837,3435795,1902288,1961852,1663410,6279130,1180304,6888124,7889361,2406506,1893089,2329654,1906303,4400401,3697323,2507075,2132927,2795157,1713836,5516088,5307412,6742707,2064318,2213233,3352646,2036084,3120358,1559113,1401357,2273729,3224375,6399233,1842714,1442743,1433110,1090930,1957573,3941159,3032289,2241852,1515095,1636879,11082303,1906072,2510071,3039879,4385712,3281466,5142561,15857195,2557722,11437384,3005024,3601659,4016035,3526002,6222527,3023572,6775993,2491744,5120809,7561134,1524041,1485139,12061535,1907874,2516263,2205099,2575690,3043553,3537122,2780292,1891096,2093917,3171655,6762821,8215619,2132443,9029019,2343480,2565202],[4322,4485,2771,2037,471,547,4057,1882,1050,4265,2589,1516,4175,1234,1172,3291,621,4726,3690,3406,5422,2711,928,5436,4874,5094,4982,5474,5107,5291,5132,5099,4834,5119,5357,4925,5268,5267,5187,5115,4911,5101,4944,4698,5051,4806,5310,5110,4930,4872,4608,5212,5293,4735,5115,4760,5281,4552,5048,4633,5251,5007,5202,5221,5312,5241,5079,4902,5183,5341,5156,4874,5105,5200,5170,5268,5274,5058,4673,5060,4644,5291,4600,4787,4712,4901,4834,4607,4518,4904,5164,4685,5275,5382,4786,4767,4820,4656,5189,4894,5165,4900,5167,4781,5039,5160,5218,4964,4692,5208,4651,5226,4571,5230,4573,5063,4901,4885,4614,4957,4624,4950,5294,4587,4764,4537,4967,5287,4821,5279,5332,5071,5240,5046,5342,5063,5291,5236,5132,5248,4982,5202,5137,5181,4968,5101,5033,5110,4851,5169,4851,4928,5203,4975,4928,5034,4998,4931,4913,5316,5054,4950,4850,4983,5164,4802,5312,5437,5027,5133,5045,5128,5220,5286,5123,5178,5138,5134,5397,5262,5425,5020,5175,5283,4926,5225,4835,4958,5028,5240,5252,5168,5050,4848,4933,4952,5308,5124,5145,4874,5033,5317,5087,5236,5137,5332,5293,5253,5558,5162,5496,5277,5218,5333,5210,5446,5155,5437,5099,5384,5311,4939,5014,5338,5056,5062,5123,5179,5168,5347,5221,5026,5151,5166,5433,5303,5146,5267,5172,5057],[1.169855295462002,6.484781152548684,1.496665460373262,1.948051948051948,2.554399243140965,1.332899869960988,0.6984025046158786,1.51423900886174,1.875586120662707,1.024531471346405,1.411802030456853,1.84044233807267,3.227124714652728,2.607549656405912,1.57294904398606,3.012323836251602,1.138392857142857,8.455997290537724,1.853076364809289,2.253569374946474,4.356561415067287,1.441799763843142,0.5913858513724615,3.73802679300282,0.9618348850114063,1.330377134214124,0.9192739729944925,2.348733486417418,1.393859499807183,1.455021219630587,1.0892629315728,1.353696117622307,1.126688162998271,1.294765068278113,1.556968999760529,1.193676148926079,1.417716459377301,1.789758061826377,1.387024183013882,1.479542496407487,1.061823450522081,1.504587320686582,1.458116810965029,0.7549887259319307,1.099157729300174,0.8957391285594128,1.258311802394786,1.434338439399805,0.9882525580936462,1.420598987634276,0.9701196222212867,1.389593271096013,1.214877712665795,0.9722246422968379,1.469665439511466,0.9502972456972397,1.18446813757853,1.065480713474789,1.214503901891645,0.8582973798092872,1.284530886891842,1.39631907383148,1.461478111820593,1.029781518472582,1.351976656629303,0.7855795659488178,1.328737866608572,1.062654726792462,1.636314743675348,1.892158115342925,1.423136377180404,0.9034958238357442,1.405772996189233,1.250174138024384,1.486275862949046,2.469035065873706,1.657682873726757,1.364500953681104,0.987948130280163,1.43922928739763,1.045021255889992,1.557541460773404,0.7879732190998594,1.316555534806059,0.7952913122893259,1.336205437508655,0.8519187220899773,1.505525725196151,0.8944287820026618,1.361192872475377,2.370884310506153,0.8369689125832469,1.175270826438428,2.573951411679845,1.330162057739013,0.9294048630417849,1.267239430250961,0.9370258611451684,1.394811699602674,0.8167594758728345,1.644533425181571,1.08688314318504,1.373229913245217,1.26081129541048,1.250803353100907,1.344242637160465,1.393708729646162,1.261071931258638,0.9890044066676084,2.000117362378943,0.8794992951603767,1.600821475316379,0.7606075069452094,1.255866228253562,0.9672485072034368,1.256430850878423,0.9493786390892499,1.194624768885818,0.9578096059293523,1.203933832915346,0.9918262623340042,1.262320448979699,1.197102138236986,0.8892509628462225,1.444746238593707,0.7912035980689329,1.522842250145387,1.756304666810799,1.346430882677464,2.095664327862249,1.560611715108969,2.004914125115346,1.259252315491322,2.059048757397315,1.630066083094194,1.951228784901613,1.898677733481578,1.592329061667421,1.880679152944818,1.853931219947391,2.181009070408285,1.497988075037165,2.14331061920684,1.385884469361496,1.761768453978491,1.220910636603113,2.08230396367497,1.838194399636771,1.237579765772427,1.521462223877903,1.904429940730506,1.368319874616595,1.82111121960746,1.352421203459959,1.605696323997402,1.361674706412078,1.183026449772923,1.992673720570689,2.365516694544398,1.854336970673632,1.282818095957413,1.77565121579908,1.317530578249531,1.846989016538316,1.291803163814095,1.987538803562472,2.553380281771931,1.748760134059019,2.059167938912265,1.720521327840371,1.920156383737671,1.804802279595636,2.162757439606072,1.605918660609311,1.865520576767747,1.836209115454959,2.539964660303518,1.835706567022749,1.878686489410611,2.193291193523322,1.487874232114787,2.286905408953465,1.539783655855484,1.524199095281757,1.836712041349964,1.67471168372347,1.855221526598778,1.654610495398389,1.756189941721287,1.697383213801124,1.881928662388133,1.913916104181115,1.712432498372891,1.914088939439401,1.698184118137736,1.986030661436381,1.726040487074995,1.868918167100827,1.415080032044934,2.087195852405295,1.580019048445243,1.655341854486382,2.777492140905485,1.69007171510288,1.898068969192524,1.487466573272481,1.472908754806541,2.504258092417377,1.789793213743036,2.189565558727649,1.593913433351543,1.51576160456622,2.755480182882388,1.976128196093908,2.27237534181773,1.548117027053479,2.393129715449144,1.645958016780714,2.003456213800455,1.741658398116391,1.225900770969011,1.88006753099162,1.498310932512041,1.528072504867747,1.511630222960217,1.653920913672378,1.409415178184744,1.425520928372591,1.873632560366125,1.591039268648353,1.501317127841248,2.103171917237411,1.774234604332454,1.754888220818469,1.608943959924416,2.130807185678888,1.465408454059499,1.547909025332652,1.516249338590472,1.567400929829308],[60.83764742391011,35.24568102849904,45.95460890244533,53.84829894032348,49.24314096499527,45.54616384915474,30.9923278936685,39.64983222920072,48.06710430342815,57.02367221185778,35.23556472081219,58.1437598736177,40.80680720806079,39.76277887602372,46.91532447960818,25.69397840278202,57.12053571428572,34.3617132138391,27.20748326747843,8.242289293710314,23.56701442393081,50.03262693431111,64.92970319125195,19.66220624323892,61.09253306704686,42.35242491773585,63.00714346995634,30.79728473955494,49.03310500788375,32.55538016027508,52.9217714228342,45.60060253851172,65.34732931445394,38.74196025864421,29.41679592280414,60.00277786899764,30.45680621677917,45.78244729863666,37.97959922920047,37.06278549845796,64.95611409138806,37.4332830504371,50.57661356397127,68.52351026864943,52.00921959911899,68.17167577539388,39.37973267284999,50.25951548727695,61.77998863699509,58.42605657221475,67.57157232517498,34.41515687940782,45.81271612025328,66.5957423465947,45.99391719623198,64.51304057019246,30.07929873066704,57.93106197068288,48.06246987457282,67.35881466821867,36.8811385580792,49.72939427769998,44.79416081494276,58.95810201096683,24.67187682667179,46.13861347886862,42.25990641484701,61.51310464367925,34.043223928962,27.21568710572679,43.02560776278937,65.53151725383056,44.46209234267427,45.53117724604593,43.84254819030357,31.81453083205626,41.08404650479664,45.27186491594095,64.96343396728489,48.96252218727153,63.02180007715634,36.13533881544421,72.83020790435636,53.23911841493378,68.30236908233532,55.96651947132768,69.97505522117902,59.21895693460527,68.99456664329226,55.32662373561051,40.73693130660453,70.65444712053552,56.31372326551477,19.33931538741268,51.82001087721885,68.33287226062315,61.34144568973871,68.80795441024425,46.1368824941617,68.9018623517151,39.93896194948545,62.8116159558013,43.73067304774223,62.00707178760933,50.57329488454548,45.48471767513346,30.92800228839862,46.77587105227688,64.62020500359026,26.93739956313495,69.40541300294751,43.76460354905711,73.69257199153112,40.70692918570707,70.5030128754272,58.78778507293551,68.12092639263058,57.86785048107254,71.1660715028687,61.48481731216867,67.34087592172614,58.166121434352,46.95650167537898,66.00690444831943,55.71103040660827,69.44274844270967,42.29341623331886,31.40874300485051,56.92964664635129,37.13613515259203,36.06616875124389,55.66520350686004,46.34974288656926,53.53846603723239,40.11576861225502,52.97346536951589,17.82128552633229,45.35978054591905,56.39064996026507,38.40502928252072,43.34842287283209,33.83982575340699,49.99115418800525,40.90510774932469,50.26416269235968,50.26423393035277,52.88793616767893,54.50768714950228,57.58689833878213,52.55002509054455,61.24134073314952,60.77170111785237,42.26224461076297,48.23660710926082,63.8464065646079,47.29809668429109,56.73435587816476,62.2868971953071,40.34508603080269,17.84481338889251,48.49267782274554,62.32920567232721,56.47434159151658,49.79259473010262,54.57940829382415,56.69047974081253,32.3853200087571,30.44214861000783,63.06242328088939,46.95642941245763,64.31289796682255,51.42702917636913,50.49073936670771,42.26617474318581,58.75105451572051,43.80670318299688,50.38232199479313,40.0531906203394,30.90614217902252,51.95172336347734,32.28302816658057,59.57977404644051,46.41720957531358,45.38140919142671,68.42443631991607,53.16264992670713,70.81205788162886,57.21489955807121,64.22436446911659,50.11222329908897,63.3815177537683,44.49605310427988,49.54243409948965,67.46474450670222,50.94909847561255,66.92317476793968,45.95336042012007,61.97288582981372,52.67261175135558,66.55516650771074,56.1667661445959,63.56411659201161,45.94558862414431,45.93977620553363,58.75585837462609,42.46051724326632,44.20569952576074,46.89976453366329,14.48139472334168,42.77622822183177,13.33339861632695,37.56612260001917,41.65888553025147,29.4936423611846,49.87674425595902,30.92671192909247,51.91879009330685,31.03955390744943,57.71267032247294,31.88398551869441,50.51227765570614,61.02152107456427,54.8409273475412,29.8031386552375,57.36290761339585,63.25892802143495,50.72801720013478,49.07306391685335,43.20345333233888,37.57871512489533,46.90827438269074,57.62859209685812,43.60330423794257,55.50389307790412,28.37049805103521,45.9757566654442,51.16980852477651,64.56183113580778,51.05923669073344,64.43106624741444]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>sample<\/th>\n      <th>keep<\/th>\n      <th>sample<\/th>\n      <th>sum<\/th>\n      <th>detected<\/th>\n      <th>subsets_Mito_percent<\/th>\n      <th>subsets_Ribo_percent<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render"],"jsHooks":[]}</script>
</div>
<h2 id="summary-4">Summary</h2>
<p>We have opted not to exclude any replicates at this stage.
However, we have identified some replicates with unusually small library sizes or number of genes detected or unusually high mitochondrial or mitochondrial gene expression percentage compared to the other samples.
We will keep an eye on these replicates and may later remove some of these in downstream analysis.</p>
<h1 id="examining-gene-level-metrics">Examining gene-level metrics</h1>
<h2 id="inspecting-the-most-highly-expressed-genes">Inspecting the most highly expressed genes</h2>
<p>Figure <a href="#fig:topgenes">23</a> shows the most highly expressed genes across all samples.
The two most highly expressed genes, <em>PF3D7_0726000.1</em> and <em>PF3D7_0532000.1</em>, are rRNAs.
Figure <a href="#fig:topgenes-filtered">24</a> shows the most highly expressed endogenous genes after excluding the mitochondrial and ribosomal protein genes, where we observe genes that are more relevant to the cell types and biology we are trying to study.</p>
<div class="panelset">
<section id="all-genes" class="panel">
<h3>All genes</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHighestExprs.html'>plotHighestExprs</a></span><span class='op'>(</span><span class='va'>sce</span>, n <span class='op'>=</span> <span class='fl'>50</span>, exprs_values <span class='op'>=</span> <span class='st'>"read_counts"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:topgenes"></span>
<img src="G000396_Danu.preprocess_files/topgenes-1.png" alt="Percentage of total counts assigned to the top 50 most   highly-abundant features in the combined data set. For each feature, each   bar represents the percentage assigned to that feature for a single cell,   while the circle represents the average across all cells." width="624" />
<p class="caption">
Figure 23: Percentage of total counts assigned to the top 50 most highly-abundant features in the combined data set. For each feature, each bar represents the percentage assigned to that feature for a single cell, while the circle represents the average across all cells.
</p>
</div>
</div>
</section>
<section id="filtered-genes" class="panel">
<h3>Filtered genes</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plotHighestExprs.html'>plotHighestExprs</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>,</span>
<span>  drop_features <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'><a href='https://rdrr.io/pkg/BiocGenerics/man/match.html'>%in%</a></span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='va'>mito_set</span>, <span class='va'>ribo_set</span><span class='op'>)</span>,</span>
<span>  n <span class='op'>=</span> <span class='fl'>50</span>,</span>
<span>  exprs_values <span class='op'>=</span> <span class='st'>"read_counts"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:topgenes-filtered"></span>
<img src="G000396_Danu.preprocess_files/topgenes-filtered-1.png" alt="Percentage of total counts assigned to the top 50 most   highly-abundant features in the combined data set. For each feature,   each bar represents the percentage assigned to that feature for a single   cell, while the circle represents the average across all cells. Bars are   coloured by the total number of expressed features in each cell, while   circles are coloured according to whether the feature is labelled as a   control feature." width="624" />
<p class="caption">
Figure 24: Percentage of total counts assigned to the top 50 most highly-abundant features in the combined data set. For each feature, each bar represents the percentage assigned to that feature for a single cell, while the circle represents the average across all cells. Bars are coloured by the total number of expressed features in each cell, while circles are coloured according to whether the feature is labelled as a control feature.
</p>
</div>
</div>
</section>
</div>
<h1 id="mds-plot">MDS plot</h1>
<p>We conclude with some MDS visualisations of the data.</p>
<h2 id="invividual-technical-replicates">Invividual technical replicates</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"logCPM_reads"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"read_counts"</span><span class='op'>)</span>, log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='va'>dec</span> <span class='op'>&lt;-</span> <span class='fu'>scran</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/scran/man/modelGeneVar.html'>modelGeneVar</a></span><span class='op'>(</span><span class='va'>sce</span>, assay.type <span class='op'>=</span> <span class='st'>"logCPM_reads"</span><span class='op'>)</span></span>
<span><span class='va'>hvg</span> <span class='op'>&lt;-</span> <span class='fu'>scran</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/scran/man/getTopHVGs.html'>getTopHVGs</a></span><span class='op'>(</span><span class='va'>dec</span>, n <span class='op'>=</span> <span class='fl'>1000</span><span class='op'>)</span></span>
<span><span class='va'>sce</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/runMDS.html'>runMDS</a></span><span class='op'>(</span><span class='va'>sce</span>, subset_row <span class='op'>=</span> <span class='va'>hvg</span>, exprs_values <span class='op'>=</span> <span class='st'>"logCPM_reads"</span><span class='op'>)</span></span>
<span><span class='va'>mds_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/makePerCellDF.html'>makePerCellDF</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span></span>
<span><span class='va'>bg</span> <span class='op'>&lt;-</span> <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span></span>
<span>  <span class='va'>mds_df</span>, </span>
<span>  <span class='op'>-</span><span class='va'>sample_type</span>, <span class='op'>-</span><span class='va'>original_plate</span>, <span class='op'>-</span><span class='va'>cell_line</span>, <span class='op'>-</span><span class='va'>timepoint</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<div class="panelset">
<section id="sample-type" class="panel">
<h3>Sample type</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>MDS.1</span>, y <span class='op'>=</span> <span class='va'>MDS.2</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>mds_df</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>bg</span>, colour <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/alpha.html'>alpha</a></span><span class='op'>(</span><span class='st'>"grey"</span>, <span class='fl'>0.5</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>sample_type</span><span class='op'>)</span>, alpha <span class='op'>=</span> <span class='fl'>1</span>, size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='op'>~</span> <span class='va'>sample_type</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Faceted by sample_type"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:mds-by-sample-type"></span>
<img src="G000396_Danu.preprocess_files/mds-by-sample-type-1.png" alt="MDS plot of the dataset. Each point represents a sample and   is coloured by `sample_type` with each panel highlighting droplets from   a particular `sample_type`." width="624" />
<p class="caption">
Figure 25: MDS plot of the dataset. Each point represents a sample and is coloured by <code>sample_type</code> with each panel highlighting droplets from a particular <code>sample_type</code>.
</p>
</div>
</div>
</section>
<section id="original-plate" class="panel">
<h3>Original plate</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>MDS.1</span>, y <span class='op'>=</span> <span class='va'>MDS.2</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>mds_df</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>bg</span>, colour <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/alpha.html'>alpha</a></span><span class='op'>(</span><span class='st'>"grey"</span>, <span class='fl'>0.5</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>original_plate</span><span class='op'>)</span>, alpha <span class='op'>=</span> <span class='fl'>1</span>, size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_wrap.html'>facet_wrap</a></span><span class='op'>(</span><span class='op'>~</span> <span class='va'>original_plate</span>, ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Faceted by sample_type"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:mds-by-original-plate"></span>
<img src="G000396_Danu.preprocess_files/mds-by-original-plate-1.png" alt="MDS plot of the dataset. Each point represents a sample and   is coloured by `original_plate` with each panel highlighting droplets   from a particular `original_plate`." width="624" />
<p class="caption">
Figure 26: MDS plot of the dataset. Each point represents a sample and is coloured by <code>original_plate</code> with each panel highlighting droplets from a particular <code>original_plate</code>.
</p>
</div>
</div>
</section>
<section id="cell-line" class="panel">
<h3>Cell line</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>MDS.1</span>, y <span class='op'>=</span> <span class='va'>MDS.2</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>mds_df</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>bg</span>, colour <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/alpha.html'>alpha</a></span><span class='op'>(</span><span class='st'>"grey"</span>, <span class='fl'>0.5</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>cell_line</span><span class='op'>)</span>, alpha <span class='op'>=</span> <span class='fl'>1</span>, size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_wrap.html'>facet_wrap</a></span><span class='op'>(</span><span class='op'>~</span> <span class='va'>cell_line</span>, ncol <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Faceted by cell_line"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:mds-by-cell-line"></span>
<img src="G000396_Danu.preprocess_files/mds-by-cell-line-1.png" alt="MDS plot of the dataset. Each point represents a sample and   is coloured by `cell_line` with each panel highlighting droplets from   a particular `cell_line`." width="624" />
<p class="caption">
Figure 27: MDS plot of the dataset. Each point represents a sample and is coloured by <code>cell_line</code> with each panel highlighting droplets from a particular <code>cell_line</code>.
</p>
</div>
</div>
</section>
<section id="timepoint" class="panel">
<h3>Timepoint</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>MDS.1</span>, y <span class='op'>=</span> <span class='va'>MDS.2</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>mds_df</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>bg</span>, colour <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/alpha.html'>alpha</a></span><span class='op'>(</span><span class='st'>"grey"</span>, <span class='fl'>0.5</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>timepoint</span><span class='op'>)</span>, alpha <span class='op'>=</span> <span class='fl'>1</span>, size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_wrap.html'>facet_wrap</a></span><span class='op'>(</span><span class='op'>~</span> <span class='va'>timepoint</span>, ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Faceted by timepoint"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:mds-by-timepoint"></span>
<img src="G000396_Danu.preprocess_files/mds-by-timepoint-1.png" alt="MDS plot of the dataset. Each point represents a sample and   is coloured by `timepoint` with each panel highlighting droplets from   a particular `timepoint`." width="624" />
<p class="caption">
Figure 28: MDS plot of the dataset. Each point represents a sample and is coloured by <code>timepoint</code> with each panel highlighting droplets from a particular <code>timepoint</code>.
</p>
</div>
</div>
</section>
<section id="cell-line-timepoint" class="panel">
<h3>Cell line + timepoint</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>MDS.1</span>, y <span class='op'>=</span> <span class='va'>MDS.2</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>mds_df</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>bg</span>, colour <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/alpha.html'>alpha</a></span><span class='op'>(</span><span class='st'>"grey"</span>, <span class='fl'>0.5</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.5</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>cell_line</span><span class='op'>)</span>, alpha <span class='op'>=</span> <span class='fl'>1</span>, size <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span><span class='op'>~</span> <span class='va'>cell_line</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Faceted by cell_line + timepoint"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:mds-by-cell-line-timepoint"></span>
<img src="G000396_Danu.preprocess_files/mds-by-cell-line-timepoint-1.png" alt="MDS plot of the dataset. Each point represents a sample and   is coloured by `cell_line` with each panel highlighting droplets from   a particular `cell_line`." width="624" />
<p class="caption">
Figure 29: MDS plot of the dataset. Each point represents a sample and is coloured by <code>cell_line</code> with each panel highlighting droplets from a particular <code>cell_line</code>.
</p>
</div>
</div>
</section>
<section id="library-size" class="panel">
<h3>Library size</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'>scater</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plot_reddim.html'>plotMDS</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>`log10(sum)` <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log10</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>$</span><span class='va'>sum</span><span class='op'>)</span>, check.names <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>, </span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  point_size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:mds-by-library-size"></span>
<img src="G000396_Danu.preprocess_files/mds-by-library-size-1.png" alt="MDS plot of the dataset. Each point represents a sample and   is coloured by library size (log10(`sum`))." width="624" />
<p class="caption">
Figure 30: MDS plot of the dataset. Each point represents a sample and is coloured by library size (log10(<code>sum</code>)).
</p>
</div>
</div>
</section>
</div>
<h2 id="aggregated-technical-replicates-1">Aggregated technical replicates</h2>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce_aggregated</span>, <span class='st'>"logCPM_reads"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'>edgeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cpm.html'>cpm</a></span><span class='op'>(</span></span>
<span>  <span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce_aggregated</span>, <span class='st'>"read_counts"</span><span class='op'>)</span>, </span>
<span>  log <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span></span>
<span><span class='va'>dec</span> <span class='op'>&lt;-</span> <span class='fu'>scran</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/scran/man/modelGeneVar.html'>modelGeneVar</a></span><span class='op'>(</span><span class='va'>sce_aggregated</span>, assay.type <span class='op'>=</span> <span class='st'>"logCPM_reads"</span><span class='op'>)</span></span>
<span><span class='va'>hvg</span> <span class='op'>&lt;-</span> <span class='fu'>scran</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/scran/man/getTopHVGs.html'>getTopHVGs</a></span><span class='op'>(</span><span class='va'>dec</span>, n <span class='op'>=</span> <span class='fl'>1000</span><span class='op'>)</span></span>
<span><span class='va'>sce_aggregated</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scater/man/runMDS.html'>runMDS</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce_aggregated</span>, </span>
<span>  subset_row <span class='op'>=</span> <span class='va'>hvg</span>, </span>
<span>  exprs_values <span class='op'>=</span> <span class='st'>"logCPM_reads"</span><span class='op'>)</span></span>
<span><span class='va'>mds_df</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/scuttle/man/makePerCellDF.html'>makePerCellDF</a></span><span class='op'>(</span><span class='va'>sce_aggregated</span><span class='op'>)</span></span>
<span><span class='va'>bg</span> <span class='op'>&lt;-</span> <span class='fu'>dplyr</span><span class='fu'>::</span><span class='fu'><a href='https://dplyr.tidyverse.org/reference/select.html'>select</a></span><span class='op'>(</span></span>
<span>  <span class='va'>mds_df</span>, </span>
<span>  <span class='op'>-</span><span class='va'>sample_type</span>, <span class='op'>-</span><span class='va'>original_plate</span>, <span class='op'>-</span><span class='va'>cell_line</span>, <span class='op'>-</span><span class='va'>timepoint</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<div class="panelset">
<section id="original-plate-1" class="panel">
<h3>Original plate</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>MDS.1</span>, y <span class='op'>=</span> <span class='va'>MDS.2</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>mds_df</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>bg</span>, colour <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/alpha.html'>alpha</a></span><span class='op'>(</span><span class='st'>"grey"</span>, <span class='fl'>0.5</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>original_plate</span><span class='op'>)</span>, alpha <span class='op'>=</span> <span class='fl'>1</span>, size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_wrap.html'>facet_wrap</a></span><span class='op'>(</span><span class='op'>~</span> <span class='va'>original_plate</span>, ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Faceted by sample_type"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:aggregated-mds-by-original-plate"></span>
<img src="G000396_Danu.preprocess_files/aggregated-mds-by-original-plate-1.png" alt="MDS plot of the aggregated dataset. Each point represents a   sample and is coloured by `original_plate` with each panel highlighting   droplets from a particular `original_plate`." width="624" />
<p class="caption">
Figure 31: MDS plot of the aggregated dataset. Each point represents a sample and is coloured by <code>original_plate</code> with each panel highlighting droplets from a particular <code>original_plate</code>.
</p>
</div>
</div>
</section>
<section id="cell-line-1" class="panel">
<h3>Cell line</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>MDS.1</span>, y <span class='op'>=</span> <span class='va'>MDS.2</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>mds_df</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>bg</span>, colour <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/alpha.html'>alpha</a></span><span class='op'>(</span><span class='st'>"grey"</span>, <span class='fl'>0.5</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>cell_line</span><span class='op'>)</span>, alpha <span class='op'>=</span> <span class='fl'>1</span>, size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_wrap.html'>facet_wrap</a></span><span class='op'>(</span><span class='op'>~</span> <span class='va'>cell_line</span>, ncol <span class='op'>=</span> <span class='fl'>3</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Faceted by cell_line"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:aggregated-mds-by-cell-line"></span>
<img src="G000396_Danu.preprocess_files/aggregated-mds-by-cell-line-1.png" alt="MDS plot of the aggregated dataset. Each point represents a   sample and is coloured by `cell_line` with each panel highlighting   droplets from a particular `cell_line`." width="624" />
<p class="caption">
Figure 32: MDS plot of the aggregated dataset. Each point represents a sample and is coloured by <code>cell_line</code> with each panel highlighting droplets from a particular <code>cell_line</code>.
</p>
</div>
</div>
</section>
<section id="timepoint-1" class="panel">
<h3>Timepoint</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>MDS.1</span>, y <span class='op'>=</span> <span class='va'>MDS.2</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>mds_df</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>bg</span>, colour <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/alpha.html'>alpha</a></span><span class='op'>(</span><span class='st'>"grey"</span>, <span class='fl'>0.5</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>timepoint</span><span class='op'>)</span>, alpha <span class='op'>=</span> <span class='fl'>1</span>, size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_wrap.html'>facet_wrap</a></span><span class='op'>(</span><span class='op'>~</span> <span class='va'>timepoint</span>, ncol <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Faceted by timepoint"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:aggregated-mds-by-timepoint"></span>
<img src="G000396_Danu.preprocess_files/aggregated-mds-by-timepoint-1.png" alt="MDS plot of the aggregated dataset. Each point represents a   sample and is coloured by `timepoint` with each panel highlighting   droplets from a particular `timepoint`." width="624" />
<p class="caption">
Figure 33: MDS plot of the aggregated dataset. Each point represents a sample and is coloured by <code>timepoint</code> with each panel highlighting droplets from a particular <code>timepoint</code>.
</p>
</div>
</div>
</section>
<section id="cell-line-timepoint-1" class="panel">
<h3>Cell line + timepoint</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/ggplot.html'>ggplot</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>MDS.1</span>, y <span class='op'>=</span> <span class='va'>MDS.2</span><span class='op'>)</span>, data <span class='op'>=</span> <span class='va'>mds_df</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span>data <span class='op'>=</span> <span class='va'>bg</span>, colour <span class='op'>=</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org/reference/alpha.html'>alpha</a></span><span class='op'>(</span><span class='st'>"grey"</span>, <span class='fl'>0.5</span><span class='op'>)</span>, size <span class='op'>=</span> <span class='fl'>0.5</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/geom_point.html'>geom_point</a></span><span class='op'>(</span><span class='fu'><a href='https://ggplot2.tidyverse.org/reference/aes.html'>aes</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='va'>cell_line</span><span class='op'>)</span>, alpha <span class='op'>=</span> <span class='fl'>1</span>, size <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/theme_cowplot.html'>theme_cowplot</a></span><span class='op'>(</span>font_size <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>xlab</a></span><span class='op'>(</span><span class='st'>"Dimension 1"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ylab</a></span><span class='op'>(</span><span class='st'>"Dimension 2"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/facet_grid.html'>facet_grid</a></span><span class='op'>(</span><span class='va'>timepoint</span><span class='op'>~</span> <span class='va'>cell_line</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/guides.html'>guides</a></span><span class='op'>(</span>colour <span class='op'>=</span> <span class='st'>"none"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://ggplot2.tidyverse.org/reference/labs.html'>ggtitle</a></span><span class='op'>(</span><span class='st'>"Faceted by cell_line + timepoint"</span><span class='op'>)</span> <span class='op'>+</span></span>
<span>  <span class='fu'><a href='https://wilkelab.org/cowplot/reference/panel_border.html'>panel_border</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:aggregated-mds-by-cell-line-timepoint"></span>
<img src="G000396_Danu.preprocess_files/aggregated-mds-by-cell-line-timepoint-1.png" alt="MDS plot of the aggregated dataset. Each point represents a   sample and is coloured by `cell_line` with each panel highlighting   droplets from a particular `cell_line`." width="624" />
<p class="caption">
Figure 34: MDS plot of the aggregated dataset. Each point represents a sample and is coloured by <code>cell_line</code> with each panel highlighting droplets from a particular <code>cell_line</code>.
</p>
</div>
</div>
</section>
<section id="library-size-1" class="panel">
<h3>Library size</h3>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'>scater</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/scater/man/plot_reddim.html'>plotMDS</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce_aggregated</span>,</span>
<span>  colour_by <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span>    `log10(sum)` <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>log10</a></span><span class='op'>(</span><span class='va'>sce_aggregated</span><span class='op'>$</span><span class='va'>sum</span><span class='op'>)</span>,</span>
<span>    check.names <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>, </span>
<span>  point_alpha <span class='op'>=</span> <span class='fl'>1</span>,</span>
<span>  point_size <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<div class="figure"><span style="display:block;" id="fig:aggregated-mds-by-library-size"></span>
<img src="G000396_Danu.preprocess_files/aggregated-mds-by-library-size-1.png" alt="MDS plot of the aggregated dataset. Each point represents a   sample and is coloured by library size (log10(`sum`))." width="624" />
<p class="caption">
Figure 35: MDS plot of the aggregated dataset. Each point represents a sample and is coloured by library size (log10(<code>sum</code>)).
</p>
</div>
</div>
</section>
</div>
<h2 id="summary-5">Summary</h2>
<p>Although Figures <a href="#fig:mds-by-sample-type">25</a> - <a href="#fig:aggregated-mds-by-library-size">35</a> are only a preliminary summary of the data, there are few patterns worth highlighting:</p>
<ul>
<li>Strong association of MDS co-ordinates and <code>timepoint</code></li>
<li>No strong association of MDS co-ordinates and <code>cell_line</code>, but this may become more apparent after accounting for <code>timepoint</code>.</li>
<li>Replicates with small library sizes (i.e.low-quality replicates) cluster separately from the others and may need to be removed or downweighted in downstream analyses.</li>
</ul>
<h1 id="concluding-remarks">Concluding remarks</h1>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='co'># Remove some unneeded assays and reducedDims.</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span><span class='op'>(</span><span class='va'>sce</span>, <span class='st'>"logCPM_reads"</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span></span>
<span><span class='fu'><a href='https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html'>reducedDims</a></span><span class='op'>(</span><span class='va'>sce</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='cn'>NULL</span></span>
<span></span>
<span><span class='co'># Save</span></span>
<span><span class='fu'><a href='https://rdrr.io/r/base/readRDS.html'>saveRDS</a></span><span class='op'>(</span></span>
<span>  <span class='va'>sce</span>,</span>
<span>  <span class='fu'><a href='https://here.r-lib.org//reference/here.html'>here</a></span><span class='op'>(</span><span class='st'>"data"</span>, <span class='st'>"SCEs"</span>, <span class='st'>"G000396_Danu.preprocessed.SCE.rds"</span><span class='op'>)</span>,</span>
<span>  compress <span class='op'>=</span> <span class='st'>"xz"</span><span class='op'>)</span></span></code></pre>
</div>
</details>
</div>
<p>The processed <em>SingleCellExperiment</em> object is available (see <a href="../data/SCEs/G000396_Danu.preprocessed.SCE.rds"><code>data/SCEs/G000396_Danu.preprocessed.SCE.rds</code></a>).
This will be used in downstream analyses, i.e.differential expression analysis.</p>
<h1 class="appendix" id="additional-information">Additional information</h1>
<p>The following are available on request:</p>
<ul>
<li>Full CSV tables of any data presented.</li>
<li>PDF/PNG files of any static plots.</li>
</ul>
<h1 class="appendix" id="session-info">Session info</h1>
<summary>
The analysis and this document were prepared using the following software (click triangle to expand)
</summary>
<details>
<div class="layout-chunk" data-layout="l-body">
<details>
<summary>
Show code
</summary>
<div class="sourceCode">
<pre class="sourceCode r"><code class="sourceCode r"><span><span class='fu'>sessioninfo</span><span class='fu'>::</span><span class='fu'><a href='https://r-lib.github.io/sessioninfo/reference/session_info.html'>session_info</a></span><span class='op'>(</span><span class='op'>)</span></span></code></pre>
</div>
</details>
<pre><code> Session info 
 setting  value
 version  R version 4.3.1 (2023-06-16)
 os       CentOS Linux 7 (Core)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       UTC
 date     2023-11-14
 pandoc   3.1.1 @ /usr/lib/rstudio-server/bin/quarto/bin/tools/ (via rmarkdown)

 Packages 
 ! package              * version   date (UTC) lib source
 P abind                  1.4-5     2016-07-21 [?] RSPM (R 4.3.0)
 P AnnotationDbi        * 1.64.1    2023-11-03 [?] Bioconductor
   basilisk               1.14.0    2023-10-24 [1] Bioconductor
   basilisk.utils         1.14.0    2023-10-24 [1] Bioconductor
   beachmat               2.18.0    2023-10-24 [1] Bioconductor
 P beeswarm               0.4.0     2021-06-01 [?] CRAN (R 4.3.0)
   Biobase              * 2.62.0    2023-10-24 [1] Bioconductor
   BiocFileCache          2.10.1    2023-10-26 [1] Bioconductor
 P BiocGenerics         * 0.48.1    2023-11-01 [?] Bioconductor
   BiocIO                 1.12.0    2023-10-24 [1] Bioconductor
   BiocNeighbors          1.20.0    2023-10-24 [1] Bioconductor
   BiocParallel           1.36.0    2023-10-24 [1] Bioconductor
   BiocSingular           1.18.0    2023-10-24 [1] Bioconductor
   biomaRt                2.58.0    2023-10-24 [1] Bioconductor
   Biostrings             2.70.1    2023-10-25 [1] Bioconductor
 P bit                    4.0.5     2022-11-15 [?] CRAN (R 4.3.0)
 P bit64                  4.0.5     2020-08-30 [?] CRAN (R 4.3.0)
 P bitops                 1.0-7     2021-04-24 [?] CRAN (R 4.3.0)
 P blob                   1.2.4     2023-03-17 [?] CRAN (R 4.3.0)
   bluster                1.12.0    2023-10-24 [1] Bioconductor
 P bslib                  0.5.1     2023-08-11 [?] RSPM (R 4.3.0)
 P cachem                 1.0.8     2023-05-01 [?] CRAN (R 4.3.0)
 P cli                    3.6.1     2023-03-23 [?] CRAN (R 4.3.0)
 P cluster                2.1.4     2022-08-22 [3] CRAN (R 4.3.1)
 P codetools              0.2-19    2023-02-01 [3] CRAN (R 4.3.1)
 P colorspace             2.1-0     2023-01-23 [?] CRAN (R 4.3.0)
 P commonmark             1.9.0     2023-03-17 [?] CRAN (R 4.3.0)
 P cowplot              * 1.1.1     2020-12-30 [?] CRAN (R 4.3.0)
 P crayon                 1.5.2     2022-09-29 [?] CRAN (R 4.3.0)
 P crosstalk              1.2.0     2021-11-04 [?] CRAN (R 4.3.0)
 P curl                   5.1.0     2023-10-02 [?] RSPM (R 4.3.0)
 P data.table             1.14.8    2023-02-17 [?] CRAN (R 4.3.0)
 P DBI                    1.1.3     2022-06-18 [?] CRAN (R 4.3.0)
   dbplyr                 2.4.0     2023-10-26 [1] RSPM (R 4.3.0)
   DelayedArray           0.28.0    2023-10-24 [1] Bioconductor
   DelayedMatrixStats     1.24.0    2023-10-24 [1] Bioconductor
 P DEoptimR               1.1-3     2023-10-07 [?] CRAN (R 4.3.1)
 P digest                 0.6.33    2023-07-07 [?] CRAN (R 4.3.0)
   dir.expiry             1.10.0    2023-10-24 [1] Bioconductor
 P distill                1.6       2023-10-06 [?] CRAN (R 4.3.1)
 P downlit                0.4.3     2023-06-29 [?] CRAN (R 4.3.0)
 P dplyr                  1.1.3     2023-09-03 [?] CRAN (R 4.3.0)
 P dqrng                  0.3.1     2023-08-30 [?] CRAN (R 4.3.0)
   DropletUtils           1.22.0    2023-10-24 [1] Bioconductor
 P DT                     0.30      2023-10-05 [?] RSPM (R 4.3.0)
   edgeR                  4.0.1     2023-10-29 [1] Bioconductor
 P ellipsis               0.3.2     2021-04-29 [?] CRAN (R 4.3.0)
 P evaluate               0.23      2023-11-01 [?] RSPM (R 4.3.0)
 P fansi                  1.0.5     2023-10-08 [?] RSPM (R 4.3.0)
 P farver                 2.1.1     2022-07-06 [?] CRAN (R 4.3.0)
 P fastmap                1.1.1     2023-02-24 [?] CRAN (R 4.3.0)
 P filelock               1.0.2     2018-10-05 [?] CRAN (R 4.3.0)
 P generics               0.1.3     2022-07-05 [?] CRAN (R 4.3.0)
 P GenomeInfoDb         * 1.38.1    2023-11-08 [?] Bioconductor
   GenomeInfoDbData       1.2.11    2023-10-30 [1] Bioconductor
   GenomicAlignments      1.38.0    2023-10-24 [1] Bioconductor
 P GenomicFeatures      * 1.54.1    2023-10-29 [?] Bioconductor
   GenomicRanges        * 1.54.1    2023-10-29 [1] Bioconductor
 P ggbeeswarm             0.7.2     2023-04-29 [?] CRAN (R 4.3.0)
 P ggplot2              * 3.4.4     2023-10-12 [?] RSPM (R 4.3.0)
 P ggrepel              * 0.9.4     2023-10-13 [?] RSPM (R 4.3.0)
 P glue                   1.6.2     2022-02-24 [?] CRAN (R 4.3.0)
 P gridExtra              2.3       2017-09-09 [?] CRAN (R 4.3.0)
 P gt                   * 0.10.0    2023-10-07 [?] RSPM (R 4.3.0)
 P gtable                 0.3.4     2023-08-21 [?] RSPM (R 4.3.0)
   HDF5Array              1.30.0    2023-10-24 [1] Bioconductor
 P here                 * 1.0.1     2020-12-13 [?] CRAN (R 4.3.0)
 P highr                  0.10      2022-12-22 [?] CRAN (R 4.3.0)
 P hms                    1.1.3     2023-03-21 [?] CRAN (R 4.3.0)
 P htmltools              0.5.7     2023-11-03 [?] RSPM (R 4.3.0)
 P htmlwidgets            1.6.2     2023-03-17 [?] CRAN (R 4.3.0)
 P httr                   1.4.7     2023-08-15 [?] RSPM (R 4.3.0)
 P igraph                 1.5.1     2023-08-10 [?] RSPM (R 4.3.0)
   IRanges              * 2.36.0    2023-10-24 [1] Bioconductor
 P irlba                  2.3.5.1   2022-10-03 [?] RSPM (R 4.3.1)
 P jquerylib              0.1.4     2021-04-26 [?] CRAN (R 4.3.0)
 P jsonlite               1.8.7     2023-06-29 [?] CRAN (R 4.3.0)
   KEGGREST               1.42.0    2023-10-24 [1] Bioconductor
 P knitr                  1.45      2023-10-30 [?] RSPM (R 4.3.0)
 P labeling               0.4.3     2023-08-29 [?] CRAN (R 4.3.0)
   lattice                0.22-5    2023-10-24 [1] RSPM (R 4.3.0)
 P lifecycle              1.0.4     2023-11-07 [?] RSPM (R 4.3.0)
 P limma                * 3.58.1    2023-10-31 [?] Bioconductor
 P locfit                 1.5-9.8   2023-06-11 [?] CRAN (R 4.3.0)
 P magrittr               2.0.3     2022-03-30 [?] CRAN (R 4.3.0)
 P markdown               1.11      2023-10-19 [?] CRAN (R 4.3.1)
 P Matrix                 1.6-2     2023-11-08 [?] RSPM (R 4.3.0)
   MatrixGenerics       * 1.14.0    2023-10-24 [1] Bioconductor
 P matrixStats          * 1.1.0     2023-11-07 [?] RSPM (R 4.3.0)
 P mclust                 6.0.0     2022-10-31 [?] CRAN (R 4.3.0)
 P memoise                2.0.1     2021-11-26 [?] CRAN (R 4.3.0)
   metapod                1.10.0    2023-10-24 [1] Bioconductor
 P munsell                0.5.0     2018-06-12 [?] CRAN (R 4.3.0)
   org.Hs.eg.db           3.18.0    2023-10-31 [1] Bioconductor
   org.Mm.eg.db           3.18.0    2023-10-31 [1] Bioconductor
 P patchwork            * 1.1.3     2023-08-14 [?] RSPM (R 4.3.0)
 P pillar                 1.9.0     2023-03-22 [?] CRAN (R 4.3.0)
 P pkgconfig              2.0.3     2019-09-22 [?] CRAN (R 4.3.0)
 P plyr                   1.8.9     2023-10-02 [?] RSPM (R 4.3.0)
 P png                    0.1-8     2022-11-29 [?] CRAN (R 4.3.0)
 P prettyunits            1.2.0     2023-09-24 [?] RSPM (R 4.3.0)
 P progress               1.2.2     2019-05-16 [?] CRAN (R 4.3.0)
 P R.methodsS3            1.8.2     2022-06-13 [?] CRAN (R 4.3.0)
 P R.oo                   1.25.0    2022-06-12 [?] CRAN (R 4.3.0)
 P R.utils                2.12.2    2022-11-11 [?] CRAN (R 4.3.0)
 P R6                     2.5.1     2021-08-19 [?] CRAN (R 4.3.0)
 P rappdirs               0.3.3     2021-01-31 [?] CRAN (R 4.3.0)
 P Rcpp                   1.0.11    2023-07-06 [?] CRAN (R 4.3.0)
 P RCurl                  1.98-1.13 2023-11-02 [?] RSPM (R 4.3.0)
 P reshape                0.8.9     2022-04-12 [?] CRAN (R 4.3.0)
 P restfulr               0.0.15    2022-06-16 [?] CRAN (R 4.3.0)
 P reticulate             1.34.0    2023-10-12 [?] RSPM (R 4.3.0)
   rhdf5                  2.46.0    2023-10-24 [1] Bioconductor
 P rhdf5filters           1.14.1    2023-11-06 [?] Bioconductor
   Rhdf5lib               1.24.0    2023-10-24 [1] Bioconductor
   Rhtslib                2.4.0     2023-10-24 [1] Bioconductor
 P rjson                  0.2.21    2022-01-09 [?] CRAN (R 4.3.0)
 P rlang                  1.1.2     2023-11-04 [?] RSPM (R 4.3.0)
 P rmarkdown              2.25      2023-09-18 [?] RSPM (R 4.3.0)
 P robustbase             0.99-0    2023-06-16 [?] CRAN (R 4.3.0)
 P rprojroot              2.0.4     2023-11-05 [?] RSPM (R 4.3.0)
   Rsamtools              2.18.0    2023-10-24 [1] Bioconductor
 P RSQLite                2.3.3     2023-11-04 [?] RSPM (R 4.3.0)
 P rstudioapi             0.15.0    2023-07-07 [?] CRAN (R 4.3.0)
 P rsvd                   1.0.5     2021-04-16 [?] CRAN (R 4.3.0)
 P rtracklayer            1.62.0    2023-10-24 [?] Bioconductor
   S4Arrays               1.2.0     2023-10-24 [1] Bioconductor
   S4Vectors            * 0.40.1    2023-10-26 [1] Bioconductor
 P sass                   0.4.7     2023-07-15 [?] CRAN (R 4.3.0)
   ScaledMatrix           1.10.0    2023-10-24 [1] Bioconductor
 P scales               * 1.2.1     2022-08-20 [?] CRAN (R 4.3.0)
   scater               * 1.30.0    2023-10-24 [1] Bioconductor
   scPipe               * 2.2.0     2023-10-24 [1] Bioconductor
   scran                  1.30.0    2023-10-24 [1] Bioconductor
   scuttle              * 1.12.0    2023-10-24 [1] Bioconductor
 P sessioninfo            1.2.2     2021-12-06 [?] CRAN (R 4.3.0)
   SingleCellExperiment * 1.24.0    2023-10-24 [1] Bioconductor
 P SparseArray            1.2.2     2023-11-07 [?] Bioconductor
   sparseMatrixStats      1.14.0    2023-10-24 [1] Bioconductor
 P statmod                1.5.0     2023-01-06 [?] CRAN (R 4.3.0)
 P stringi                1.7.12    2023-01-11 [?] CRAN (R 4.3.0)
 P stringr                1.5.0     2022-12-02 [?] CRAN (R 4.3.0)
   SummarizedExperiment * 1.32.0    2023-10-24 [1] Bioconductor
 P tibble                 3.2.1     2023-03-20 [?] CRAN (R 4.3.0)
 P tidyselect             1.2.0     2022-10-10 [?] CRAN (R 4.3.0)
   utf8                   1.2.4     2023-10-22 [1] RSPM (R 4.3.0)
 P vctrs                  0.6.4     2023-10-12 [?] RSPM (R 4.3.0)
 P vipor                  0.4.5     2017-03-22 [?] CRAN (R 4.3.0)
 P viridis                0.6.4     2023-07-22 [?] RSPM (R 4.3.0)
 P viridisLite            0.4.2     2023-05-02 [?] CRAN (R 4.3.0)
 P withr                  2.5.2     2023-10-30 [?] RSPM (R 4.3.0)
 P xaringanExtra          0.7.0     2022-07-16 [?] CRAN (R 4.3.0)
 P xfun                   0.41      2023-11-01 [?] RSPM (R 4.3.0)
 P XML                    3.99-0.15 2023-11-02 [?] RSPM (R 4.3.0)
 P xml2                   1.3.5     2023-07-06 [?] CRAN (R 4.3.0)
   XVector                0.42.0    2023-10-24 [1] Bioconductor
 P yaml                   2.3.7     2023-01-23 [?] CRAN (R 4.3.0)
   zlibbioc               1.48.0    2023-10-24 [1] Bioconductor

 [1] /stornext/Projects/score/Analyses/G000396_Danu/renv/library/R-4.3/x86_64-pc-linux-gnu
 [2] /stornext/Home/data/allstaff/h/hickey/.cache/R/renv/sandbox/R-4.3/x86_64-pc-linux-gnu/4087f812
 [3] /stornext/System/data/apps/R/R-4.3.1/lib64/R/library

 P  Loaded and on-disk path mismatch.

</code></pre>
</div>
</details>
<div class="sourceCode" id="cb2"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr />
<ol>
<li id="fn1"><p>In the terminology of Bioconductor this is called a <em>SingleCellExperiment</em> object.<a href="#fnref1" class="footnote-back" role="doc-backlink"></a></p></li>
<li id="fn2"><p>The negative control samples have <code>N/A</code> recorded for the various sample metadata.<a href="#fnref2" class="footnote-back" role="doc-backlink"></a></p></li>
<li id="fn3"><p>This file was originally supplied by Danielle Clucas as part of the <code>C122_Clucas</code> project<a href="#fnref3" class="footnote-back" role="doc-backlink"></a></p></li>
<li id="fn4"><p>Allowing up to 1 mismatch<a href="#fnref4" class="footnote-back" role="doc-backlink"></a></p></li>
<li id="fn5"><p>This may appear higher than typical but it is because we did not use an Illumina index when sequencing in <em>S000448</em> and so this includes what would normally be undetermined reads.<a href="#fnref5" class="footnote-back" role="doc-backlink"></a></p></li>
<li id="fn6"><p>I.e. the library size when using read counts<a href="#fnref6" class="footnote-back" role="doc-backlink"></a></p></li>
<li id="fn7"><p>The choice of a 6 bp UMI in the mini-bulk protocol is because it is adapted from SCOREs CEL-Seq2 single-cell RNA-sequencing protocol. With scRNA-seq data it is very rare to observe more than <span class="math inline">\(4^k = 4^6 = 4096\)</span> unique molecules for any one gene.<a href="#fnref7" class="footnote-back" role="doc-backlink"></a></p></li>
<li id="fn8"><p>The number of expressed genes refers to the number of genes which have non-zero counts (ie. they have been identified in the cell at least once)<a href="#fnref8" class="footnote-back" role="doc-backlink"></a></p></li>
</ol>
</section>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
