---
title: "Preprocessing the G000396 mini-bulk data set"
description: |
author:
  - name: Peter Hickey
    url: https://peterhickey.org
    affiliation: Cellular Genomics Projects Team (formerly SCORE), WEHI
    affiliation_url: https://www.wehi.edu.au/people/rory-bowden/4536/wehi-advanced-genomics-facility
  - name: Millie Dunstone
    affiliation: Research Assistant, WEHI
date: "`r Sys.Date()`"
editor_options:
  chunk_output_type: console
---

```{r setup}
library(SingleCellExperiment)
library(here)
library(scales)
library(ggplot2)
library(cowplot)
library(patchwork)
library(ggrepel)
library(gt)

knitr::opts_chunk$set(fig.path = "G000396_Danu.preprocess_files/")

# NOTE: The following is a workaround to the lack of support for tabsets in 
#       distill (see https://github.com/rstudio/distill/issues/11 and 
#       https://github.com/rstudio/distill/issues/11#issuecomment-692142414 in 
#       particular).
xaringanExtra::use_panelset()
```

# Setting up the data

We start from the count matrices created in [`code/scPipe.R`](../code/scPipe.R`), specifically the [UMI counts (a.k.a. UMI-deduplicated) data](../data/SCEs/G000396_Danu.UMI_deduped.scPipe.SCE.rds) and [read counts (a.k.a. not UMI-deduplicated) data](../data/SCEs/G000396_Danu.not_UMI_deduped.scPipe.SCE.rds).

<aside>
UMI stands for unique molecular identifier.
</aside>

```{r}
sce_deduped <- readRDS(
  here("data", "SCEs", "G000396_Danu.UMI_deduped.scPipe.SCE.rds"))
sce_not_deduped <- readRDS(
  here("data", "SCEs", "G000396_Danu.not_UMI_deduped.scPipe.SCE.rds"))

colnames(sce_not_deduped) <- sub(
  "\\.not_UMI_deduped",
  "",
  colnames(sce_not_deduped))
stopifnot(
  identical(rownames(sce_deduped), rownames(sce_not_deduped)),
  identical(colnames(sce_deduped), colnames(sce_not_deduped))) 

# Combine UMI and read counts a single SCE.
sce <- sce_deduped
assay(sce, "UMI_counts") <- assay(sce_deduped, "counts")
assay(sce, "read_counts") <- assay(sce_not_deduped, "counts")
# NOTE: Nullify some now-unrequired data.
assay(sce, "counts") <- NULL
sce$UMI_deduped <- NULL
```

We combine these into a single object^[In the terminology of Bioconductor this is called a *SingleCellExperiment* object.] containing both the UMI counts and the read counts for `r number(nrow(sce), big.mark = ",")` genes and `r number(ncol(sce), big.mark = ",")` samples.

## Experimental design

```{r}
# Initial tidying of sample metadata
sce$plate_number <- factor(sce$plate_number)
sce$well_position <- factor(
  sce$well_position,
  unlist(lapply(LETTERS[1:16], function(x) paste0(x, 1:24))))
sce$sample_type <- factor(sce$sample_type)
sce$original_plate <- factor(sce$original_plate)
sce$day <- factor(
  sub("Day ", "Day", sce$day), 
  levels = c(paste0("Day", c(3, 6, 9, 12)), "N/A"))
sce$sample <- factor(sce$sample)
sce$cell_line <- factor(
  sapply(strsplit(as.character(sce$sample), "_"), "[[", 1),
  levels = c("GID1KO", "GID2KO", "GID7KO", "GID8KO", "GID9KO", "WT", "N/A"))
sce$rep <- sub(".* ", "", as.character(sce$sample))
sce$rep <- factor(sce$rep, levels = c(sort(as.integer(unique(sce$rep))), "N/A"))
sce$technical_replicate <- factor(sce$technical_replicate)
sce$sequencing_run <- factor(sce$sequencing_run)

# TODO: A 'real' sample ID would by <cell_line>.<day>.<rep>
# TODO: Remove `old_sample` once I'm confident it's no longer required.
sce$old_sample <- sce$sample
sce$sample <- interaction(
  sce$cell_line, 
  sce$day, 
  sce$rep, 
  drop = TRUE, 
  lex.order = TRUE)
```

### Introduction

This is a large and somewhat complex experiment.
There are `r number(sum(sce$sample_type != "empty"), big.mark = ",")` mini-bulk libraries and `r number(sum(sce$sample_type == "empty"), big.mark = ",")` negative control 'empty' libraries^[The negative control samples have `N/A` recorded for the various sample metadata.].
The tables below summarise the samples in this experiment.

::::: {.panelset}

::: {.panel}

#### Technical replicates

```{r}
# TODO: Why are there 4 reps of each of GID7KO.12.Day12, GID8KO.12.Day12, and 
#       GID9KO.12.Day12? Have emailed Danu to clarify (2023-11-10).
tech_reps_tbl <- as.data.frame(colData(sce)) |>
  dplyr::group_by(cell_line, day, rep) |>
  dplyr::count() |> 
  dplyr::group_by(cell_line, day)

gt(
  tech_reps_tbl,
  caption = md(
    "Number of technical replicates per `cell_line`-`day`-`rep` combination.")) |>
  fmt_number(columns = n, decimals = 0) |>
  summary_rows(
    columns = n, 
    fns = list(subtotal ~ sum(.)), 
    fmt = ~ fmt_number(., decimals = 0)) |>
  grand_summary_rows(
    columns = n, 
    fns = total ~ sum(.), 
    fmt = ~ fmt_number(., decimals = 0)) |>
  tab_options(
    summary_row.background.color = "lightgrey",
    grand_summary_row.background.color = "darkgrey")
```

:::

::: {.panel}

#### Biological replicates (per cell line)

```{r}
bio_reps_tbl <- as.data.frame(colData(sce)) |>
  dplyr::select(cell_line, day, rep) |>
  dplyr::group_by(cell_line, day) |>
  dplyr::distinct() |>
  dplyr::count() |>
  dplyr::group_by(cell_line)

gt(
  bio_reps_tbl,
  caption = md(
    "Number of biological replicates per `cell_line`-`day` combination.")) |>
  fmt_number(columns = n, decimals = 0) |>
  summary_rows(
    columns = n, 
    fns = list(subtotal ~ sum(.)), 
    fmt = ~ fmt_number(., decimals = 0)) |>
  grand_summary_rows(
    columns = n, 
    fns = total ~ sum(.), 
    fmt = ~ fmt_number(., decimals = 0)) |>
  tab_options(
    summary_row.background.color = "lightgrey",
    grand_summary_row.background.color = "darkgrey")
```

:::

::: {.panel}

#### Biological replicates (per day)

```{r}
bio_reps_tbl <- as.data.frame(colData(sce)) |>
  dplyr::select(day, cell_line, rep) |>
  dplyr::group_by(day, cell_line) |>
  dplyr::distinct() |>
  dplyr::count() |>
  dplyr::group_by(day)

gt(
  bio_reps_tbl,
  caption = md(
    "Number of biological replicates per `cell_line`-`day` combination.")) |>
  fmt_number(columns = n, decimals = 0) |>
  summary_rows(
    columns = n, 
    fns = list(subtotal ~ sum(.)), 
    fmt = ~ fmt_number(., decimals = 0)) |>
  grand_summary_rows(
    columns = n, 
    fns = total ~ sum(.), 
    fmt = ~ fmt_number(., decimals = 0)) |>
  tab_options(
    summary_row.background.color = "lightgrey",
    grand_summary_row.background.color = "darkgrey")
```

:::

:::::

Note that all samples have been run in technical duplicate.
The `Day3` and `Day6` samples were originally plated together on one 96-well plate (`Plate A`) while the `Day 9` and `Day 12` were originally plated together on another 96-well plate (`Plate B`).

We will retain the negative control 'empty' libraries mini-bulk libraries for QC purposes.

## Incorporating gene-based annotation

We obtain gene-based annotations, such as the chromosome, using the information in the GFF file, `PlasmoDB-51_Pfalciparum3D7.gff`^[This file was originally supplied by Danielle Clucas as part of the `C122_Clucas` project].

**TODO** Move/remove these notes?

- The GFF seemingly doesn't include some simple/useful information like gene name or symbol. 
  - E.g., `PF3D7_0222700` is `RIF` or `rifin` according to https://plasmodb.org/plasmo/app/search?q=PF3D7_0222700 but that's not in the GFF file (actually, it is in the value of the `description` tag in the `attribute` field but the `makeTxDbFromGFF()` doesn't parse this field).
- I tried using a BioC resource, but there are many features in the SCE with the same `GENENAME`, e.g., `rowData(sce)[which(any(rowData(sce)$GENENAME == "rifin (RIF)")), ]`.

```{r}
ids <- rownames(sce)

library(GenomicFeatures)

txdb <- makeTxDbFromGFF(
  here("extdata/PlasmoDB-51_Pfalciparum3D7/PlasmoDB-51_Pfalciparum3D7.gff"))

txdb_columns <- columns(txdb)
txdb_columns <- setNames(txdb_columns, txdb_columns)
txdb_df <- DataFrame(
  lapply(txdb_columns, function(column) {
    mapIds(
      x = txdb, 
      # NOTE: Need to remove gene version number prior to lookup.
      keys = gsub("\\.[0-9]+$", "", ids),
      keytype = "GENEID",
      column = column,
      multiVals = "CharacterList")
  }),
  row.names = ids)

# Pull out useful gene-based annotations from the database.
library(org.Pf.plasmo.db)

# NOTE: These columns were customised for this project.
db_columns <- c("GENENAME", "SYMBOL")
names(db_columns) <- db_columns
stopifnot(all(db_columns %in% columns(org.Pf.plasmo.db)))
db_df <- DataFrame(
  lapply(db_columns, function(column) {
    mapIds(
      x = org.Pf.plasmo.db, 
      # NOTE: Need to remove gene version number prior to lookup.
      keys = gsub("\\.[0-9]+$", "", ids),
      keytype = "ORF",
      column = column,
      multiVals = "CharacterList")
  }),
  row.names = ids)

rowData(sce) <- cbind(txdb_df, db_df)
```

```{r}
# Some useful gene sets
mito_set <- rownames(sce)[any(rowData(sce)$TXCHROM == "Pf3D7_MIT_v3")]
pseudogene_set <- rownames(sce)[
  any(rowData(sce)$TXTYPE == "pseudogenic_transcript")]
ribo_set <- union(
  rownames(sce)[any(rowData(sce)$TXTYPE == "rRNA")],
  # NOTE: Include mRNA that are ribosomal protein genes.
  rownames(sce)[
    any(grepl("ribosomal", rowData(sce)$GENENAME)) &
    any(grepl("protein", rowData(sce)$GENENAME))])
```

